---
title: "Epigenetic Correlations - Generation Scotland"
author: "Alesha Hatton"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  epuRate::PCTG:
    toc: TRUE
    number_sections: FALSE
    code_folding: "hide"
editor_options: 
  chunk_output_type: console
---

```{r, echo=FALSE, message=FALSE, warning=FALSE}
# You need these libraries to run this template:
library(rmarkdown)    # install.packages("rmarkdown")
library(epuRate)      # devtools::install_github("holtzy/epuRate", force=TRUE)
library(tidyverse)  
library(knitr)
library(stringr)
library(readr)

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(eval = FALSE)


mycolours <- c("#00AFBB", "#E7B800", "#FC4E07", "#0072B2", "#009E73",
          "#F0E442", "#D55E00", "#CC79A7") 



```

<br>

> Epigenetic Correlations - Generation Scotland pipeline

# Overview

------------------------------------------------------------------------

<br><br>

Cluster login

ssh -L 8017:cfw:8017 v1ahatt4\@ccace-sshgate.psy.ed.ac.uk

ssh -p 8017 v1ahatt4\@localhost

scp -P 8017 v1ahatt4\@localhost

```{r}

du -d 1 -h
top -H

```

<br> <br>

#### Data directories

OSCA files=Local_Data/methylation/GS_20k/OSCA_files

scratch=/Local_Scratch/Alesha

<br><br>

#### Data description

-   18,349 individuals

-   831,733 DNAm probes

-   3 Waves

    -   Wave 1 = 5,087
    -   Wave 3 = 4,450
    -   Wave 4 = 8,876

<br>

#### Phenotypes

-   Body Mass Index (kg/m\^2)

-   Waist / Hip Ratio

-   Body Fat Composition (bio-impedance) (%)

-   Glucose (mmol/l)

-   HDL cholesterol (mmol/l)

-   Total cholesterol (mmol/l)

-   Height (control)

<br><br>

# Quality Control

------------------------------------------------------------------------

**Samples with covariate information**

n = 18377

```{r Covariates}
library(tidyr)
library(dplyr)

data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")
# wave1 wave3 wave4 
# 5087  4450  8876 

# Fam
grm.id <- read.table("/Local_Scratch/Alesha/unrelated/GS_GWAS.grm.id", header=F)
colnames(grm.id) <- c("FID", "IID")

# OSCA ID, V1==V2
oii <- read.table("/Local_Data/methylation/GS_20k/OSCA_files/mvals-norm20k-18413-831733.oii", header=F)
colnames(oii) <- c("FID_DNAm", "IID_DNAm", "V3", "V4", "V5")

# Withdrawn
withdrawn <- read.table("/Local_Scratch/Alesha/unrelated/withdrawals_v3.txt", header=F)
colnames(withdrawn) <- c("FID", "IID")

# Phenotypes
# /Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/body.csv. 
# They are labelled as bmi, whr and body_fat there. 

# /Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/biochemistry.csv. 
# These are glucose, HDL cholesterol and Total cholesterol. 

body <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/body.csv", header=T, sep=",")
biochemistry <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/biochemistry.csv", header=T, sep=",")

# Merge
ds <- oii %>% 
select(FID_DNAm, IID_DNAm) %>%
inner_join(data, by=c("FID_DNAm"="Sample_Sentrix_ID")) %>%
inner_join(grm.id, by=c("Sample_Name"="IID")) %>%
left_join(body, by=c("Sample_Name"="id")) %>%
left_join(biochemistry, by=c("Sample_Name"="ID")) %>% 
filter(!Sample_Name %in% withdrawn$IID) %>%
select(FID_DNAm, IID_DNAm, FID, IID=Sample_Name, Sample_Name, everything())

# 18349 with DNAm 

write.table(ds, file="/Local_Scratch/Alesha/covariates/oii_cov.txt", col.name=T, row.name=F, quote=F)
```

<br>

### Remove related individuals

```{bash Related individuals}

mkdir /Local_Scratch/Alesha/unrelated
gcta64 \
--bfile /Cluster_Filespace/Marioni_Group/GS/GS_GWAS/GS_GWAS \
--remove /Local_Scratch/Alesha/unrelated/withdrawals_v3.txt \
--make-grm \
--out /Local_Scratch/Alesha/unrelated/GS_GWAS
#   19989 samples, 561125 markers, 199790055 GRM elements

gcta64 --grm /Local_Scratch/Alesha/unrelated/GS_GWAS \
 --grm-cutoff 0.05 \
 --make-grm \
 --out GS_GWAS_0.05
# After pruning the GRM, there are 8041 individuals (11948 individuals removed).

# However we are going to select only participants with DNAm first to make sure we dont loose more people than necessary
# Check unrelated doesnt change numbers when filtering for DNAm ind first

awk 'NR>1 {print $3, $4}' /Local_Scratch/Alesha/covariates/oii_cov.txt > /Local_Scratch/Alesha/covariates/oii_cov_FID_IID.txt
# 0.05 cut off for those with DNAm first
gcta64 --grm /Local_Scratch/Alesha/unrelated/GS_GWAS \
 --keep /Local_Scratch/Alesha/covariates/oii_cov_FID_IID.txt \
 --grm-cutoff 0.05 \
 --make-grm \
 --out /Local_Scratch/Alesha/unrelated/GS_GWAS_DNAm_0.05
# After pruning the GRM, there are 7758 individuals (10618 individuals removed).
# Note: Resulted in more individuals this way by ~ 300.


# Check unrelated makes sense
# 1. Set by waves and determine unrelated
awk 'NR>1 &&  $7=="wave4" {print $3, $4}' oii_cov.txt > wave4_FID.txt
awk 'NR>1 &&  $7=="wave3" {print $3, $4}' oii_cov.txt > wave3_FID.txt
awk 'NR>1 &&  $7=="wave1" {print $3, $4}' oii_cov.txt > wave1_FID.txt

gcta64 --grm /Local_Scratch/Alesha/unrelated/GS_GWAS \
 --keep /Local_Scratch/Alesha/covariates/wave1_FID.txt \
 --grm-cutoff 0.05  --make-grm  --out GS_GWAS_DNAm_0.05_wave1

gcta64 --grm /Local_Scratch/Alesha/unrelated/GS_GWAS \
 --keep /Local_Scratch/Alesha/covariates/wave3_FID.txt \
 --grm-cutoff 0.05  --make-grm  --out GS_GWAS_DNAm_0.05_wave3

 gcta64 --grm /Local_Scratch/Alesha/unrelated/GS_GWAS \
 --keep /Local_Scratch/Alesha/covariates/wave4_FID.txt \
 --grm-cutoff 0.05  --make-grm  --out GS_GWAS_DNAm_0.05_wave4

#  7758 GS_GWAS_DNAm_0.05.grm.id
#  2710 GS_GWAS_DNAm_0.05_wave1.grm.id
#  4444 GS_GWAS_DNAm_0.05_wave3.grm.id
#  4267 GS_GWAS_DNAm_0.05_wave4.grm.id

# However when run all together
# wave1 wave3 wave4 
# 2035  4276  1435 
```

7,758 unrelated individuals

-   wave1 - 2,035
-   wave3 - 4,276
-   wave4 - 1,435

```{r Remove related from covariates}
# Check how many are unrelated from each wave individually
# 
# -   wave1 - 2710
# -   wave3 - 4444
# -   wave4 - 4267

# Related with DNAm
 # wave1 wave3 wave4 
 # 5080  4445  8851 

# Looks like almost everyone in wave4 was related or related to individuals in previous waves

ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)
unrelated <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_DNAm_0.05.grm.id", header=F)
colnames(unrelated) <- c("FID", "IID")

ds_unrelated <- ds %>%
inner_join(unrelated, by=c("FID", "IID")) 
# [1] 7758   22 # Unrelated with DNAm and covariates

write.table(ds_unrelated, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_unrelated.txt", col.name=T, row.name=F, quote=F)
```

<br>

### Phenotypes

```{r Phenotype QC}
# For each trait, we adjusted the phenotype for age in each gender group and standardized the residuals by rank-based inverse normal transformation, which removed the age effect and potential difference in mean and variance between two gender groups. There was no adjustment for wave as there were minimal differences in the mean across waves.

# -   BMI
# -   Waist-to-hip ratio
# -   Body fat %
# -   Biochemical traits
#     -   Glucose
#     -   HDL cholesterol
#     -   Total cholesterol
# Other traits available include
# 
# -   Sodium
# -   Potassium
# -   Urea
# -   Creatinine
# -   Creat_mgdl

library(RNOmni) 
library(tidyverse)
# Unrelated
ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_unrelated.txt", header=T)
# Related
# ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)


summary(ds)
dim(ds)

# Look at the data
library(psych) 
describe(ds %>% select(-FID_DNAm, -IID_DNAm, -IID, -FID, -Set, -Batch, -sex)) %>% 
  select(n, mean, median, sd, min, max, range)

# Plot each var
ds %>% str

t_ds_numeric_sex <- reshape2::melt(ds %>% select(FID_DNAm, IID_DNAm, FID, IID, sex, where(is.numeric), where(is.integer)), 
                               id = c('FID_DNAm','IID_DNAm', 'FID', 'IID', "sex"))

t_ds_numeric_Set <- reshape2::melt(ds %>% select(FID_DNAm, IID_DNAm, FID, IID, Set, where(is.numeric), where(is.integer)), 
                               id = c('FID_DNAm','IID_DNAm', 'FID', 'IID', "Set"))

# By sex
p1 <- t_ds_numeric_sex %>% 
  filter(variable %in% c("age", "bmi", "whr", "body_fat", "Glucose", "HDL_cholesterol", "Total_cholesterol", "height", "waist", "hips")) %>%  
  ggplot(aes(x = value, fill=sex)) +
  geom_density(alpha=0.5)  +
  facet_wrap( ~ variable, ncol = 4,  scales = "free")

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/density_by_sex.png", width = 9, height = 6, dpi = 150, units = "in", device='png')

# By Set
p1 <- t_ds_numeric_Set  %>% 
  filter(variable %in% c("age", "bmi", "whr", "body_fat", "Glucose", "HDL_cholesterol", "Total_cholesterol", "height", "waist", "hips")) %>% 
  ggplot(aes(x = value, fill=Set)) +
  geom_density(alpha=0.5)  +
  facet_wrap( ~ variable, ncol = 4,  scales = "free")

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/density_by_Set.png", width = 9, height = 6, dpi = 150, units = "in", device='png')


# Mean differences by set?
ds_Set <-   ds %>% group_by(Set) %>% # or Batch?
  summarise(sex_M = sum(sex == "M")/sum(!is.na(sex)) * 100,
            sex_F = sum(sex=="F")/sum(!is.na(sex)) * 100) %>%
  left_join(
    ds %>% group_by(Set) %>% # or Batch?
      select(-FID_DNAm, -IID_DNAm, -FID, -IID, -Batch) %>%
      summarise_if(is.numeric, list(mean = mean), na.rm = TRUE), 
    by=c("Set")) %>% as.data.frame()
# Minimal effect of set

ds_Set %>% reshape2::melt(id = c("Set")) %>% filter(Set=="wave1") %>% rename(wave1=value) %>% select(-Set) %>%
  left_join(
    ds_Set %>% reshape2::melt(id = c("Set")) %>% filter(Set=="wave3") %>% rename(wave3=value) %>% select(-Set),
            by=c("variable")) %>%
  left_join(
    ds_Set %>% reshape2::melt(id = c("Set")) %>% filter(Set=="wave4") %>% rename(wave4=value) %>% select(-Set),
            by=c("variable"))

# Large effect of Sex: Within sex, adjust for age
ds_Sex <- 
  ds %>% group_by(sex) %>% 
      select(-FID_DNAm, -IID_DNAm, -FID, -IID, -Batch, -Set) %>%
      summarise_if(is.numeric, list(mean = mean), na.rm = TRUE) %>% 
      as.data.frame()

ds_Sex %>% reshape2::melt(id = c("sex")) %>% filter(sex=="M") %>% rename(Male=value) %>% select(-sex) %>%
  left_join(
    ds_Sex %>% reshape2::melt(id = c("sex")) %>% filter(sex=="F") %>% rename(Female=value) %>% select(-sex),
            by=c("variable")) 

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/density_by_*.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/pheno_QC

```

| Trait             | n    | mean   | median | sd    | min    | max    | range  |
|-------------------|------|--------|--------|-------|--------|--------|--------|
| age               | 7758 | 51.57  | 53.58  | 13.23 | 17.92  | 93.33  | 75.42  |
| sex\*             | 7758 | 1.43   | 1.00   | 0.50  | 1.00   | 2.00   | 1.00   |
| height            | 7729 | 167.99 | 167.50 | 9.52  | 135.00 | 203.00 | 68.00  |
| weight            | 7713 | 76.22  | 74.40  | 16.31 | 34.30  | 183.90 | 149.60 |
| bmi               | 7711 | 26.94  | 26.18  | 5.09  | 14.19  | 71.35  | 57.16  |
| waist             | 7650 | 90.13  | 90.00  | 13.76 | 51.00  | 167.00 | 116.00 |
| hips              | 7649 | 103.42 | 103.00 | 10.37 | 30.80  | 180.90 | 150.10 |
| whr               | 7645 | 0.87   | 0.87   | 0.09  | 0.52   | 2.79   | 2.27   |
| body_fat          | 7555 | 30.41  | 30.10  | 9.30  | 2.00   | 57.20  | 55.20  |
| Glucose           | 7601 | 4.88   | 4.70   | 1.15  | 1.40   | 32.00  | 30.60  |
| HDL_cholesterol   | 7687 | 1.47   | 1.40   | 0.42  | 0.50   | 5.10   | 4.60   |
| Total_cholesterol | 7700 | 5.21   | 5.20   | 1.08  | 0.90   | 12.00  | 11.10  |

![Density by sex](data/pheno_QC/density_by_sex.png)

![Density by set](data/pheno_QC/density_by_Set.png)

<br>

### Removing outliers

Biochemical and anthropometric traits were trimmed for outliers (values that were ± 4 SDs from the mean). In addition BMI was trimmed for extreme values at \<17 and \>50 kg/m2.

```{r outliers}
library(RNOmni) 
library(tidyverse)
ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_unrelated.txt", header=T)
# ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)

# 1. Remove outliers - as per Robs papers (4SD from the mean except for BMI (17-50))
ds_outliers <- ds %>% 
  mutate(
    across(
    .cols = c("height", "weight", "bmi", "waist", "hips", "whr", "body_fat", "Glucose", "HDL_cholesterol", "Total_cholesterol"),
    .fns = list(mean4sd_up = ~ mean(.x, na.rm=T) + 4*sqrt(var(.x, na.rm=T)),
                mean4sd_low = ~ mean(.x, na.rm=T) - 4*sqrt(var(.x, na.rm=T))
                ))) 
ds_outliers %>% select(ends_with("mean4sd_up"), ends_with("mean4sd_low")) %>% unique()

ds_clean <- ds_outliers %>%
  mutate(bmi = case_when(bmi >= 17 & bmi <= 50  ~ bmi,    TRUE ~ as.numeric(NA)),
         height = case_when(height <= height_mean4sd_up & height >= height_mean4sd_low ~ height,    TRUE ~ as.numeric(NA)),
         weight = case_when(weight <= weight_mean4sd_up & weight >= weight_mean4sd_low ~ weight,    TRUE ~ as.numeric(NA)),
         waist = case_when(waist <= waist_mean4sd_up & waist >= waist_mean4sd_low ~ waist,    TRUE ~ as.numeric(NA)),
         hips = case_when(hips <= hips_mean4sd_up & hips >= hips_mean4sd_low ~ hips,    TRUE ~ as.numeric(NA)),
         whr = case_when(whr <= whr_mean4sd_up & whr >= whr_mean4sd_low ~ whr,    TRUE ~ as.numeric(NA)),
         body_fat = case_when(body_fat <= body_fat_mean4sd_up & body_fat >= body_fat_mean4sd_low ~ body_fat,    TRUE ~ as.numeric(NA)),
         Glucose = case_when(Glucose <= Glucose_mean4sd_up & Glucose >= Glucose_mean4sd_low ~ Glucose,    TRUE ~ as.numeric(NA)),
         HDL_cholesterol = case_when(HDL_cholesterol <= HDL_cholesterol_mean4sd_up & HDL_cholesterol >= HDL_cholesterol_mean4sd_low ~ HDL_cholesterol,    TRUE ~ as.numeric(NA)),
         Total_cholesterol = case_when(Total_cholesterol <= Total_cholesterol_mean4sd_up & Total_cholesterol >= Total_cholesterol_mean4sd_low ~ Total_cholesterol,    TRUE ~ as.numeric(NA)))

# 2. Adjust each phenotype for age withine ach sex and rank normalise

ds_bmi <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, bmi) %>%
    group_by(sex) %>% 
    filter(!is.na(bmi)) %>%
    mutate(bmi_RankNorm = residuals(lm(bmi ~ age, na.action = na.exclude)) %>% RankNorm)

ds_whr <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, whr) %>%
    group_by(sex) %>% 
    filter(!is.na(whr)) %>%
    mutate(whr_RankNorm = residuals(lm(whr ~ age, na.action = na.exclude)) %>% RankNorm)

ds_body_fat <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, body_fat) %>%
    group_by(sex) %>% 
    filter(!is.na(body_fat)) %>%
    mutate(body_fat_RankNorm = residuals(lm(body_fat ~ age, na.action = na.exclude)) %>% RankNorm)

ds_Glucose <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, Glucose) %>%
    group_by(sex) %>% 
    filter(!is.na(Glucose)) %>%
    mutate(Glucose_RankNorm = residuals(lm(Glucose ~ age, na.action = na.exclude)) %>% RankNorm)

ds_HDL_cholesterol <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, HDL_cholesterol) %>%
    group_by(sex) %>% 
    filter(!is.na(HDL_cholesterol)) %>%
    mutate(HDL_cholesterol_RankNorm = residuals(lm(HDL_cholesterol ~ age, na.action = na.exclude)) %>% RankNorm)

 ds_Total_cholesterol <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, Total_cholesterol) %>%
    group_by(sex) %>% 
    filter(!is.na(Total_cholesterol)) %>%
    mutate(Total_cholesterol_RankNorm = residuals(lm(Total_cholesterol ~ age, na.action = na.exclude)) %>% RankNorm)
 
ds_Height <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, height) %>%
    group_by(sex) %>% 
    filter(!is.na(height)) %>%
    mutate(height_RankNorm = residuals(lm(height ~ age, na.action = na.exclude)) %>% RankNorm)

ds_weight <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, weight) %>%
    group_by(sex) %>% 
    filter(!is.na(weight)) %>%
    mutate(weight_RankNorm = residuals(lm(weight ~ age, na.action = na.exclude)) %>% RankNorm)
  
ds_Waist <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, waist) %>%
    group_by(sex) %>% 
    filter(!is.na(waist)) %>%
    mutate(waist_RankNorm = residuals(lm(waist ~ age, na.action = na.exclude)) %>% RankNorm)
   
ds_Hips <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, hips) %>%
    group_by(sex) %>% 
    filter(!is.na(hips)) %>%
    mutate(hips_RankNorm = residuals(lm(hips ~ age, na.action = na.exclude)) %>% RankNorm)

# Merge back together
ds_pheno <- ds_clean %>% 
left_join(ds_bmi, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "bmi")) %>%
left_join(ds_whr, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "whr")) %>%
left_join(ds_body_fat, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "body_fat")) %>%
left_join(ds_Glucose, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "Glucose")) %>%
left_join(ds_HDL_cholesterol, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "HDL_cholesterol")) %>%
left_join(ds_Total_cholesterol, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "Total_cholesterol")) %>%
left_join(ds_Height, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "height")) %>%
left_join(ds_Waist, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "waist")) %>%
left_join(ds_Hips, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "hips")) %>%
left_join(ds_weight, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "weight")) %>%
select(FID_DNAm, IID_DNAm, age, sex, Set, bmi, bmi_RankNorm, whr, whr_RankNorm, body_fat, body_fat_RankNorm, Glucose, Glucose_RankNorm,  HDL_cholesterol, HDL_cholesterol_RankNorm, Total_cholesterol, Total_cholesterol_RankNorm, height, height_RankNorm, weight, weight_RankNorm, waist, waist_RankNorm, hips, hips_RankNorm)

write.table(ds_pheno, file="/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", col.name=T, row.name=F, quote=F)

# Split for correlation analysis
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)

# Plot again
t_ds_numeric_sex <- reshape2::melt(ds_pheno %>% select(FID_DNAm, IID_DNAm, sex, where(is.numeric), where(is.integer)), 
                               id = c('FID_DNAm','IID_DNAm', "sex"))

t_ds_numeric_Set <- reshape2::melt(ds_pheno %>% select(FID_DNAm, IID_DNAm, Set, where(is.numeric), where(is.integer)), 
                               id = c('FID_DNAm','IID_DNAm', "Set"))

# By sex
library(sjmisc)

t_ds_numeric_sex <- t_ds_numeric_sex %>% mutate(Method = case_when(grepl("RankNorm", variable) ~ "RINT",
                                                                   TRUE ~ "Raw"),
                                                Trait = 
                                                  case_when (variable %in% c("bmi", "bmi_RankNorm") ~ "bmi",
                                                             variable %in% c("body_fat", "body_fat_RankNorm") ~ "body_fat",
                                                             variable %in% c("whr", "whr_RankNorm") ~ "whr",
                                                             variable %in% c("Glucose", "Glucose_RankNorm") ~ "Glucose",
                                                   variable %in% c("HDL_cholesterol", "HDL_cholesterol_RankNorm") ~ "HDL_cholesterol",
                                                   variable %in% c("Total_cholesterol", "Total_cholesterol_RankNorm") ~ "Total_cholesterol"))

t_ds_numeric_sex$Trait <- recode_factor( t_ds_numeric_sex$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol")

p1 <- t_ds_numeric_sex %>% na.omit() %>%  ggplot(aes(x = value, fill=sex)) +
  geom_density(alpha=0.5)  +
  scale_fill_manual(values = c(mycolours[2], mycolours[1])) +
  facet_wrap( Method ~ Trait , scales = "free")

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/norm_density_by_sex_outliers4sd.png")

# By Set
p1 <- t_ds_numeric_Set  %>%  ggplot(aes(x = value, fill=Set)) +
  geom_density(alpha=0.5)  +
  facet_wrap( ~ variable, ncol = 4,  scales = "free")

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/norm_density_by_Set_outliers4sd.png")


scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/related_norm*.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/pheno_QC

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/pheno_QC


awk '{print $1, $2, $3, $4}' oii_cov.txt > iid.txt
scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/iid.txt /Users/uqahatt2/Documents/2_project/4_GS/3_height/BayesRR/data

```

### Adjust and normalise

For each trait, we adjusted the phenotype for age in each gender group and standardized the residuals by RINT which removed the age effect and potential difference in mean and variance between two gender groups. There was no adjustment for wave as there were minimal differences in the mean across waves.

![Normalised phenotypes by sex](data/pheno_QC/norm_density_by_sex_outliers4sd.png)

![Normalised phenotypes by set](data/pheno_QC/norm_density_by_Set_outliers4sd.png)

<br>

```{r Additonal phenotypes}
# /Cluster_Filespace/Marioni_Group/GS/GS_dataset/

# Desired traits
# T2D, years of education, Ever smoke, Pack years, alcohol consumption, heart disease (CAD), blood pressure, bipolar, RA, Despression
# seven biochemical traits (creatinine, glucose, high-density lipoprotein cholesterol, potassium, sodium, total cholesterol and urea)  
# ten complex traits (alcohol consumption, body fat percentage, body mass index, diastolic blood pressure, forced expiratory volume in one second (FEV), forced vital capacity (FVC), heart rate (average beats/minute), smoking pack years, systolic blood pressure, waist-to-hip ratio)
 # ever_smoke,  pack_years,  education_years, alcohol_units, heart_disease, high_BP, diabetes, depression, osteo_arthritis, rheum_arthritis, asthma, avg_sys, avg_dia, avg_hr, FEV, FVC, T2D

library(RNOmni) 
library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")
ever_smoke <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/ever_smoke.csv", header=T, sep=",")
pack_years <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/pack_years.csv", header=T, sep=",")

diabetes <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/diabetes.csv", header=T, sep=",")
education <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/PCQ/education.csv", header=T, sep=",")
alcohol <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/PCQ/alcohol.csv", header=T, sep=",")
disease <- data.table::fread("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/PCQ/disease.csv", header=T, sep=",", fill=TRUE)
BPHR <- data.table::fread("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/BPHR.csv", header=T, sep=",", fill=TRUE)
spirometry <- data.table::fread("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/clinical/spirometry.csv", header=T, sep=",", fill=TRUE)

data_pheno <- data %>% select(Sample_Name, age, sex, Set, Batch, Sample_Sentrix_ID) %>% 
                       # inner_join(ds_pheno %>% select(FID_DNAm, bmi, bmi_RankNorm), by=c("Sample_Sentrix_ID"="FID_DNAm")) %>%
                       left_join(ever_smoke, by=c("Sample_Name")) %>%
                       left_join(pack_years, by=c("Sample_Name")) %>%
                       left_join(diabetes %>% select(id, diabetes=tname), by=c("Sample_Name"="id")) %>%
                       left_join(education %>% select(id, education_years=years), by=c("Sample_Name"="id")) %>%
                       left_join(alcohol %>% select(id, alcohol_units=units), by=c("Sample_Name"="id")) %>%
                       left_join(disease %>% select(ID, heart_disease_Y, high_BP_Y, diabetes_Y, alzheimers_Y, parkinsons_Y, depression_Y, bowel_cancer_Y,lung_cancer_Y, prostate_cancer_Y,  hip_fracture_Y,  osteo_arthritis_Y,  rheum_arthritis_Y, asthma_Y), by=c("Sample_Name"="ID")) %>%
                        left_join(BPHR %>% select(id, avg_sys, avg_dia, avg_hr), by=c("Sample_Name"="id")) %>%
                        left_join(spirometry %>% select(id, FEV, FVC), by=c("Sample_Name"="id"))


data_pheno$ever_smoke <- data_pheno$ever_smoke %>% as.factor
data_pheno$diabetes <- data_pheno$diabetes %>% as.factor
data_pheno$heart_disease_Y <- data_pheno$heart_disease_Y %>% as.factor
data_pheno$high_BP_Y <- data_pheno$high_BP_Y %>% as.factor
data_pheno$diabetes_Y <- data_pheno$diabetes_Y %>% as.factor
data_pheno$alzheimers_Y <- data_pheno$alzheimers_Y %>% as.factor
data_pheno$parkinsons_Y <- data_pheno$parkinsons_Y %>% as.factor
data_pheno$depression_Y <- data_pheno$depression_Y %>% as.factor
data_pheno$bowel_cancer_Y <- data_pheno$bowel_cancer_Y %>% as.factor
data_pheno$lung_cancer_Y <- data_pheno$lung_cancer_Y %>% as.factor
data_pheno$prostate_cancer_Y <- data_pheno$prostate_cancer_Y %>% as.factor
data_pheno$hip_fracture_Y <- data_pheno$hip_fracture_Y %>% as.factor
data_pheno$osteo_arthritis_Y <- data_pheno$osteo_arthritis_Y %>% as.factor
data_pheno$rheum_arthritis_Y <- data_pheno$rheum_arthritis_Y %>% as.factor
data_pheno$asthma_Y <- data_pheno$asthma_Y %>% as.factor

cor(data_pheno$bmi, data_pheno$high_BP_Y, use="complete.obs") # 0.2146242
cor(data_pheno$bmi, data_pheno$diabetes_Y, use="complete.obs") # 0.1577002
cor(data_pheno$bmi, data_pheno$avg_sys, use="complete.obs") # 0.2630081
cor(data_pheno$bmi, data_pheno$avg_dia, use="complete.obs") # 0.3318025
cor(data_pheno$bmi, data_pheno$avg_hr, use="complete.obs") # 0.1080104

data_pheno <- data_pheno %>% mutate(T2D = case_when(diabetes=="Type 2" ~ 1,
                                                   TRUE ~ 0),
                                    FID_DNAm=Sample_Sentrix_ID, IID_DNAm=Sample_Sentrix_ID) %>%
                                      select(-diabetes, -alzheimers_Y, -parkinsons_Y, -bowel_cancer_Y,
                                            -lung_cancer_Y, -prostate_cancer_Y, -hip_fracture_Y, -Sample_Sentrix_ID)  %>%# Too low numbers < 50
                              rename_at(.vars = vars(ends_with("_Y")), ~str_remove(., "_Y")) 

write.table(data_pheno, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/additional_phenotypes.txt", col.name=T, row.name=F, quote=F)

t_ds_numeric_sex <- reshape2::melt(data_pheno %>% select(Sample_Name,  sex, where(is.numeric), where(is.integer)), 
                               id = c('Sample_Name', "sex"))

p1 <- t_ds_numeric_sex %>% 
  filter(variable %in% c("age", "pack_years", "education_years", "alcohol_units", "avg_sys", "avg_dia", "avg_hr",  "FEV",  "FVC")) %>%  
  ggplot(aes(x = value, fill=sex)) +
  geom_density(alpha=0.5)  +
  facet_wrap( ~ variable, ncol = 4,  scales = "free")

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/density_by_sex_additional_pheno.png", width = 9, height = 6, dpi = 150, units = "in", device='png')


data_pheno$education_years <- data_pheno$education_years%>% as.numeric
data_pheno$alcohol_units <- data_pheno$alcohol_units%>% as.numeric
data_pheno$avg_sys <- data_pheno$avg_sys%>% as.numeric
data_pheno$avg_dia <- data_pheno$avg_dia%>% as.numeric
data_pheno$avg_hr <- data_pheno$avg_hr%>% as.numeric

# Remove outliers, adjust for age within gender and RINT
ds_outliers <- data_pheno %>% 
  mutate(
    across(
    .cols = c("pack_years", "education_years", "alcohol_units", "avg_sys", "avg_dia", "avg_hr", "FEV", "FVC"),
    .fns = list(mean4sd_up = ~ mean(.x, na.rm=T) + 4*sqrt(var(.x, na.rm=T)),
                mean4sd_low = ~ mean(.x, na.rm=T) - 4*sqrt(var(.x, na.rm=T))
                ))) 
ds_outliers %>% select(ends_with("mean4sd_up"), ends_with("mean4sd_low")) %>% unique()

ds_clean <- ds_outliers %>%
  mutate(pack_years = case_when(pack_years <= pack_years_mean4sd_up & pack_years >= pack_years_mean4sd_low ~ pack_years,  
                                TRUE ~ as.numeric(NA)),
         education_years = case_when(education_years <= education_years_mean4sd_up & education_years >= education_years_mean4sd_low ~ education_years,    
                                     TRUE ~ as.numeric(NA)),
         alcohol_units = case_when(alcohol_units <= alcohol_units_mean4sd_up & alcohol_units >= alcohol_units_mean4sd_low ~ alcohol_units,    
                                   TRUE ~ as.numeric(NA)),
         avg_sys = case_when(avg_sys <= avg_sys_mean4sd_up & avg_sys >= avg_sys_mean4sd_low ~ avg_sys,    TRUE ~ as.numeric(NA)),
         avg_dia = case_when(avg_dia <= avg_dia_mean4sd_up & avg_dia >= avg_dia_mean4sd_low ~ avg_dia,    TRUE ~ as.numeric(NA)),
         avg_hr = case_when(avg_hr <= avg_hr_mean4sd_up & avg_hr >= avg_hr_mean4sd_low ~ avg_hr,    TRUE ~ as.numeric(NA)),
         FEV = case_when(FEV_mean4sd_up & FEV >= FEV_mean4sd_low ~ FEV,    TRUE ~ as.numeric(NA)),
         FVC = case_when(FVC <= FVC_mean4sd_up & FVC >= FVC_mean4sd_low ~ FVC,    TRUE ~ as.numeric(NA)))

# 2. Adjust each phenotype for age withine ach sex and rank normalise

ds_pack_years <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, pack_years) %>%
    group_by(sex) %>% 
    filter(!is.na(pack_years)) %>%
    mutate(pack_years_RankNorm = residuals(lm(pack_years ~ age, na.action = na.exclude)) %>% RankNorm)

ds_education_years <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, education_years) %>%
    group_by(sex) %>% 
    filter(!is.na(education_years)) %>%
    mutate(education_years_RankNorm = residuals(lm(education_years ~ age, na.action = na.exclude)) %>% RankNorm)

ds_alcohol_units <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, alcohol_units) %>%
    group_by(sex) %>% 
    filter(!is.na(alcohol_units)) %>%
    mutate(alcohol_units_RankNorm = residuals(lm(alcohol_units ~ age, na.action = na.exclude)) %>% RankNorm)

ds_avg_sys <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, avg_sys) %>%
    group_by(sex) %>% 
    filter(!is.na(avg_sys)) %>%
    mutate(avg_sys_RankNorm = residuals(lm(avg_sys ~ age, na.action = na.exclude)) %>% RankNorm)

ds_avg_dia <- ds_clean %>% 
	select(FID_DNAm, IID_DNAm, sex, age, avg_dia) %>%
    group_by(sex) %>% 
    filter(!is.na(avg_dia)) %>%
    mutate(avg_dia_RankNorm = residuals(lm(avg_dia ~ age, na.action = na.exclude)) %>% RankNorm)

 ds_avg_hr <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, avg_hr) %>%
    group_by(sex) %>% 
    filter(!is.na(avg_hr)) %>%
    mutate(avg_hr_RankNorm = residuals(lm(avg_hr ~ age, na.action = na.exclude)) %>% RankNorm)
 
  ds_FEV <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, FEV) %>%
    group_by(sex) %>% 
    filter(!is.na(FEV)) %>%
    mutate(FEV_RankNorm = residuals(lm(FEV ~ age, na.action = na.exclude)) %>% RankNorm)
  
  ds_FVC <- ds_clean %>% 
 	select(FID_DNAm, IID_DNAm, sex, age, FVC) %>%
    group_by(sex) %>% 
    filter(!is.na(FVC)) %>%
    mutate(FVC_RankNorm = residuals(lm(FVC ~ age, na.action = na.exclude)) %>% RankNorm)
 

 	
# Merge back together
ds_pheno <- ds_clean %>%
left_join(ds_pack_years, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "pack_years")) %>%
left_join(ds_education_years, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "education_years")) %>%
left_join(ds_alcohol_units, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "alcohol_units")) %>%
left_join(ds_avg_sys, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "avg_sys")) %>%
left_join(ds_avg_dia, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "avg_dia")) %>%
left_join(ds_avg_hr, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "avg_hr")) %>%
left_join(ds_FEV, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "FEV")) %>%
left_join(ds_FVC, by=c("FID_DNAm", "IID_DNAm", "sex", "age", "FVC"))  %>%
select(FID_DNAm, IID_DNAm, sex, Set, bmi, bmi_RankNorm, ever_smoke, heart_disease, high_BP, diabetes, depression, osteo_arthritis, rheum_arthritis, asthma,
       T2D, pack_years, pack_years_RankNorm, education_years, education_years_RankNorm, alcohol_units, alcohol_units_RankNorm, 
       avg_sys, avg_sys_RankNorm, avg_dia, avg_dia_RankNorm, avg_hr, avg_hr_RankNorm,  FEV, FEV_RankNorm, FVC, FVC_RankNorm)
       


write.table(ds_pheno, file="/Local_Scratch/Alesha/DNAm/phenotypes/additional_covariates_agesex_ranknorm_outliers4sd.txt", col.name=T, row.name=F, quote=F)


# sample sizes
pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T) %>% left_join(read.table("/Local_Scratch/Alesha/DNAm/phenotypes/additional_covariates_agesex_ranknorm_outliers4sd.txt", header=T), by=c("FID_DNAm", "IID_DNAm", "sex", "Set"))

csum <- colSums(!is.na(pheno)) %>% as.data.frame
# bmi                        7689
# whr                        7638
# body_fat                   7555
# Glucose                    7523
# HDL_cholesterol            7674
# Total_cholesterol          7692
# height                     7729
# weight                     7700
# waist                      7642
# hips                       7614
# ever_smoke                 7519
# heart_disease              7615
# high_BP                    7615
# diabetes                   7615
# depression                 7615
# osteo_arthritis            7615
# rheum_arthritis            7615
# asthma                     7615
# T2D                        7758
# pack_years                 7449
# education_years            7353
# alcohol_units              6966
# avg_sys                    7741
# avg_dia                    7749
# avg_hr                     7739
# FEV                        6547
# FVC                        6547
```

### Cell type proportions

Includes technical effects and samplesheets

```{r cell types}

require(data.table)
require(tidyverse)

wave1 <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/stradl-samples-5087.rds")  # > dim(wave1) [1] 5087   15
wave3 <- read.csv("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/wave3-final/samplesheet.final.csv") # > dim(wave3) [1] 4450   13
wave4 <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/wave4/w4-samplesheet_v3.rds") # > dim(wave4) [1] 8877   12
wave4_cellcomp <- as.data.frame(fread("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/wave4/wave4_cellcomp.tsv")) # > dim(wave4_cellcomp) [1] 8877    7

comb <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds") %>% separate(col=Sample_Sentrix_ID, into=c('Sentrix_ID', 'Sentrix_Position'), sep='_', remove=FALSE)

# Merge all cell type information together
wave1_comb <- comb %>% 
  inner_join(wave1 %>% 
              mutate(Sentrix_ID = as.character(Sentrix_ID)) %>%
              select(-age, -sex, -father, -mother), 
            by=c("Sample_Name", "Sentrix_ID", "Sentrix_Position")) %>%
  select(Sample_Name, Sample_Sentrix_ID, Sentrix_ID, Sentrix_Position,
         age, sex, Set, Batch, CD8T,  CD4T,  NK, 
         Bcell , Mono,  Gran, plate_processing_batch)

wave3_comb <- comb %>%
  inner_join(wave3 %>%
              mutate(Sentrix_ID = as.character(Sentrix_ID)) %>%
              select(-age, -sex, -Batch), 
            by=c("Sample_Name", "Sample_Sentrix_ID", "Sentrix_ID", "Sentrix_Position")) %>%
  mutate(plate_processing_batch=NA) %>%
  select(Sample_Name, Sample_Sentrix_ID, Sentrix_ID, Sentrix_Position,
         age, sex, Set, Batch, CD8T,  CD4T,  NK, 
         Bcell , Mono,  Gran, plate_processing_batch)
  
wave4_comb <- comb %>%
  inner_join(wave4_cellcomp,
             by=c("Sample_Sentrix_ID" = "ID")) %>%
  mutate(plate_processing_batch=NA) %>%
  select(Sample_Name, Sample_Sentrix_ID, Sentrix_ID, Sentrix_Position,
         age, sex, Set, Batch, CD8T,  CD4T,  NK, 
         Bcell , Mono,  Gran, plate_processing_batch)

waves_comb <- rbind(wave1_comb, wave3_comb, wave4_comb)

write.table(waves_comb, file="/Local_Scratch/Alesha/DNAm/GS_20K_samplesheet.txt", col.name=T, row.name=F, quote=F)

ds_unrelated <- read.table("/Local_Scratch/Alesha/covariates/oii_cov_unrelated.txt", header=T)

waves_comb_unrelated <- waves_comb %>% filter(Sample_Sentrix_ID %in% ds_unrelated$IID_DNAm)
write.table(waves_comb_unrelated, file="/Local_Scratch/Alesha/DNAm/GS_20K_samplesheet_unrelated.txt", col.name=T, row.name=F, quote=F)
```

<br>

### **DNAm QC**

-   GS20K 3 waves normalised together

-   We removed probes with almost invariable beta values across individuals (standard deviation \< 0.02).

-   We removed cross-hybridising/polymorphic probes based on publications by Chen *et al.* 2013 for 450k array data (<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3592906/>) or McCartney *et al.* 2016 for EPIC array data (<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4909830/>).

-   Adjust for covariates and cell types (Age, sex, wave, slide and cell type proportions)

-   Keep only unrelated individuals and exclude withdrawn individuals

-   781,379 DNAm probes

<br>

```{bash DNAm QC}

in_path=/Local_Data/methylation/GS_20k/OSCA_files
filename=mvals-norm20k-18413-831733
scratch=/Local_Scratch/Alesha

awk '{print $1, $2}' /Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt > /Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_related_FID_IID_DNAm.txt

osca_Linux --befile ${in_path}/${filename} \
--keep /Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_unrelated_FID_IID_DNAm.txt \
--exclude-probe ${scratch}/DNAm/CpGs_to_remove/exclude_epic_autosomes.txt \
--sd-min 0.02 \
--make-bod \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated

# 0 of 781379 sites are Excluded with low variance (std < 0.020000) and 781379 sites left.
# 7758 individuals have been saved in the file /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated.oii .
# 781379 probes have been saved in the file /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated.opi .
# M-values of Methylation data for 781379 probes of 7758 individuals have been saved in the file /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated.bod.

# related individuals
<!-- 18413 individuals to be included from  /Local_Data/methylation/GS_20k/OSCA_files/mvals-norm20k-18413-831733.oii . -->
<!-- 18376 individuals are kept from /Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_related_FID_IID_DNAm.txt. -->
<!-- Reading probe information from /Local_Data/methylation/GS_20k/OSCA_files/mvals-norm20k-18413-831733.opi ... -->
<!-- WARNING: at least one gene id is missing. -->
<!-- 831733 probes to be included from  /Local_Data/methylation/GS_20k/OSCA_files/mvals-norm20k-18413-831733.opi . -->
<!-- Reading a list of probes from /Local_Scratch/Alesha/DNAm/CpGs_to_remove/exclude_epic_autosomes.txt . -->
<!-- 50354 probes are excluded from /Local_Scratch/Alesha/DNAm/CpGs_to_remove/exclude_epic_autosomes.txt and there are 781379 probe remaining. -->
<!-- Reserving 2047 MB memory for I/O buffer. -->




# Adjust probes for covariates

# Attempt 1: Sentrix_Position, Set, Sex, age, CD8T, CD4T, NK, Bcell, Mono, Gran
# Remove Sentrix_ID and Batch as too many levels

# Sentrix_ID        Sentrix_Position Set          Batch      
# 200784530081:   8   R02C01 : 989   wave1:2036   W3_21  : 186  
# 201057020031:   8   R06C01 : 986   wave3:4296   W3_18  : 185  
# 201057020068:   8   R04C01 : 977   wave4:1426   W3_22  : 184  
# 201057030041:   8   R08C01 : 975                W3_23  : 183  
# 201057030094:   8   R01C01 : 969                W3_25  : 183  
# 201057150040:   8   R05C01 : 962                W3_14  : 181  
# (Other)     :7710   (Other):1900                (Other):6656  

# Related individuals covariate list - age, sex, batch, cell types, smoking status, pack years
#  Qcov - "FID", "IID", "age", "CD8T", "CD4T", "NK", "Bcell", "Mono", "Gran", "pack years"
# catcov -  "Sentrix_Position", "sex", "Batch", "smoking status"

awk -v OFS='\t' 'BEGIN { print "FID", "IID", "Sentrix_Position", "sex", "Set"} NR>1{print $2, $2,  $4, $6, $7}' /Local_Scratch/Alesha/DNAm/samplesheet/GS_20K_samplesheet_unrelated.txt >  /Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates.txt

# Add batch - remove set? Are they colinear?
awk -v OFS='\t' 'BEGIN { print "FID", "IID", "Sentrix_Position", "sex", "Batch"} NR>1{print $2, $2,  $4, $6, $8}' /Local_Scratch/Alesha/DNAm/samplesheet/GS_20K_samplesheet_unrelated.txt >  /Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch.txt

awk -v OFS='\t' 'BEGIN { print "FID", "IID", "age", "CD8T", "CD4T", "NK", "Bcell", "Mono", "Gran"} NR>1{print $2, $2, $5, $9, $10, $11, $12, $13, $14}' /Local_Scratch/Alesha/DNAm/samplesheet/GS_20K_samplesheet_unrelated.txt > /Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates.txt

# qcov and PCs
qcov <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates.txt", header=T)
path="/Local_Scratch/Alesha/DNAm/ORM/PC"
eigenvec <- read.table(paste0(path,"/mvals-norm20k-18413-831733_sd0.02_unrelated_covar-10PCS.eigenvec"))
colnames(eigenvec) <- c("FID", "IID", "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10")
qcov <- left_join(qcov, eigenvec, by=c("FID", "IID"))
write.table(qcov, file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates_PC.txt", col.name=T, row.name=F, quote=F)

# Smoking
library(tidyverse)
ever_smoke <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/ever_smoke.csv", header=T, sep=",")
pack_years <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/pack_years.csv", header=T, sep=",")
cov_batch <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch.txt", header=T)
qcov <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates.txt", header=T)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)

cov_batch_smoking <- cov_batch %>% 
  left_join(ever_smoke %>% 
        inner_join(data %>% select(Sample_Name, IID=Sample_Sentrix_ID), by="Sample_Name") %>% 
    select(-Sample_Name), by="IID") 
cov_batch_smoking %>% write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch_smoking.txt", col.name=T, row.name=F, quote=F)
    
qcov_smoking <- qcov %>% 
  left_join(pack_years %>% 
        inner_join(data %>% select(Sample_Name, IID=Sample_Sentrix_ID), by="Sample_Name") %>% 
  select(-Sample_Name), by="IID") 
           
qcov_smoking %>% write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates_smoking.txt", col.name=T, row.name=F, quote=F)

cov_batch_smoking %>% drop_na() %>% select(FID, IID) %>% 
              inner_join(qcov_smoking %>% drop_na() %>% select(IID), by="IID") %>%
              write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_keep_IID_smoking.txt", col.name=F, row.name=F, quote=F)
              



# Related individuals covariate list - age, sex, batch, cell types, smoking status, pack years
#  Qcov - "FID", "IID", "age", "CD8T", "CD4T", "NK", "Bcell", "Mono", "Gran", "pack years"
# catcov -  "Sentrix_Position", "sex", "Batch", "smoking status"

library(tidyverse)
ever_smoke <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/ever_smoke.csv", header=T, sep=",")
pack_years <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/pack_years.csv", header=T, sep=",")
samplesheet <- read.table("/Local_Scratch/Alesha/DNAm/GS_20K_samplesheet_related.txt", header=T)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd_related.txt", header=T)

cov_batch_smoking <- samplesheet  %>%
  left_join(ever_smoke %>% 
        inner_join(data %>% select(Sample_Name, IID=Sample_Sentrix_ID), by="Sample_Name") %>% 
    select(-Sample_Name), by=c("Sample_Sentrix_ID"="IID") ) %>% 
    select(FID=Sample_Sentrix_ID, IID=Sample_Sentrix_ID, Sentrix_Position, sex,  Batch, ever_smoke)
cov_batch_smoking %>% write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch_smoking_related.txt", col.name=T, row.name=F, quote=F) 

qcov_smoking <- samplesheet %>% 
  left_join(pack_years %>% 
        inner_join(data %>% select(Sample_Name, IID=Sample_Sentrix_ID), by="Sample_Name") %>% 
  select(-Sample_Name), by=c("Sample_Sentrix_ID"="IID") ) %>%
select(FID=Sample_Sentrix_ID, IID=Sample_Sentrix_ID, age, CD8T, CD4T, NK, Bcell, Mono, Gran, pack_years)

qcov_smoking %>% write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates_smoking_related.txt", col.name=T, row.name=F, quote=F)

              
# Smoking and Weight
library(tidyverse)
ever_smoke <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/ever_smoke.csv", header=T, sep=",")
pack_years <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/updated_smoking_jan_2019/pack_years.csv", header=T, sep=",")
cov_batch <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch.txt", header=T)
qcov <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates.txt", header=T)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)

qcov_smoking_bodyfat <- qcov %>% 
  left_join(pack_years %>% 
        inner_join(data %>% select(Sample_Name, IID=Sample_Sentrix_ID), by="Sample_Name") %>% 
  select(-Sample_Name), by="IID") %>%
            inner_join(ds_pheno %>% select(IID=IID_DNAm, body_fat), by="IID") 
qcov_smoking_bodyfat %>% write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates_smoking_bodyfat.txt", col.name=T, row.name=F, quote=F)

cov_batch_smoking %>% drop_na() %>% select(FID, IID) %>% 
              inner_join(qcov_smoking_bodyfat %>% drop_na() %>% select(IID), by="IID") %>%
              write.table(file="/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_keep_IID_smoking_bodyfat.txt", col.name=F, row.name=F, quote=F)





# Adjust DNAm
filename=mvals-norm20k-18413-831733
scratch=/Local_Scratch/Alesha

osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates.txt \
--adj-probe --make-bod \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated_covar

osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates_batch.txt \
--adj-probe --make-bod \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated_covar_batch
# Note cant adjust for batch -  "Maybe there is multicollinearity in the covariates."
# Is this because I cant adjust for batch and set simultaneously? - removed set

# Adjust for standard covariates and PCs
osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates_PC.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates.txt \
--adj-probe --make-bod \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated_covar_PC



# Std Cov + batch + smoking (pack years) 
# Note some individuals missing covariates so keep only those with complete information - 
osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates_smoking.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates_batch_smoking.txt \
--adj-probe --make-bod \
--thread-num 20 \
--keep ${scratch}/DNAm/OSCA/GS_20K_keep_IID_smoking.txt \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated_covar_batch_eversmoke_packyears

# Std Cov + batch + smoking (pack years) + weight 
# Note some individuals missing covariates so keep only those with complete information - --keep extracts a subset of individuals.
osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates_smoking_bodyfat.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates_batch_smoking.txt \
--adj-probe --make-bod \
--thread-num 20 \
--keep ${scratch}/DNAm/OSCA/GS_20K_keep_IID_smoking_bodyfat.txt \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_unrelated_covar_batch_eversmoke_packyears_bodyfat




# Related
# Std Cov + batch + smoking (pack years) 
# Note some individuals missing covariates so keep only those with complete information - 
osca_Linux --befile ${scratch}/DNAm/OSCA/${filename}_sd0.02_related \
--qcovar ${scratch}/DNAm/OSCA/GS_20K_qcovariates_smoking_related.txt \
--covar ${scratch}/DNAm/OSCA/GS_20K_covariates_batch_smoking_related.txt \
--adj-probe --make-bod \
--thread-num 20 \
--out ${scratch}/DNAm/OSCA/${filename}_sd0.02_related_covar_batch_eversmoke_packyears
# Non-missing quantitative covariates of 17831 individuals are included from /Local_Scratch/Alesha/DNAm/OSCA/GS_20K_qcovariates_smoking_related.txt.


```

```{r checks}
library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)

# Confirm it is the same as..
# ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt", header=T) 

# And this too
# ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds1v3.txt", header=T)

ds_pheno %>% 
  select(ends_with("RankNorm")) %>% 
  summarise(across(
    .cols = where(is.numeric), 
    .fns = list(Var=var), na.rm = TRUE, 
    .names = "{col}_{fn}"
    )) %>%
  reshape2::melt()

ds_pheno %>% 
  group_by(Set) %>% 
  select(ends_with("RankNorm")) %>% 
  summarise(across(
    .cols = where(is.numeric), 
    .fns = list(Var=var), na.rm = TRUE, 
    .names = "{col}_{fn}"
    ))%>% as.data.frame

# Check phenotypic variance matches expected & within wave
#                         variable     value
# 1               bmi_RankNorm_Var 0.9988650
# 2               whr_RankNorm_Var 0.9988578
# 3          body_fat_RankNorm_Var 0.9988474
# 4           Glucose_RankNorm_Var 0.9988308
# 5   HDL_cholesterol_RankNorm_Var 0.9988639
# 6 Total_cholesterol_RankNorm_Var 0.9988664

#     Set bmi_RankNorm_Var whr_RankNorm_Var body_fat_RankNorm_Var
# 1 wave1        1.0344391        1.0337696             0.9913359
# 2 wave3        0.9868144        0.9839824             1.0148148
# 3 wave4        0.9728171        0.9816899             0.9518543
#   Glucose_RankNorm_Var HDL_cholesterol_RankNorm_Var
# 1            1.0358339                    1.0357456
# 2            0.9623267                    0.9845110
# 3            1.0435387                    0.9873337
#   Total_cholesterol_RankNorm_Var
# 1                      1.0250887
# 2                      0.9760417
# 3                      1.0269719

# All phenotypic variances are 1
```

<br><br>

### ORM

```{bash Make ORM}

# Make ORM adjusted for standard covariates
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar
# Make ORM adjusted for standard covariates and PCs
# also make one adjusted for PCs
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_PC
# And adjusted for batch (removing set)
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch
# And adjusted for batch (removing set), ever_smoke, packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears
# Adjusted for std cov, batch, eversmoke, packyears, bodyfat
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_bodyfat
 
# Adjusted for std cov, batch, eversmoke, packyears, RELATED!!
# ${scratch}/DNAm/OSCA/${filename}_sd0.02_related_covar_batch_eversmoke_packyears -  Extract IID and match with GRM
filename=mvals-norm20k-18413-831733_sd0.02_related_covar_batch_eversmoke_packyears

ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd_related.txt", header=T)
GRM <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS.grm.id", header=F)
ORM <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_related.oii", header=F)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")

colnames(GRM) <- c("FID", "IID")
colnames(ORM) <- c("FID_DNAm", "IID_DNAm")

keep <- ds_pheno %>% select("FID_DNAm", "IID_DNAm") %>% 
  filter(IID_DNAm %in% ORM$IID_DNAm) %>%
  inner_join(data %>% select(Sample_Name, Sample_Sentrix_ID), by=c("IID_DNAm"="Sample_Sentrix_ID")) %>%
  inner_join(GRM, by=c("Sample_Name"="IID")) %>%
  select(FID_DNAm, IID_DNAm, FID, IID=Sample_Name)
write.table(keep, file="/Local_Scratch/Alesha/DNAm/OSCA/related_with_DNAm_DNA.txt", col.names=T, row.names=F, quote=F)


scratch=/Local_Scratch/Alesha
<!-- awk '{print $1, $2}' /Local_Scratch/Alesha/DNAm/OSCA/related_with_DNAm_DNA.txt > tmp -->

osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
 --make-orm --out ${scratch}/DNAm/ORM/${filename}

# Make by chr then merge together
for chr in {1..22}
do

osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
 --chr ${chr} \
 --make-orm --out ${scratch}/DNAm/ORM/${filename}_chr${chr}

done

# Make flist
echo ${scratch}/DNAm/ORM/${filename}_chr1 > ${scratch}/DNAm/ORM/myorm.flist 
for chr in {2..22}
do
echo ${scratch}/DNAm/ORM/${filename}_chr${chr} >> ${scratch}/DNAm/ORM/myorm.flist
done

# Merge orms
osca_Linux --multi-orm ${scratch}/DNAm/ORM/myorm.flist \
--make-orm \
--out ${scratch}/DNAm/ORM/${filename}



# Create sparse GRM for related individuals
scratch=/Local_Scratch/Alesha
awk '{print $3, $4}' /Local_Scratch/Alesha/DNAm/OSCA/related_with_DNAm_DNA.txt > tmp


gcta64 --grm /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS \
--make-bK-sparse 0.05 \
--keep tmp \
--out /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related_sp 
 

# Fix ID's for both sparse and normal GRM
# Rename GRM ID's to match DNAm ID's
# Save GRM ready for analysis
library(tidyverse)
ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)
id <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related_sp.grm.id", header=F)

id %>%
left_join(ds, by=c("V2"="IID")) %>%
select(FID_DNAm, IID_DNAm) %>%
write.table("/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related.grm.id", col.names=F, row.names=F, quote=F)

cp /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS.grm.N.bin /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related.grm.N.bin
cp /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS.grm.bin /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related.grm.bin




# ORM for mQTLs from GoDMC and those not mQTLs
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears
scratch=/Local_Scratch/Alesha

osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
 --extract-probe /Local_Scratch/Alesha/DNAm/GoDMC/cpg_1e-8.txt \
 --make-orm --out ${scratch}/DNAm/ORM/${filename}_mQTLs
# 173099
 
 osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
 --exclude-probe /Local_Scratch/Alesha/DNAm/GoDMC/cpg_1e-8.txt \
 --make-orm --out ${scratch}/DNAm/ORM/${filename}_nomQTLs --thread-num 20
# 608280



## Imputed GRM
gcta64 \
--bfile /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra 
--make-grm 
--threads 30 
--keep /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/unrelated_ID.txt 
--out /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated 
# 24161581 SNPs to be included from BIM file. --- that seems like too many SNPs?

# Subset to HM3 SNPs
scp -P 8017 /Users/uqahatt2/Downloads/gera_hapmap3_m01_imprsq0.3_hwe1e6.bim v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/

cd /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/

library(tidyverse)
hm3 <- read.table("/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/gera_hapmap3_m01_imprsq0.3_hwe1e6.bim", header=F)
GS <- read.table("/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra.bim", header=F)

hm3_filter <- hm3 %>% left_join(GS, by=c("V1", "V4"))
hm3_filter %>% na.omit() %>% select(V2.y) %>% write.table("/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/hm3_snps.txt", col.names=F, row.names=F, quote=F)

gcta64 \
--bfile /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra \
--make-grm \
--threads 30 \
--extract hm3_snps.txt \
--keep /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/unrelated_ID.txt \
--out /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3

# Check GRM from hm3 still all under 0.05

gcta64 \
--threads 30 \
--grm /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3 \
--grm-cutoff 0.05 \
--make-grm \
--out /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3_0.05



# Fix IDs to match DNAm IDs
cd /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS
cp GS20K_HRC.r1-1_nomono_I4_cpra_unrelated.orm.id GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.grm.id
cp GS20K_HRC.r1-1_nomono_I4_cpra_unrelated.orm.id GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.orm.id
cp GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.grm.bin GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.orm.bin
cp GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.grm.N.bin GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.orm.N.bin
```

```{bash Rename ORM to GRM}

filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_mQTLs
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_nomQTLs

cp ${scratch}/DNAm/ORM/${filename}.orm.bin ${scratch}/DNAm/ORM/GRM/${filename}.grm.bin
cp ${scratch}/DNAm/ORM/${filename}.orm.id ${scratch}/DNAm/ORM/GRM/${filename}.grm.id
cp ${scratch}/DNAm/ORM/${filename}.orm.N.bin ${scratch}/DNAm/ORM/GRM/${filename}.grm.N.bin


```

```{r Smoking}
### Smoking

library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")

ever_smoke <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/ever_smoke.csv", header=T, sep=",")
pack_years <- read.table("/Cluster_Filespace/Marioni_Group/GS/GS_dataset/pack_years.csv", header=T, sep=",")

data_smoke <- data %>% left_join(ever_smoke, by=c("Sample_Name")) %>%
  left_join(pack_years, by=c("Sample_Name")) %>%
  inner_join(ds_pheno, by=c("Sample_Sentrix_ID"="FID_DNAm"))

# 239 NA's when using both update_2019 and file in initial directory

# Regree variables against smoking

lm(bmi_RankNorm ~  ever_smoke + pack_years , data= data_smoke) %>% summary
                                                                     
     Set               Batch             ever_smoke      pack_years     
 Length:7758        Length:7758        Min.   :1.000   Min.   :  0.000  
 Class :character   Class :character   1st Qu.:3.000   1st Qu.:  0.000  
 Mode  :character   Mode  :character   Median :4.000   Median :  0.000  
                                       Mean   :3.189   Mean   :  8.076  
                                       3rd Qu.:4.000   3rd Qu.: 10.000  
                                       Max.   :5.000   Max.   :133.000  
                                       NA's   :239     NA's   :239     
```

<br> <br>

# Proportion of Variance Captured

------------------------------------------------------------------------

```{r Format phenotypes, eval=FALSE}

library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)

# Testing Sets
ds_pheno1 <- ds_pheno %>% filter(Set=="wave1") %>% 
select(FID_DNAm, IID_DNAm, bmi_RankNorm1=bmi_RankNorm, whr_RankNorm1=whr_RankNorm, body_fat_RankNorm1=body_fat_RankNorm, Glucose_RankNorm1=Glucose_RankNorm,
HDL_cholesterol_RankNorm1=HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm1=Total_cholesterol_RankNorm, height_RankNorm1=height_RankNorm, weight_RankNorm1=weight_RankNorm, waist_RankNorm1=waist_RankNorm,  hips_RankNorm1=hips_RankNorm)

ds_pheno3 <- ds_pheno %>% filter(Set=="wave3") %>% 
select(FID_DNAm, IID_DNAm, bmi_RankNorm3=bmi_RankNorm, whr_RankNorm3=whr_RankNorm, body_fat_RankNorm3=body_fat_RankNorm, Glucose_RankNorm3=Glucose_RankNorm,
HDL_cholesterol_RankNorm3=HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm3=Total_cholesterol_RankNorm, height_RankNorm3=height_RankNorm, weight_RankNorm3=weight_RankNorm, waist_RankNorm3=waist_RankNorm,  hips_RankNorm3=hips_RankNorm)

ds_pheno4 <- ds_pheno %>% filter(Set=="wave4") %>% 
select(FID_DNAm, IID_DNAm, bmi_RankNorm4=bmi_RankNorm, whr_RankNorm4=whr_RankNorm, body_fat_RankNorm4=body_fat_RankNorm, Glucose_RankNorm4=Glucose_RankNorm,
HDL_cholesterol_RankNorm4=HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm4=Total_cholesterol_RankNorm, height_RankNorm4=height_RankNorm, weight_RankNorm4=weight_RankNorm, waist_RankNorm4=waist_RankNorm,  hips_RankNorm4=hips_RankNorm)

ds1v3 <- ds_pheno1 %>% full_join(ds_pheno3, by=c("FID_DNAm", "IID_DNAm")) %>%
select(FID_DNAm, IID_DNAm, starts_with("bmi"), starts_with("whr"), starts_with("body"), 
starts_with("Glucose"), starts_with("HDL"), starts_with("Total"), starts_with("height"),  starts_with("weight"),  starts_with("waist"),  starts_with("hips"))
write.table(ds1v3, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds1v3.txt", col.name=T, row.name=F, quote=F)

ds1v4 <- ds_pheno1 %>% full_join(ds_pheno4, by=c("FID_DNAm", "IID_DNAm")) %>%
select(FID_DNAm, IID_DNAm, starts_with("bmi"), starts_with("whr"), starts_with("body"), 
starts_with("Glucose"), starts_with("HDL"), starts_with("Total"), starts_with("height"),  starts_with("weight"),  starts_with("waist"),  starts_with("hips"))
write.table(ds1v4, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds1v4.txt", col.name=T, row.name=F, quote=F)

ds3v4 <- ds_pheno3 %>% full_join(ds_pheno4, by=c("FID_DNAm", "IID_DNAm")) %>%
select(FID_DNAm, IID_DNAm, starts_with("bmi"), starts_with("whr"), starts_with("body"), 
starts_with("Glucose"), starts_with("HDL"), starts_with("Total"), starts_with("height"),  starts_with("weight"),  starts_with("waist"),  starts_with("hips"))
write.table(ds3v4, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds3v4.txt", col.name=T, row.name=F, quote=F)


write.table(ds_pheno1, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_wave1.txt", col.name=T, row.name=F, quote=F)
write.table(ds_pheno3, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_wave3.txt", col.name=T, row.name=F, quote=F)
write.table(ds_pheno4, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_wave4.txt", col.name=T, row.name=F, quote=F)


# Testing Gender
# By it is adjusted for gender? Allan said shouldn't matter at this stage..
# Male
ds_phenoM <- ds_pheno %>% filter(sex=="M") %>% 
select(FID_DNAm, IID_DNAm, bmi_RankNormM=bmi_RankNorm, whr_RankNormM=whr_RankNorm, body_fat_RankNormM=body_fat_RankNorm, Glucose_RankNormM=Glucose_RankNorm,
HDL_cholesterol_RankNormM=HDL_cholesterol_RankNorm, Total_cholesterol_RankNormM=Total_cholesterol_RankNorm, height_RankNormM=height_RankNorm, weight_RankNormM=weight_RankNorm, waist_RankNormM=waist_RankNorm,  hips_RankNormM=hips_RankNorm)

ds_phenoF <- ds_pheno %>% filter(sex=="F") %>% 
select(FID_DNAm, IID_DNAm, bmi_RankNormF=bmi_RankNorm, whr_RankNormF=whr_RankNorm, body_fat_RankNormF=body_fat_RankNorm, Glucose_RankNormF=Glucose_RankNorm,
HDL_cholesterol_RankNormF=HDL_cholesterol_RankNorm, Total_cholesterol_RankNormF=Total_cholesterol_RankNorm, height_RankNormF=height_RankNorm, weight_RankNormF=weight_RankNorm, waist_RankNormF=waist_RankNorm,  hips_RankNormF=hips_RankNorm)

dsMvF <- ds_phenoM %>% full_join(ds_phenoF, by=c("FID_DNAm", "IID_DNAm")) %>%
select(FID_DNAm, IID_DNAm, starts_with("bmi"), starts_with("whr"), starts_with("body"), 
starts_with("Glucose"), starts_with("HDL"), starts_with("Total"), starts_with("height"), starts_with("weight"), starts_with("waist"),  starts_with("hips"))
write.table(dsMvF, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/dsMvF.txt", col.name=T, row.name=F, quote=F)

write.table(ds_phenoM, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Males.txt", col.name=T, row.name=F, quote=F)
write.table(ds_phenoF, file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Females.txt", col.name=T, row.name=F, quote=F)




# Analysis Sets
ds_pheno %>% select(FID_DNAm, IID_DNAm, bmi_RankNorm, whr_RankNorm, body_fat_RankNorm ,  Glucose_RankNorm, HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm, height_RankNorm, weight_RankNorm, waist_RankNorm, hips_RankNorm) %>%
write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt", col.name=T, row.name=F, quote=F)


# Analysis Sets - related
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd_related.txt", header=T)
ds_pheno %>% select(FID_DNAm, IID_DNAm, bmi_RankNorm, whr_RankNorm, body_fat_RankNorm ,  Glucose_RankNorm, HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm, height_RankNorm, weight_RankNorm, waist_RankNorm, hips_RankNorm) %>%
write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_related.txt", col.name=T, row.name=F, quote=F)


# Descriptive tables
library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)
ORM <- read.table("/Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.orm.id", header=F)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")

ds <- ds_pheno %>% filter(IID_DNAm %in% ORM$V2) %>% left_join(data %>% select(Sample_Sentrix_ID, age), by=c("IID_DNAm"="Sample_Sentrix_ID"))

```

We estimate te proportion of variance captured by DNAm probes and compare this with the proportion of variance explained by genetic factors.

Adjusting DNAm for age, sex, cell types, batch, smoking status and pack years

```{bash epigenetic}

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi


awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.prop

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--pheno temp.${trait}.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/${trait}

done
```

```{r genetic factors}
pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
# pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_related.txt

scratch=/Local_Scratch/Alesha

mkdir ${scratch}/DNAm/GREML/prop_var_explained/GRM/

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

cov=covar_batch_eversmoke_packyears
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}_GRM

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.prop

# --grm ${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated \

# gcta64 --reml --grm /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related \
# --pheno temp.${trait}.prop \
# --threads 10 \
# --out ${scratch}/DNAm/GREML/prop_var_explained/${cov}_GRM/related_${trait}
osca_Linux --reml --orm /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3 \
--pheno temp.${trait}.prop \
--thread-num 10 \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}_GRM/${trait}_imputed_hm3

done
```

```{r joint}

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

mkdir ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/
mkdir ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}

echo ${scratch}/DNAm/ORM/GRM/${filename} > ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}.txt
echo ${scratch}/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated >> ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}.txt


echo ${scratch}/DNAm/ORM/${filename} > ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}_imputed_hm3.txt
echo /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3 >> ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}_imputed_hm3.txt



cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height  
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.joint

# gcta64 --reml \
# --mgrm ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}.txt \
# --pheno temp.${trait}.prop \
# --threads 10 \
# --out ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/${trait}_imputed

osca_Linux --reml \
--multi-orm ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/GRM_ORM_${cov}_imputed_hm3.txt \
--pheno temp.${trait}.joint \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/prop_var_explained/GRM_ORM/${cov}/${trait}_imputed_hm3

done
```

```{bash collect output}
path=${scratch}/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears
cd ${path}

awk  -F',' '{print FILENAME, "allsamples", "ORM",  $1, $2, $3, $4}' *.rsq  > ${path}/traits_imputed_hm3.hsq
cd ../${cov}_GRM
awk  -F',' '{print FILENAME, "allsamples", "GRM",  $1, $2, $3, $4}' *imputed_hm3.rsq  >> ${path}/traits_imputed_hm3.hsq

cd ../GRM_ORM/covar_batch_eversmoke_packyears
awk  -F',' '{print FILENAME, "allsamples", "Joint",  $1, $2, $3, $4}' *imputed_hm3.rsq  >> ${path}/traits_imputed_hm3.hsq

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/traits_imputed_hm3.hsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/


```

```{r output, eval=TRUE, warning=FALSE, message=FALSE}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated
library(tidyverse)
library(stringr)
library(readr)
library(xlsx)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/traits_imputed_hm3.hsq") %>% 
select(-X7) %>% 
  filter(Variance!="Variance" & !is.na(SE)) %>%
  mutate(Var = case_when(Source %in% c("V(O)","V(O1)") & ORM %in% c("ORM", "Joint") ~ "VO",
                       Source %in% c("V(O)", "V(O2)") & ORM %in% c("GRM", "Joint") ~ "Vg",
                       Source == "Vp" ~ "Vp",
                       Source == "V(e)" ~ "Ve",
                       Source %in% c("V(O)/Vp", "V(O1)/Vp") & ORM %in% c("ORM", "Joint") ~ "VO_Vp",
                       Source %in% c("V(O)/Vp", "V(O2)/Vp") & ORM %in% c("GRM", "Joint") ~ "Vg_Vp",
                       TRUE ~ Source)) %>%
select(-Source, -allsamples) 
colnames(traits)[1] <- "Trait"
colnames(traits)[2] <- "Source"
traits$Trait <- str_extract(traits$Trait, ".*(?=\\.)")
traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference",
                             bmi_imputed_hm3  = "BMI", 
                             body_fat_imputed_hm3 = "Body Fat %", 
                             whr_imputed_hm3 = "Waist to Hip Ratio",
                             Glucose_imputed_hm3 = "Glucose",
                             HDL_cholesterol_imputed_hm3 = "HDL Cholesterol",
                             Total_cholesterol_imputed_hm3 = "Total Cholesterol",
                             height_imputed_hm3 = "Height")
# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits <- traits %>% mutate(Group = paste0(Source,"_", Var))

traits_wide <- traits %>%
  select(-Source, -Var) %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Group, value = col_value) %>%
  unnest(c(ORM_Ve, ORM_VO, ORM_VO_Vp, ORM_Vp,
           GRM_Ve, GRM_Vg, GRM_Vg_Vp, GRM_Vp,
           Joint_Ve, Joint_Vg, Joint_Vg_Vp, Joint_VO, Joint_VO_Vp, Joint_Vp),  .sep = '_') %>%
  select(Trait, 
         ORM_VO_Variance, ORM_VO_SE, ORM_Vp_Variance, ORM_Vp_SE, 
         ORM_VO_Vp_Variance, ORM_VO_Vp_SE, ORM_Ve_Variance, ORM_Ve_SE,
         GRM_Vg_Variance, GRM_Vg_SE, GRM_Vp_Variance, GRM_Vp_SE,
         GRM_Vg_Vp_Variance, GRM_Vg_Vp_SE, GRM_Ve_Variance, GRM_Ve_SE,
         Joint_VO_Variance, Joint_VO_SE, Joint_Vg_Variance, Joint_Vg_SE,
         Joint_Vp_Variance, Joint_Vp_SE, Joint_VO_Vp_Variance, Joint_VO_Vp_SE, 
         Joint_Vg_Vp_Variance, Joint_Vg_Vp_SE, Joint_Ve_Variance, Joint_Ve_SE)

traits_wide %>% 
  filter(!Trait %in% c("Weight", "Waist Circumference", "Hip Circumference")) %>%
  write.xlsx(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/epigenetic_correlations.xlsx",
             sheetName="Proportion_variance", append=TRUE)


traits_wide_table <- traits %>%
  mutate(Var_source = paste0(Var,"_",Source)) %>%
  filter(Var %in% c("VO_Vp", "Vg_Vp")) %>%
  select(-Source, -Var) %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var_source, value = col_value) %>%
  unnest(`Vg_Vp_GRM`, `Vg_Vp_Joint`, `VO_Vp_Joint`, `VO_Vp_ORM`, .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO/Vp` = paste0(VO_Vp_ORM_Variance, " (", VO_Vp_ORM_SE, ")"),
         `Vg/Vp` = paste0(Vg_Vp_GRM_Variance, " (", Vg_Vp_GRM_SE, ")"),
         `VO/Vp Joint` = paste0(VO_Vp_Joint_Variance, " (", VO_Vp_Joint_SE, ")"),
         `Vg/Vp Joint` = paste0(Vg_Vp_Joint_Variance, " (", Vg_Vp_Joint_SE, ")")) 

kable(traits_wide_table %>% 
  select(Trait, 
         `VO/Vp`, 
         `Vg/Vp`,
         `VO/Vp Joint`,
         `Vg/Vp Joint`), caption="Proportion of Variance captured") 



# Adjusting for batch & smoking
# Table: Proportion of Variance captured
# 
# |Trait               |VO/Vp         |Vg/Vp         |VO/Vp Joint   |Vg/Vp Joint   |
# |:-------------------|:-------------|:-------------|:-------------|:-------------|
# |BMI                 |0.595 (0.02)  |0.35 (0.046)  |0.594 (0.02)  |0.245 (0.041) |
# |Body Fat %          |0.613 (0.021) |0.363 (0.047) |0.611 (0.022) |0.234 (0.041) |
# |Waist to Hip Ratio  |0.422 (0.028) |0.231 (0.045) |0.419 (0.028) |0.123 (0.042) |
# |Glucose             |0.293 (0.038) |0.195 (0.047) |0.289 (0.039) |0.149 (0.045) |
# |HDL Cholesterol     |0.423 (0.025) |0.329 (0.047) |0.422 (0.025) |0.27 (0.043)  |
# |Total Cholesterol   |0.384 (0.036) |0.142 (0.046) |0.382 (0.036) |0.119 (0.043) |
# |Height              |0.415 (0.038) |0.647 (0.044) |0.355 (0.038) |0.543 (0.043) |
# |Weight              |0.575 (0.021) |NA (NA)       |0.572 (0.021) |0.255 (0.041) |
# |Waist Circumference |0.55 (0.021)  |0.258 (0.046) |0.549 (0.021) |0.149 (0.041) |
# |Hip Circumference   |0.457 (0.025) |0.275 (0.047) |0.452 (0.026) |0.221 (0.043) |


figure_ds <- traits %>% mutate(Var_source = paste0(Var,"_",Source)) %>%
  filter(Var %in% c("VO_Vp", "Vg_Vp")) %>%
  mutate(Var_comp  = case_when (Var_source == "VO_Vp_ORM" ~ "MRM",
                                Var_source == "Vg_Vp_GRM" ~ "GRM",
                                Var_source == "VO_Vp_Joint" ~ "MRM Joint",
                                 Var_source == "Vg_Vp_Joint" ~ "GRM Joint" ) %>% as.factor)  %>%
  mutate(Source  = case_when (Var_source == "VO_Vp_ORM" ~ "Marginal",
                                Var_source == "Vg_Vp_GRM" ~ "Marginal",
                                Var_source == "VO_Vp_Joint" ~ "Joint",
                                 Var_source == "Vg_Vp_Joint" ~ "Joint" ) %>% as.factor)
figure_ds$Source <- recode_factor( figure_ds$Source, 
                             Marginal  = "Marginal", 
                             Joint = "Joint")
figure_ds$Var_comp <- recode_factor( figure_ds$Var_comp, 
                             MRM  = "MRM", 
                             GRM = "GRM",
                             `MRM Joint`="MRM Joint",
                             `GRM Joint` ="GRM Joint")

figure_ds <- figure_ds %>% filter(Trait %in% c("BMI", "Body Fat %", "Waist to Hip Ratio", "Glucose", "HDL Cholesterol", "Total Cholesterol"))

# figure_ds %>% write.table(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Genome_Biology/Feb2023/Data/Variance_traits.csv", col.names=T, row.names=F, quote=F, sep=",")

figure_ds %>% ggplot(aes(fill=Var_comp, y=Variance, x=Source)) +
  geom_bar( stat="identity", width=0.8, position = position_dodge(width=0.9)) +
  geom_errorbar(aes(ymin=Variance-SE, ymax=Variance+SE), width=.2,
                 position=position_dodge(0.9))  +
  facet_wrap(~Trait , nrow = 2) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Variance component") +
  labs(y="Proportion of variance captured", x="Model") 
  # theme(legend.position="bottom")
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_imputed+hm3.png", width = 10, height = 7, dpi = 300, units = "in", device='png')


```


<br>

<br>

### mQTLs and non-mQTLs


```{r mQTL and non-mQTLs, eval=FALSE}
# Make ORM for mQTL and non-mQTL
osca \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/GoDMC/cpg_1e-8.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_mQTLs


pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}

printf '%s\n' "${scratch}/DNAm/ORM/GRM/${filename}_mQTLs" "${scratch}/DNAm/ORM/GRM/${filename}_nomQTLs" "${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated" > ${scratch}/DNAm/ORM/${filename}_mQTL_ORM_GRM.flist

printf '%s\n' "${scratch}/DNAm/ORM/GRM/${filename}_nomQTLs" "${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated" > ${scratch}/DNAm/ORM/${filename}_nomQTL_ORM_GRM.flist


for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
#for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi


awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.prop

gcta64 --reml \
--mgrm  ${scratch}/DNAm/ORM/${filename}_nomQTL_ORM_GRM.flist \
--pheno temp.${trait}.prop \
--threads 10 \
--reml-maxit 200 \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/${trait}_nomQTL_ORM_GRM
# --reml-no-constrain \

done


cd ${scratch}/DNAm/GREML/prop_var_explained/${cov}/
awk  -F',' '{print FILENAME, $1, $2, $3, $4}' *_mQTL_ORM_GRM.hsq  > traits_mQTL_ORM_GRM.rsq
awk  -F',' '{print FILENAME, $1, $2, $3, $4}' *_nomQTL_ORM_GRM.hsq  > traits_nomQTL_ORM_GRM.rsq


scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/traits_mQTL_ORM_GRM.rsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/
```

```{r 450K subset}
scp -P 8017 /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/probes/humanmethylation450_15017482_v1-2_edit.txt v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K


# 1. 450K with mQTL
# 2. 450K without mQTL

library(dplyr)
mQTL <- read.table("/Local_Scratch/Alesha/DNAm/GoDMC/cpg_1e-8.txt", header=F)
colnames(mQTL) <- "probeID"
DNAm450K <- read.table("/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/humanmethylation450_15017482_v1-2_edit.txt", header=T)
probelist <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.opi", header=F)
colnames(probelist) <- c("chr", "probeID", "probe_POS", "gene", "orientation")

mQTL_probes <- probelist %>% filter(probeID %in% DNAm450K$IlmnID & probeID %in% mQTL$probeID)
nomQTL_probes <- probelist %>% filter(probeID %in% DNAm450K$IlmnID & !probeID %in% mQTL$probeID)
probes_450K <- probelist %>% filter(probeID %in% DNAm450K$IlmnID )
probes_not450K <- probelist %>% filter(!probeID %in% DNAm450K$IlmnID )

write.table(mQTL_probes$probeID, file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mQTL_450K.txt", col.names=F, row.names=F, quote=F)
write.table(nomQTL_probes$probeID, file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/nomQTL_450K.txt", col.names=F, row.names=F, quote=F)
write.table(probes_450K$probeID, file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/allprobes_450K.txt", col.names=F, row.names=F, quote=F)
write.table(probes_not450K$probeID, file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/probes_excluding450K.txt", col.names=F, row.names=F, quote=F)


# Make ORMs


osca_Linux \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mQTL_450K.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_mQTL450K

osca_Linux \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/nomQTL_450K.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_nomQTL450K

osca_Linux \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/allprobes_450K.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_all450K

osca_Linux \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/probes_excluding450K.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/450K/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_probes_excluding450K




# OREML


pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cp /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.grm.bin /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.orm.bin
cp /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.grm.id /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.orm.id
cp /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.grm.N.bin /Local_Scratch/Alesha/DNAm/GREML/traits/GRM/GS_GWAS_unrelated.orm.N.bin

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

# 3 component model (mQTL 450K, nomQTL 450K, GRM)
printf '%s\n' "${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_mQTL450K" "${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_nomQTL450K" "${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated" > ${scratch}/DNAm/ORM/ORM_subsets/450K/450K_mQTL_ORM_GRM.flist

# 2 component model (nomQTL 450K, GRM)
printf '%s\n' "${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_nomQTL450K" "${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated" > ${scratch}/DNAm/ORM/ORM_subsets/450K/450K_nomQTL_ORM_GRM.flist

# 2 component model (450K vs else)
printf '%s\n' "${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_all450K" "${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_probes_excluding450K"  > ${scratch}/DNAm/ORM/ORM_subsets/450K/ORM_450K_vs_else.flist


for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}5.prop

# 3 comp model 450K
osca_Linux --reml \
--multi-orm ${scratch}/DNAm/ORM/ORM_subsets/450K/450K_mQTL_ORM_GRM.flist \
--pheno temp.${trait}.prop \
--reml-maxit 200 \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K/${trait}_450KmQTL_ORM_GRM

# 2 comp model 450K
osca_Linux --reml \
--multi-orm ${scratch}/DNAm/ORM/ORM_subsets/450K/450K_nomQTL_ORM_GRM.flist \
--pheno temp.${trait}.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K/${trait}_450KnomQTL_ORM_GRM

# 1 comp 450K
osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/ORM_subsets/450K/${filename}_all450K \
--pheno temp.${trait}.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K/${trait}_all450K

# 2 comp 450K vs else
osca_Linux --reml \
--multi-orm ${scratch}/DNAm/ORM/ORM_subsets/450K/ORM_450K_vs_else.flist \
--pheno temp.${trait}5.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K/${trait}_450K_vs_else

done


# collect output

cd ${scratch}/DNAm/GREML/prop_var_explained/${cov}/mQTL_450K

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

awk  -F',' -v "traits=${trait}" '{print traits, "450KmQTL_ORM_GRM", $1, $2, $3, $4}'   ${trait}_450KmQTL_ORM_GRM.rsq  >> traits_mQTL_450K_split.rsq
awk  -F',' -v "traits=${trait}" '{print traits, "450KnomQTL_ORM_GRM", $1, $2, $3, $4}' ${trait}_450KnomQTL_ORM_GRM.rsq >> traits_mQTL_450K_split.rsq
awk  -F','  -v "traits=${trait}" '{print traits, "450Kall", $1, $2, $3, $4}' ${trait}_all450K.rsq  >> traits_mQTL_450K_split.rsq
awk  -F','  -v "traits=${trait}" '{print traits, "450Kvselse", $1, $2, $3, $4}' ${trait}_450K_vs_else.rsq  >> traits_mQTL_450K_split.rsq

done

# bmi_450KmQTL_ORM_GRM.rsq' for reading (No such file or directory)
# bmi_450K_vs_else.rsq' for reading (No such file or directory)
# body_fat_450KmQTL_ORM_GRM.rsq' for reading (No such file or directory)
# body_fat_450K_vs_else.rsq' for reading (No such file or directory)
# HDL_cholesterol_450KmQTL_ORM_GRM.rsq' for reading (No such file or directory)
# Total_cholesterol_450KmQTL_ORM_GRM.rsq' for reading (No such file or directory)

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/mQTL_450K/traits_mQTL_450K_split.rsq  /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/


  
```


```{r ORM subsets}

# Lets make two random ORMs 
set.seed(999)
library(dplyr)
opi <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.opi", header=F)
dim(opi)
dnam1 <- sample_n(opi, 781379/2)
dnam2 <- opi %>% filter(!V2 %in% dnam1$V2)
write.table(dnam1 %>% select(V2), file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/ORM1_equalN.txt", col.names=F, row.names=F, quote=F)
write.table(dnam2 %>% select(V2), file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/ORM2_equalN.txt", col.names=F, row.names=F, quote=F)

dnam3 <- sample_n(opi, 781379*0.75)
dnam4 <- opi %>% filter(!V2 %in% dnam3$V2)
write.table(dnam3 %>% select(V2), file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/ORM1_0.75.txt", col.names=F, row.names=F, quote=F)
write.table(dnam4 %>% select(V2), file="/Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/ORM2_0.25.txt", col.names=F, row.names=F, quote=F)


subset=ORM2_0.25

osca_Linux \
--befile /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears \
--extract-probe /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/${subset}.txt \
--make-orm \
--out /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets/${subset}

# Make grm suffix
cd /Local_Scratch/Alesha/DNAm/ORM/ORM_subsets

subset=ORM1_0.75
cp ${subset}.orm.bin ${subset}.grm.bin
cp ${subset}.orm.id ${subset}.grm.id
cp ${subset}.orm.N.bin ${subset}.grm.N.bin



pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

printf '%s\n' "${scratch}/DNAm/ORM/ORM_subsets/ORM1_equalN" "${scratch}/DNAm/ORM/ORM_subsets/ORM2_equalN"  > ${scratch}/DNAm/ORM/ORM_subsets/ORM_equalN.flist
printf '%s\n' "${scratch}/DNAm/ORM/ORM_subsets/ORM1_0.75" "${scratch}/DNAm/ORM/ORM_subsets/ORM2_0.25"  > ${scratch}/DNAm/ORM/ORM_subsets/ORM_unequalN.flist


subset=ORM_unequalN

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi


awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.prop

gcta64 --reml \
--mgrm  ${scratch}/DNAm/ORM/ORM_subsets/${subset}.flist \
--pheno temp.${trait}.prop \
--threads 10 \
--reml-maxit 100 \
--out ${scratch}/DNAm/ORM/ORM_subsets/${subset}_${trait}

done



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/....... /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/


```


```{r mQTL vs GRM, eval=TRUE, warning=FALSE, message=FALSE}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

# 3 var comp #

library(stringr)
library(readr)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/traits_mQTL_ORM_GRM.rsq") %>% 
select(-X5) %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of" & Source!="Sum") %>%
mutate(Var = case_when(Source %in% c("V(O)","V(G1)") ~ "mQTL",
                       Source %in% c("V(G)", "V(G2)") ~ "nomQTL",
                       Source %in% c("V(G)", "V(G3)") ~ "Vg",
                       Source == "Vp" ~ "Vp",
                       Source == "V(e)" ~ "Ve",
                       Source %in% c("V(O)/Vp", "V(G1)/Vp") ~ "mQTL/Vp",
                       Source %in% c("V(O)/Vp", "V(G2)/Vp") ~ "nomQTL/Vp",
                       Source %in% c("V(G)/Vp", "V(G3)/Vp") ~ "Vg/Vp")) %>%
select(-Source) 
colnames(traits)[1] <- "Trait"

traits$Trait <- recode_factor( traits$Trait, 
                             bmi_mQTL_ORM_GRM.hsq  = "BMI", 
                             body_fat_mQTL_ORM_GRM.hsq = "Body Fat %", 
                             whr_mQTL_ORM_GRM.hsq = "Waist to Hip Ratio",
                             Glucose_mQTL_ORM_GRM.hsq = "Glucose",
                             HDL_cholesterol_mQTL_ORM_GRM.hsq = "HDL Cholesterol",
                             Total_cholesterol_mQTL_ORM_GRM.hsq = "Total Cholesterol",
                             height_mQTL_ORM_GRM.hsq = "Height",
                             weight_mQTL_ORM_GRM.hsq = "Weight",
                             waist_mQTL_ORM_GRM.hsq = "Waist Circumference",
                             hips_mQTL_ORM_GRM.hsq = "Hip Circumference")
traits$Source= "Var3Comp"


# 2 var comp #


traits2 <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/traits_nomQTL_ORM_GRM.rsq") %>% 
select(-X5) %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of" & Source!="Sum") %>%
mutate(Var = case_when(Source %in% c("V(O)","V(G1)") ~ "nomQTL",
                       Source %in% c("V(G)", "V(G2)") ~ "Vg",
                       Source == "Vp" ~ "Vp",
                       Source == "V(e)" ~ "Ve",
                       Source %in% c("V(O)/Vp", "V(G1)/Vp") ~ "nomQTL/Vp",
                       Source %in% c("V(G)/Vp", "V(G2)/Vp") ~ "Vg/Vp")) %>%
select(-Source) 
colnames(traits2)[1] <- "Trait"

traits2$Trait <- recode_factor(traits2$Trait, 
                             bmi_nomQTL_ORM_GRM.hsq   = "BMI", 
                             body_fat_nomQTL_ORM_GRM.hsq  = "Body Fat %", 
                             whr_nomQTL_ORM_GRM.hsq = "Waist to Hip Ratio",
                             Glucose_nomQTL_ORM_GRM.hsq  = "Glucose",
                             HDL_cholesterol_nomQTL_ORM_GRM.hsq  = "HDL Cholesterol",
                             Total_cholesterol_nomQTL_ORM_GRM.hsq  = "Total Cholesterol",
                             height_nomQTL_ORM_GRM.hsq  = "Height")
traits2$Source= "Var2Comp"




# Merge
traits_all <- rbind(traits, traits2)





# Recode to numeric
traits_all$Variance <- traits_all$Variance %>% as.numeric
traits_all$SE <- traits_all$SE %>% as.numeric


traits_wide <- traits_all %>%
  mutate(Source_Var = paste0(Source,"_", Var)) %>%
  select(-Source, -Var) %>%
  filter(! Trait %in% c("Weight", "Waist Circumference", "Hip Circumference")) %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Source_Var, value = col_value) %>%
  unnest(c(Var3Comp_mQTL, Var3Comp_nomQTL, Var3Comp_Vg, Var3Comp_Vp, Var3Comp_Ve, `Var3Comp_mQTL/Vp`, `Var3Comp_nomQTL/Vp`, `Var3Comp_Vg/Vp`, 
           Var2Comp_nomQTL, Var2Comp_Vg, Var2Comp_Vp, Var2Comp_Ve, `Var2Comp_nomQTL/Vp`, `Var2Comp_Vg/Vp` ),  .sep = '_') 


traits_wide %>% select(Trait, `Var3Comp_mQTL/Vp_Variance`, `Var3Comp_nomQTL/Vp_Variance`, `Var3Comp_Vg/Vp_Variance`, 
                       `Var2Comp_nomQTL/Vp_Variance`, `Var2Comp_Vg/Vp_Variance` )

traits_wide %>%
  write.xlsx(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/epigenetic_correlations.xlsx",
             sheetName="Proportion_variance_mQTL_nomQTL", append=TRUE)

traits_wide_table <- traits %>%
  filter(Var %in% c("mQTL/Vp", "nomQTL/Vp",  "Vg/Vp")) %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`mQTL/Vp`, `nomQTL/Vp`, `Vg/Vp`), .sep = '_') %>%
  mutate(across(c(`mQTL/Vp_SE`, `nomQTL/Vp_Variance`, `nomQTL/Vp_SE`, `Vg/Vp_Variance`, `Vg/Vp_SE`), as.numeric))

traits_wide_table[6,5] <- 0

traits_wide_table <- traits_wide_table %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`mQTL/Vp` = paste0(`mQTL/Vp_Variance`, " (", `mQTL/Vp_SE`, ")"),
         `nomQTL/Vp` = paste0(`nomQTL/Vp_Variance`, " (", `nomQTL/Vp_SE`, ")"),
         `Vg/Vp` = paste0(`Vg/Vp_Variance`, " (", `Vg/Vp_SE`, ")")) 


kable(traits_wide_table %>% 
  select(Trait, 
         `mQTL/Vp`, 
         `nomQTL/Vp`,
         `Vg/Vp`), caption="Proportion of Variance captured") 

# Excel table
traits_wide_table %>% 
  select(Trait, Source, `mQTL/Vp_Variance`, `mQTL/Vp_SE`, 
        `nomQTL/Vp_Variance`, `nomQTL/Vp_SE`,
        `Vg/Vp_Variance`, `Vg/Vp_SE`)



# Adding proportion of variance from above
figure_mQTL <- traits_all %>% filter(Var %in% c("mQTL/Vp", "nomQTL/Vp", "Vg/Vp")) %>% 
  select(Trait, Source, Variance, SE, Var)  %>%
  mutate(Variance = case_when (Variance < 0 ~ 0,
                               TRUE ~ Variance))

figure_mQTL$Model <-  recode_factor(figure_mQTL$Source, 
                             Var3Comp   = "Model 1", 
                             Var2Comp  = "Model 2")
figure_mQTL$Var <-  recode_factor(figure_mQTL$Var, 
                             `mQTL/Vp`   = "ORM with mQTL", 
                             `nomQTL/Vp` = "ORM without mQTL",
                             `Vg/Vp` = "GRM")

figure_mQTL %>% 
  filter(!Trait %in% c("Hip Circumference", "Weight", "Waist Circumference", "Height")) %>%
  ggplot(aes(fill=Var, y=Variance, x=Model)) +
  geom_bar(position="dodge", stat="identity") +
  geom_errorbar(aes(ymin=Variance-SE, ymax=Variance+SE), width=.2,
                 position=position_dodge(.9))  +
  facet_wrap(~Trait, nrow=2) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Variance component") +
  labs(y="Proportion of variance captured", x="Model") 
  # theme(legend.position="bottom")


ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_mQTL.png", width = 9, height = 7, dpi = 300, units = "in", device='png')



```

```{r ORM subsets, eval=TRUE, warning=FALSE, message=FALSE}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

library(stringr)
library(readr)

all_models <- read.delim("~/Documents/2_project/4_GS/1_Correlations/data/results/Tables/all_models.txt")
all_models <- all_models %>% mutate(Trait = as.factor(Trait), Model=as.factor(Model), Component = as.factor(Component),
                                        Variance = Variance %>% signif(2), SE = SE %>% signif(2))


all_models$Model %>% levels
all_models$Component %>% levels


all_models$Model <-  fct_relevel(all_models$Model, c("ORM marginal",  "GRM marginal", "ORM GRM joint" ,
                                                 "450K", "450K vs else", 
                                                 "mQTL nomQTL GRM",  "nomQTL GRM" ,
                                                 "450K mQTL nomQTL GRM", "450K nomQTL GRM"))

all_models$Component <-  fct_relevel(all_models$Component, c("GRM", "ORM", "mQTL", "nomQTL", "450K", "non-450K" ))
                            


all_models %>%
  # filter(Trait %in% c("BMI", "Waist to hip ratio", "Height")) %>%
  ggplot(aes(fill=Component, y=Variance, x=Model)) +
  # geom_bar(position="dodge", stat="identity") +
  geom_bar(position="stack", stat="identity") +
  facet_wrap(~Trait, nrow=1) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Variance component") +
  labs(y="Proportion of variance captured", x="Model") +
  # theme(legend.position="bottom")
theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))



ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_all_models.png", width = 10, height = 7, dpi = 300, units = "in", device='png')

```
<br> <br>

### **Across waves**

```{bash Waves - Proportion of variance captured}

scratch=/Local_Scratch/Alesha
# filename=mvals-norm20k-18413-831733_sd0.02_unrelated_covar
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

mkdir ${scratch}/DNAm/GREML/prop_var_explained/waves/${cov}
wave=wave4

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_${wave}.txt > temp.${trait}_${wave}.prop

<!-- osca_Linux --reml \ -->
<!-- --orm ${scratch}/DNAm/ORM/${filename} \ -->
<!-- --pheno temp.${trait}_${wave}.prop \ -->
<!-- --keep /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/sample1500_${wave}.txt \ -->
<!-- --out ${scratch}/DNAm/GREML/prop_var_explained/waves/${cov}/${trait}_${wave}_1500 -->

gcta64 --reml --grm /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_related \
--pheno temp.${trait}_${wave}.prop \
--threads 10 \
--out ${scratch}/DNAm/GREML/prop_var_explained/waves/${cov}/${trait}_${wave}_GRM


done



cd ${scratch}/GREML/prop_var_explained/waves/${cov}
for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
for wave in wave1 wave3
do
awk -v "trait=${trait}" -v wave=${wave} '{print trait, wave, $1, $2, $3}' ${trait}_${wave}_1500.rsq >> trait_waves_1500.rsq
done
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/waves/covar_batch_eversmoke_packyears/trait_waves_1500.rsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/

```



```{r, eval=TRUE, warning=FALSE, message=FALSE}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

library(stringr)
library(readr)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/trait_waves.rsq") %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of" & !(Source %in% c("logL", "logL0", "LRT", "df", "Pval"))) %>%
mutate(Var = case_when(Source %in% c("V(O)","V(G1)") ~ "VO",
                       Source == "Vp" ~ "Vp",
                       Source == "V(e)" ~ "Ve",
                       Source %in% c("V(O)/Vp", "V(G1)/Vp") ~ "VO_Vp")) %>%
  select(-Source) 
colnames(traits)[1] <- "Trait"
colnames(traits)[2] <- "Wave"

traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

traits$Set <- recode_factor( traits$Wave, 
                             wave1  = "Set 1", 
                             wave3 = "Set 2", 
                             wave4 = "Set 3")
# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(VO, Vp, VO_Vp, Ve),  .sep = '_')  %>%
  select(Trait, Wave,
         VO_Variance, VO_SE, 
         Vp_Variance, Vp_SE, 
         Ve_Variance, Ve_SE,
         `VO_Vp_Variance`, `VO_Vp_SE`)
traits_wide %>% write.xlsx(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/traits_proportion_of_variance_explained_waves.xlsx")

traits_wide_table <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(`VO`, `Vp`, `VO_Vp`, `Ve`,  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO` = paste0(VO_Variance, " (", VO_SE, ")"),
         `Vp` = paste0(Vp_Variance, " (", Vp_SE, ")"),
         `VO/Vp` = paste0(VO_Vp_Variance, " (", VO_Vp_SE, ")"))

kable(traits_wide_table %>% 
  select(Trait, Wave, 
         `VO`, 
         `Vp`,
         `VO/Vp`), caption="Proportion of Variance captured by wave")

# Figure
figure_ds <- traits %>% 
  filter(Var %in% c("VO_Vp")) %>%
  mutate(Set = as.factor(Set))

figure_ds <- figure_ds %>% filter(Trait %in% c("BMI", "Body Fat %", "Waist to Hip Ratio", "Glucose", "HDL Cholesterol", "Total Cholesterol"))

figure_ds %>% ggplot(aes(fill=Set, y=Variance, x=Set)) +
  geom_bar(position="dodge", stat="identity") +
  geom_errorbar(aes(ymin=Variance-SE, ymax=Variance+SE), width=.2,
                 position=position_dodge(.9))  +
  facet_wrap(~Trait, nrow=2) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Set") +
  labs(y="Proportion variance captured", x=" ") 
  # theme(legend.position="bottom")
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_waves.png", width = 9, height = 7, dpi = 150, units = "in", device='png')







# Check numbers by batch
library(tidyverse)
cov_batch <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/GS_20K_covariates_batch.txt", header=T)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)

ds_batch <- ds_pheno %>% 
  left_join(cov_batch %>% select(IID, Batch), by=c("IID_DNAm"="IID"))  %>%
  group_by(Set, Batch) %>% tally()
ds_batch %>% write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/descriptive/batch_numbers.txt", col.name=T, row.name=F, quote=F)
  
scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/descriptive/batch_numbers.txt /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes

# Read
library(tidyverse)
ds_batch <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes/batch_numbers.txt", header=T)

ds_batch$Set <- recode_factor(ds_batch$Set, 
                             wave1  = "Set 1", 
                             wave3 = "Set 2", 
                             wave4 = "Set 3")

ds_batch %>%
  ggplot(aes(fill=Set, y=n, x=Batch)) +
  geom_bar(position="dodge", stat="identity")  +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Set") +
  labs(y="Samples per batch", x="Batch")  +
  theme(panel.grid.major = element_blank()) +
   theme(axis.text.x=element_blank(),
        axis.ticks.x=element_blank()) +
   scale_y_continuous(expand = c(0, 0))
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/Samples_per_batch.png", width = 7, height = 4, dpi = 150, units = "in", device='png')
  

ds_batch %>% group_by(Set) %>% dplyr::summarize(mean = mean(n)) 
  
```


```{r wave subsets}
# Subsets of 1500 from each wave..
library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)
ds_pheno2 <- ds_pheno %>% na.omit
table(ds_pheno2$Set)
# wave1 wave3 wave4 
#  1887  3959  1307 
set.seed(999)
sample_wave1 <- sample_n(ds_pheno2 %>% filter(Set=="wave1"), 1500)
sample_wave3 <- sample_n(ds_pheno2 %>% filter(Set=="wave3"), 1500)

write.table(sample_wave1 %>% 
              select(FID_DNAm, IID_DNAm), file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/sample1500_wave1.txt", col.names=T, row.names=F, quote=F)
write.table(sample_wave3%>% 
              select(FID_DNAm, IID_DNAm), file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/sample1500_wave3.txt", col.names=T, row.names=F, quote=F)


# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

library(stringr)
library(readr)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/OREML_propvar/covar_batch_eversmoke_packyears/trait_waves_1500.rsq") %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of" & !(Source %in% c("logL", "logL0", "LRT", "df", "Pval"))) %>%
mutate(Var = case_when(Source %in% c("V(O)","V(G1)") ~ "VO",
                       Source == "Vp" ~ "Vp",
                       Source == "V(e)" ~ "Ve",
                       Source %in% c("V(O)/Vp", "V(G1)/Vp") ~ "VO_Vp")) %>%
  select(-Source) 
colnames(traits)[1] <- "Trait"
colnames(traits)[2] <- "Wave"

traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

traits$Set <- recode_factor( traits$Wave, 
                             wave1  = "Set 1", 
                             wave3 = "Set 2", 
                             wave4 = "Set 3")
# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(VO, Vp, VO_Vp, Ve),  .sep = '_')  %>%
  select(Trait, Wave,
         VO_Variance, VO_SE, 
         Vp_Variance, Vp_SE, 
         Ve_Variance, Ve_SE,
         `VO_Vp_Variance`, `VO_Vp_SE`)

traits_wide_table <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(`VO`, `Vp`, `VO_Vp`, `Ve`,  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO` = paste0(VO_Variance, " (", VO_SE, ")"),
         `Vp` = paste0(Vp_Variance, " (", Vp_SE, ")"),
         `VO/Vp` = paste0(VO_Vp_Variance, " (", VO_Vp_SE, ")"))

kable(traits_wide_table %>% 
  select(Trait, Wave, 
         `VO`, 
         `Vp`,
         `VO/Vp`), caption="Proportion of Variance captured by wave")

traits_wide_table %>% 
  select(Trait, Wave, 
         `VO`, 
         `Vp`,
         `VO/Vp`) %>% as.data.frame


# Figure
figure_ds_1500 <- traits %>% 
  filter(Var %in% c("VO_Vp")) %>%
  mutate(Wave = as.factor(Wave),
         Sample="1500")
figure_ds$Sample="All"
figure_ds_comb = rbind(figure_ds, figure_ds_1500)


figure_ds_1500 <- figure_ds_1500 %>% filter(Trait %in% c("BMI", "Body Fat %", "Waist to Hip Ratio", "Glucose", "HDL Cholesterol", "Total Cholesterol"))

figure_ds_1500 %>% ggplot(aes(fill=Set, y=Variance, x=Set)) +
  geom_bar(position="dodge", stat="identity") +
  geom_errorbar(aes(ymin=Variance-SE, ymax=Variance+SE), width=.2,
                 position=position_dodge(.9))  +
  facet_wrap(~Trait, nrow=2) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Set") +
  labs(y="Proportion variance captured", x=" ") 
  # theme(legend.position="bottom")
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_waves_1500.png", width = 9, height = 7, dpi = 150, units = "in", device='png')
```

<br> <br>

```{r}
# Theoretic variance
ds <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)

library(dplyr)
ds %>%  group_by(Set) %>%  summarise_all(funs(var), na.rm = TRUE) %>% as.data.frame
# bmi_RankNorm  whr_RankNorm    body_fat_RankNorm Glucose_RankNorm
# 1.0306165     1.0337387       0.9913359         1.0524132
# 0.9897482     0.9834382       1.0148148         0.9526874
# 0.9708127     0.9832835       0.9518543         1.0509880

# HDL_cholesterol_RankNorm  Total_cholesterol_RankNorm
# 1.0316112                            1.0038007
# 0.9851011                            0.9821799
# 0.9916465                            1.0396815

#  height_RankNorm    weight_RankNorm    waist waist_RankNorm
#  0.9882373          1.0434951         1.0730827
#  0.9992921          0.9760648         0.9740410
#  1.0036359          1.0013445         0.9630842

# hips_RankNorm
# 1.0709526
# 0.9898853
# .9234712

```

<br> <br>

EWAS for each trait to compare the marginal effects across waves

```{r EWAS}

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}


cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
wave=wave4

mkdir ${scratch}/DNAm/GREML/EWAS

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_${wave}.txt > temp.${trait}_${wave}.EWAS

osca_Linux  \
 --befile ${scratch}/DNAm/OSCA/${filename} \
 --pheno temp.${trait}_${wave}.EWAS \
 --linear \
 --out ${scratch}/DNAm/GREML/EWAS/waves/${trait}_${wave}

done

# Filter out SNPs where p < 1e-4
scratch=/Local_Scratch/Alesha

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height
do

wave=wave1
# awk -v "wave_var=${wave}" '{ if($8  < 0.001) { print $0, wave_var}}' ${scratch}/DNAm/GREML/EWAS/waves/${trait}_${wave}.linear > ${scratch}/DNAm/GREML/EWAS/waves/${trait}_0.001.linear
awk -v "wave_var=${wave}" '{{ print $0, wave_var}}' ${scratch}/DNAm/GREML/EWAS/waves/${trait}_${wave}.linear > ${scratch}/DNAm/GREML/EWAS/waves/${trait}.linear

for wave in wave3 wave4
do

awk -v "wave_var=${wave}" '{ { print $0, wave_var}}' ${scratch}/DNAm/GREML/EWAS/waves/${trait}_${wave}.linear >> ${scratch}/DNAm/GREML/EWAS/waves/${trait}.linear

done 
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/EWAS/waves/*.linear /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS


```

```{r EWAS_r}

marginal_effects <- lapply(1:6, function(trait_n) {
	
	  trait_list=c("bmi", "whr", "body_fat", "Glucose", "HDL_cholesterol", "Total_cholesterol", "height")
	  trait=trait_list[trait_n]
	    
	  ds <- read.table(paste0("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS/",trait,".linear"), header=F)
	  
	  colnames(ds) <- c("Chr",	"Probe", "bp",	"Gene",	"Orientation",	"b",	"se",	"p",	"NMISS", "wave")
	  ds <- ds %>% filter(Chr!="Chr") %>% mutate(b = as.numeric(b), se=as.numeric(se), p=as.numeric(p))
	  
	  ds_wide <- reshape(data=ds %>%  select(Chr, Probe, b, se, p, wave), 
	                                    idvar=c("Probe", "Chr"), v.names = c("b", "se", "p"), 
	                                    timevar = "wave", 
	                                    direction="wide") %>%
	                            na.exclude %>% mutate(trait=paste0(trait))
	  ds_0.001_wide <- ds_wide %>% filter( p.wave1 < 0.001 & p.wave3 < 0.001 & p.wave4 < 0.001 )
	  
	   ds_0.001_wide_clean <- rbind(
	    ds_0.001_wide %>% 
        select(Chr, Probe, trait, x=b.wave1, y=b.wave3) %>% 
	      mutate(comparison="wave 1 vs 3"),
	    ds_0.001_wide %>% 
	      select(Chr, Probe, trait, x=b.wave1, y=b.wave4) %>% 
	      mutate(comparison="wave 1 vs 4"),  
	    ds_0.001_wide %>% 
	      select(Chr, Probe, trait, x=b.wave3, y=b.wave4) %>% 
	      mutate(comparison="wave 3 vs 4"))
	  
	   ds_0.001_wide
	})
	
ds_marginal_effects <- do.call(rbind, marginal_effects)

# Signficiant in at least one wave..
# write.table(ds_marginal_effects, file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS/EWAS_0.001_atleast1wave.txt", col.names=T, row.names=F, quote=F)

# Signficiant in all waves..
# write.table(ds_marginal_effects, file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS/EWAS_0.001_allwave.txt", col.names=T, row.names=F, quote=F)


# Compare the difference between sig in at least one wave and sig in all waves
sig_one <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS/EWAS_0.001_atleast1wave.txt", header=T) %>%
   filter(p.wave1 < 0.001 | p.wave3 < 0.001 | p.wave4 < 0.001)

sig_all <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/EWAS/EWAS_0.001_allwave.txt", header=T) %>%
   filter(p.wave1 < 0.001 & p.wave3 < 0.001 & p.wave4 < 0.001)

sig_one %>% group_by(trait) %>% tally()
sig_all %>% group_by(trait) %>% tally()

# 0.001
# > sig_one %>% group_by(trait) %>% tally()
#   trait                 n
# 1 bmi               71136
# 2 body_fat          51521
# 3 Glucose            6210
# 4 HDL_cholesterol   56315
# 5 Total_cholesterol 10518
# 6 whr               42698

# > sig_all %>% group_by(trait) %>% tally()
#   trait                 n
# 1 bmi                2660
# 2 body_fat           1231
# 3 Glucose               8
# 4 HDL_cholesterol    1202
# 5 Total_cholesterol    13
# 6 whr                1012

sig_one$trait <- recode_factor(sig_one$trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol")

sig_all$trait <- recode_factor(sig_all$trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol")


# Clean
sig_one_clean <- rbind(
	    sig_one %>% 
        select(Chr, Probe, trait, x=b.wave1, y=b.wave3) %>% 
	      mutate(comparison="Set 1 v 2"),
	    sig_one %>% 
	      select(Chr, Probe, trait, x=b.wave1, y=b.wave4) %>% 
	      mutate(comparison="Set 1 v 3"),  
	    sig_one %>% 
	      select(Chr, Probe, trait, x=b.wave3, y=b.wave4) %>% 
	      mutate(comparison="Set 2 v 3"))

sig_all_clean <- rbind(
	    sig_all %>% 
        select(Chr, Probe, trait, x=b.wave1, y=b.wave3) %>% 
	      mutate(comparison="Set 1 v 2"),
	    sig_all %>% 
	      select(Chr, Probe, trait, x=b.wave1, y=b.wave4) %>% 
	      mutate(comparison="Set 1 v 3"),  
	    sig_all %>% 
	      select(Chr, Probe, trait, x=b.wave3, y=b.wave4) %>% 
	      mutate(comparison="Set 2 v 3"))

# Find correlations
sig_all_clean_pvals <- sig_all_clean %>% group_by(trait, comparison) %>%
  summarize(
  cor=cor.test(x, y)[["estimate"]],
  pvalue=cor.test(x,y)[["p.value"]],
  statistic=cor.test(x,y)[["statistic"]],
  parameter=cor.test(x,y)[["parameter"]]) %>%
  mutate(log.p = 2* pt(statistic,  df = parameter, lower.tail=FALSE, log.p=TRUE),
         edit_pvalue = case_when(pvalue==0 ~ 1e-100, 
                                 TRUE ~ pvalue),
         link = case_when(pvalue==0 ~ "<", 
                                 TRUE ~ "=" ),) %>% as.data.frame %>%
  select(trait, comparison, cor, edit_pvalue, link)


# plot

library(viridis)
library(ggpubr)
sig_all_clean %>% 
  left_join(sig_all_clean_pvals, by=c("trait", "comparison")) %>%
    ggplot(aes(x=x, y=y)) +
    geom_point(aes(colour=trait), size=1) +
    # stat_cor(method = "pearson", size=3,   p.accuracy=1e-100) +
    facet_grid(rows=vars(trait), cols=vars(comparison)) +
    scale_color_viridis(discrete = TRUE, option = "D")+
    labs(y="", x="") + 
    theme_linedraw() +
    scale_fill_manual(values = mycolours) +
    labs(y="Probe association coefficient", x="Probe association coefficient")  +
    theme(legend.position="none") +
    geom_text(data = sig_all_clean_pvals, aes(x = -0.6, y = 1.5, 
    #                   label = paste0("R = ", round(cor,2), ",  p",link, scientific_10(signif(edit_pvalue, 2)))))

             label=paste("R=", round(cor,2), ", p", link, signif(edit_pvalue, 2))), 
             size=3.5)  

# ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/EWAS_ES_sig_all_0.001_axis.png", width = 7, height = 9, dpi = 300, units = "in", device='png')
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Journal Submissions/2_AJHG/Reviewers Comments/Supplementary Figure 3.png", width = 7, height = 9, dpi = 300, units = "in", device='png')



```

![EWAS marginal effect size by trait and wave](data/EWAS/EWAS_ES.png){width="675"}

<br> <br>

### Males and Females

```{bash Proportion of variance for Males and Females}

scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

mkdir ${scratch}/DNAm/GREML/prop_var_explained/Male
mkdir ${scratch}/DNAm/GREML/prop_var_explained/Female

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp


for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi
if [ "${trait}" == "weight" ]; then  var=10; fi
if [ "${trait}" == "waist" ]; then  var=11; fi
if [ "${trait}" == "hips" ]; then  var=12; fi
# Male
awk -v "col1=${var}" '{print $1, $2, $col1}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Males.txt > temp.${trait}_Male.prop

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--pheno temp.${trait}_Male.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/Male/${trait}_Male


# Female
awk -v "col1=${var}" '{print $1, $2, $col1}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Females.txt > temp.${trait}_Female.prop

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--pheno temp.${trait}_Female.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/Female/${trait}_Female

done


sex=Female
cd ${scratch}/DNAm/GREML/prop_var_explained/${sex}
rm  trait_${sex}.rsq
for trait in BMI whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
awk -v "trait=${trait}" -v sex=${sex} '{print trait, sex, $1, $2, $3}' ${trait}_${sex}.rsq >> trait_${sex}.rsq
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/*ale/trait_*ale.rsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex

```


```{r, eval=TRUE, warning=FALSE, message=FALSE}

library(stringr)
library(readr)
traits_F <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/trait_Female.rsq") %>% 
  mutate(Var = case_when(Source %in% c("V(O)") ~ "VO",
                       Source %in% c("Vp") ~ "Vp",
                       Source %in% c("rG") ~ "rG",
                       Source == "V(O)/Vp" ~ "VO/Vp")) %>% 
  select(-Source) %>%
  filter(!is.na(Var)) 

traits_M <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/trait_Male.rsq") %>% 
  mutate(Var = case_when(Source %in% c("V(O)") ~ "VO",
                       Source %in% c("Vp") ~ "Vp",
                       Source %in% c("rG") ~ "rG",
                       Source == "V(O)/Vp" ~ "VO/Vp")) %>% 
  select(-Source) %>%
  filter(!is.na(Var)) 
colnames(traits_F)[1] <- "Trait"
colnames(traits_F)[2] <- "Gender"
colnames(traits_M)[1] <- "Trait"
colnames(traits_M)[2] <- "Gender"

traits <- rbind(traits_F, traits_M)

traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

traits$Gender <- recode_factor( traits$Gender, 
                             Female  = "Female", 
                             Male = "Male")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(VO, Vp, `VO/Vp`, Ve),  .sep = '_')  %>%
  select(Trait, Gender,
         VO_Variance, VO_SE, 
         Vp_Variance, Vp_SE, 
         Ve_Variance, Ve_SE,
         `VO/Vp_Variance`, `VO/Vp_SE`)
traits_wide %>% write.xlsx(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/traits_proportion_of_variance_explained_waves.xlsx")


traits_wide_table <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`VO`, `Vp`, `VO/Vp`),  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO` = paste0(`VO_Variance`, " (", `VO_SE`, ")"),
         `Vp` = paste0(`Vp_Variance`, " (", `Vp_SE`, ")"),
         `VO/Vp` = paste0(`VO/Vp_Variance`, " (", `VO/Vp_SE`, ")"))



figure_ds <- traits %>% 
  filter(Trait %in% c("BMI", "Body Fat %", "Waist to Hip Ratio", "Glucose", "HDL Cholesterol", "Total Cholesterol") &
           Var =="VO/Vp")

library(ggpubr)
ggarrange(
figure_ds %>% ggplot(aes(fill=Gender, y=Variance, x=Gender)) +
  geom_bar(position="dodge", stat="identity") +
  geom_errorbar(aes(ymin=Variance-SE, ymax=Variance+SE), width=.2,
                 position=position_dodge(.9))  +
  facet_wrap(~Trait, nrow=2) +
  theme_linedraw() +
  scale_fill_manual(values = mycolours, name = "Sex") +
  labs(y="Proportion of variance captured by DNAm", x="Sex") +
  theme(legend.position = "none"),
figure_sex, ncol=2, widths=c(0.6, 0.4))



ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/proportion_variance_sex.png", width = 9, height = 7, dpi = 150, units = "in", device='png')

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/sex_plots.png", width = 10, height = 6, dpi = 150, units = "in", device='png')
```

<br> <br>

# Epigenetic Correlation

<br>

## Correlation between waves

```{bash waves}


# Bivariate REML

# bmi - columns $3, $4
# whr - columns $5, $6
# body_fat - columns $7, $8
# Glucose - columns $9, $10
# HDL_cholesterol - columns $11, $12
# Total_cholesterol - columns $13, $14

scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

mkdir ${scratch}/DNAm/phenotypes/greml.pheno/temp
cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do

for ds in 1v3 1v4 3v4
do

if [ "${trait}" == "bmi" ];              then  var1=3 var2=4; fi
if [ "${trait}" == "whr" ];              then  var1=5 var2=6; fi
if [ "${trait}" == "body_fat" ];         then  var1=7 var2=8; fi
if [ "${trait}" == "Glucose" ];          then  var1=9 var2=10; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var1=11 var2=12; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var1=13 var2=14; fi
if [ "${trait}" == "height" ]; then  var1=15 var2=16; fi


awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' ${scratch}/DNAm/phenotypes/greml.pheno/ds${ds}.txt > temp.${trait}.waves
gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--pheno temp.${trait}.waves \
--reml-maxit 200 \
--out ${scratch}/DNAm/GREML/waves/${trait}_RankNorm${ds}_0

done
done

cd ${scratch}/DNAm/GREML/waves
# for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
for trait in Total_cholesterol height 
do
for ds in 1v3 1v4 3v4
do
awk -v "trait=${trait}" -v ds=${ds} '{print trait, ds, $1, $2, $3}' ${trait}_RankNorm${ds}.hsq >> ${trait}_waves.rsq
done
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/waves/trait_waves.rsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_waves
```

```{r, eval=TRUE, warning=FALSE, message=FALSE}

library(stringr)
library(readr)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_waves/trait_waves.rsq") %>% 
  mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "VO/Vp1",
                       Source %in% c("V(G)/Vp_tr2")   ~ "VO/Vp2",
                       Source %in% c("rG") ~ "rG",
                      Source %in% c("Pval") ~ "Pval")) %>% 
  select(-Source) %>%
  filter(!is.na(Var)) 
colnames(traits)[1] <- "Trait"
colnames(traits)[2] <- "Waves"

traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric %>% round(3)
traits$SE <- traits$SE %>% as.numeric %>% round(3)

# Wide

traits_wide_table <- traits %>%
  mutate(var2 = paste0(Waves, " ", Var)) %>%
  select(-Waves, -Var) %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = c(var2), value = col_value) %>%
  unnest(c(`1v3 VO/Vp1`, `1v3 VO/Vp2`, `1v3 rG`, `1v4 VO/Vp1`, `1v4 VO/Vp2`, `1v4 rG`, 
           `3v4 VO/Vp1`, `3v4 VO/Vp2`, `3v4 rG`),  .sep = '_') 
  
traits_wide_table %>% as.data.frame
# 
#   
#   mutate(`Vwave1/Vp` = case_when(Waves=="1v3" ~ `VO/Vp1_Variance`,
#                                  Waves=="1v4" ~ `VO/Vp1_Variance`),
#          `Vwave3/Vp` = case_when(Waves=="1v3" ~ `VO/Vp2_Variance`,
#                                  Waves=="3v4" ~ `VO/Vp1_Variance`),
#          `Vwave4/Vp` = case_when(Waves=="1v4" ~ `VO/Vp2_Variance`,
#                                  Waves=="3v4" ~ `VO/Vp1_Variance`))
#          
#          

traits %>% filter(Waves=="1v3" & Var=="rG") %>% 
  mutate(`Wave1 vs Wave3` = paste0(Variance, " (", SE, ")")) %>% select(Trait, `Wave1 vs Wave3`) %>%
  left_join(traits %>% filter(Waves=="1v4" & Var=="rG") %>% 
              mutate(`Wave1 vs Wave4` = paste0(Variance, " (", SE, ")")) %>% select(Trait, `Wave1 vs Wave4`), by="Trait") %>%
  left_join(traits %>% filter(Waves=="3v4" & Var=="rG") %>% 
              mutate(`Wave3 vs Wave4` = paste0(Variance, " (", SE, ")")) %>% select(Trait, `Wave3 vs Wave4`), by="Trait") %>%
  kable(caption="Epigenetic correlation between waves")


# Figure
# figure_ds <- traits %>% filter(Var=="rG") 
figure_ds <- traits %>% filter(Var=="rG") %>% 
  left_join(traits %>% filter(Var=="Pval") %>% select(Trait, Pval=Variance, Waves), by=c("Trait", "Waves")) %>%
  mutate(pval_annot = Pval %>% signif(1))


figure_ds$Set <- recode_factor(figure_ds$Waves, 
                             `1v3` = "Set 1 v Set 2", 
                             `1v4` = "Set 1 v Set 3", 
                             `3v4` = "Set 2 v Set 3")


figure_ds %>% filter(Trait!="Height") %>%
  ggplot(aes(y=fct_reorder(Trait, desc(Trait)), x=Variance, xmin=Variance-SE, xmax = Variance+SE, colour=Trait, label=pval_annot)) +
  geom_vline(xintercept = 1, colour="darkgrey") +
  geom_errorbar(position = position_dodge(width = 0.2), width=.1) +
  geom_point(position = position_dodge(width = 0.3)) +
  geom_text(aes(label=paste0("p = ",pval_annot), x=1.04), hjust=0, vjust=-1.5, size=2.5,
        position = position_dodge(width = 1), colour="black") +
  labs(x="rG (DNAm) with BMI", y="Trait") +
  facet_wrap(~Set) +
  theme_linedraw() +
  scale_colour_manual(values = mycolours, name = " ") +
  labs(y=" ", x="DNAm correlation") +
  theme(legend.position="bottom") +
  theme(legend.position = "none")


# ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Genome_Biology/Feb2023/Figure1.png", width = 8.5, height = 4.5, dpi = 300, units = "cm", device='png')
 
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Journal Submissions/2_AJHG/tiff_figures/Figure1.tiff", width = 8.5, height = 4.5, dpi = 300, units = "in", device='png')

```

<br> <br>

## Correlation between sexes

```{bash sexes}

# Bivariate REML

scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

mkdir ${scratch}/DNAm/phenotypes/greml.pheno/temp
cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height
do

if [ "${trait}" == "bmi" ];              then  var1=3 var2=4; fi
if [ "${trait}" == "whr" ];              then  var1=5 var2=6; fi
if [ "${trait}" == "body_fat" ];         then  var1=7 var2=8; fi
if [ "${trait}" == "Glucose" ];          then  var1=9 var2=10; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var1=11 var2=12; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var1=13 var2=14; fi
if [ "${trait}" == "height" ];            then  var1=15 var2=16; fi
if [ "${trait}" == "weight" ];            then  var1=17 var2=18; fi
if [ "${trait}" == "waist" ];             then  var1=19 var2=20; fi
if [ "${trait}" == "hips" ];              then  var1=21 var2=22; fi

awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' ${scratch}/DNAm/phenotypes/greml.pheno/dsMvF.txt > temp.${trait}.sex

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0  \
--pheno temp.${trait}.sex \
--reml-maxit 100 \
--threads 20 \
--out ${scratch}/DNAm/GREML/sex/${trait}_RankNorm_MvF_0

done

cd ${scratch}/DNAm/GREML/sex
for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
awk -v "trait=${trait}"  '{print trait, $1, $2, $3}' ${trait}_RankNorm_MvF.hsq >> trait_MvF.rsq
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/sex/trait_MvF.rsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex
```

```{r, eval=TRUE, warning=FALSE, message=FALSE}

library(stringr)
library(readr)
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/trait_MvF.rsq") %>% 
  mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "VO/Vp (Males)",
                       Source %in% c("V(G)/Vp_tr2") ~ "VO/Vp (Females)",
                       Source %in% c("rG") ~ "rG",
                       Source == "Pval" ~ "Pval"),
       SE = case_when(SE=="(one-tailed" ~ as.character(NA),
                      TRUE ~ SE)) %>%
  select(-Source) %>%
  filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait"

traits$Trait <- recode_factor( traits$Trait, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide_table <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`VO/Vp (Males)`, `VO/Vp (Females)`, rG, Pval),  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 4)) %>%
  mutate(`VO/Vp (Males)` = paste0(`VO/Vp (Males)_Variance`, " (", `VO/Vp (Males)_SE`, ")"),
         `VO/Vp (Females)` = paste0(`VO/Vp (Females)_Variance`, " (", `VO/Vp (Females)_SE`, ")"),
         `rG` = paste0(rG_Variance, " (", rG_SE, ")"),
         `Pval` = Pval_Variance)

traits_wide_table %>% select(rG, Pval)
# 
# kable(traits_wide_table %>% 
#   select(Trait,  
#          `VO/Vp (Males)`,
#          `VO/Vp (Females)`,
#          `rG`,
#          `Pval`), caption="Epigenetic correlation between genders")


# Figure
figure_ds <- traits %>% filter(Var=="rG") %>% 
  left_join(traits %>% filter(Var=="Pval") %>% select(Trait, Pval=Variance), by=c("Trait")) %>%
  mutate(pval_annot = Pval %>% signif(1))
  
# options(scipen=999)
# ggarrange(
figure_sex <- figure_ds %>% filter(! Trait %in% c("Hip Circumference", "Waist Circumference", "Weight", "Height")) %>%
  mutate(sex="Male v Female") %>%
  ggplot(aes(y=fct_reorder(Trait, desc(Trait)), x=Variance, xmin=Variance-SE, xmax = Variance+SE, colour=Trait, label=pval_annot)) +
  geom_vline(xintercept = 1, colour="darkgrey") +
  geom_errorbar(position = position_dodge(width = 0.2), width=.1) +
  geom_point(position = position_dodge(width = 0.2), size=3) +
  # geom_text(aes(label=pval_annot),hjust=0.5, vjust=3, size=2,
  #       position = position_dodge(width = 1)) +
  theme_linedraw() +
  scale_colour_manual(values = mycolours, name = " ") +
  labs(y=" ", x="DNAm correlation") +
  theme(legend.position="bottom") +
  theme(legend.position = "none")+
  facet_wrap(~sex) +
  scale_y_discrete(position = "right")




ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/rG(DNAm)_sex_edit.png", width =4, height = 8, dpi = 300, units = "in", device='png')
 

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Journal Submissions/2_AJHG/tiff_figures/Figure1.tiff", width = 8.5, height = 4.5, dpi = 300, units = "in", device='png')

```

Pvalue for the likelihood test fixing the genetic correlation at 1

```{r probe:sex}
# Pheno ~ Probe:sex

# qq-plot of pvalues of interaction effects
path=/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}


mkdir ${path}/DNAm
mkdir ${path}/pvalue_output

# Make by chr then merge together
for chr in {1..22}
do

osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
--chr ${chr} \
--make-efile --out ${path}/DNAm/DNAm_unrelated_chr${chr}.txt

done


# 1. extract DNAm 
# 2. Read into R along with ds_pheno..

CHR=22

library(tidyverse)
ds <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd_related.txt", header=T)
# 
# for (CHR in 1:22) {


DNAm <- data.table::fread(paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/DNAm/DNAm_unrelated_chr",CHR,".txt"), header=T) 

ds <- ds %>% mutate(sex=as.factor(sex))

DNAm_cov <- ds %>% select(FID=FID_DNAm, IID=IID_DNAm, trait=bmi_RankNorm, sex) %>%
  inner_join(DNAm, by=c("FID", "IID"))

# 3. Probelist to loop over..
probelist <- colnames(DNAm)[-c(1,2)] # 16874 probes in chr22
length(probelist)

# 4. LM

# Set up outut file
pval.matrix <- probelist %>% as.data.frame
colnames(pval.matrix) <- c("probeID")

pval.matrix$probe_Estimate <- NA
pval.matrix$probe_StdError <- NA
pval.matrix$probe_tvalue <- NA
pval.matrix$probe_P <- NA

pval.matrix$sex_Estimate <- NA
pval.matrix$sex_StdError <- NA
pval.matrix$sex_tvalue <- NA
pval.matrix$sex_P <- NA

pval.matrix$interaction_Estimate <- NA
pval.matrix$interaction_StdError <- NA
pval.matrix$interaction_tvalue <- NA
pval.matrix$interaction_P <- NA

 
# For each probe, create normalised residuals
for (i in 1:length(probelist)) {

  print(paste0(CHR, "   ", i))
  
  # Select probevalue
  probelist[i]
  this_probe <- probelist[i]
  this_ds <- DNAm_cov %>% select(FID, IID, trait, sex, DNAm=probelist[i])
  
  lm <- lm(trait ~ DNAm + sex + DNAm:sex, data=this_ds)
  
  # Add to normalised residual matrix
  
  # Probe
  pval.matrix[pval.matrix$probeID==this_probe,]$probe_Estimate <- summary(lm)$coefficients[2,1]
  pval.matrix[pval.matrix$probeID==this_probe,]$probe_StdError <- summary(lm)$coefficients[2,2]
  pval.matrix[pval.matrix$probeID==this_probe,]$probe_tvalue <- summary(lm)$coefficients[2,3]
  pval.matrix[pval.matrix$probeID==this_probe,]$probe_P <- summary(lm)$coefficients[2,4]
  
  # Sex
  pval.matrix[pval.matrix$probeID==this_probe,]$sex_Estimate <- summary(lm)$coefficients[3,1]
  pval.matrix[pval.matrix$probeID==this_probe,]$sex_StdError <- summary(lm)$coefficients[3,2]
  pval.matrix[pval.matrix$probeID==this_probe,]$sex_tvalue <- summary(lm)$coefficients[3,3]
  pval.matrix[pval.matrix$probeID==this_probe,]$sex_P <- summary(lm)$coefficients[3,4]
  
  # Interaction
  pval.matrix[pval.matrix$probeID==this_probe,]$interaction_Estimate <- summary(lm)$coefficients[4,1]
  pval.matrix[pval.matrix$probeID==this_probe,]$interaction_StdError <- summary(lm)$coefficients[4,2]
  pval.matrix[pval.matrix$probeID==this_probe,]$interaction_tvalue <- summary(lm)$coefficients[4,3]
  pval.matrix[pval.matrix$probeID==this_probe,]$interaction_P <- summary(lm)$coefficients[4,4]
  
  # Re-set all values at the end of each cycle
  this_probe <- NULL
  this_ds <- NULL
  lm <- NULL
  probe_Estimate = sex_Estimate = interaction_Estimate <- NULL
  probe_StdError = sex_StdError = interaction_StdError <- NULL
  probe_tvalue = sex_tvalue = interaction_tvalue <- NULL
  probe_P = sex_P = interaction_P <- NULL
  
}

write.table(pval.matrix, file=paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/pvalue_output/interaction_pval_chr",CHR,".txt"), col.name=T, row.name=F, quote=F)




# }


# Combine all chr
DNAm <- data.table::fread(paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/pvalue_output/interaction_pval_chr1.txt"), header=T)
DNAm$chr <- 1
for (i in 2:22) {
  DNAm_chr <- data.table::fread(paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/pvalue_output/interaction_pval_chr",i,".txt"), header=T)
  DNAm_chr$chr <- i
  DNAm <- rbind(DNAm, DNAm_chr)
}
write.table(DNAm, file=paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/interaction_pval.txt"), col.name=T, row.name=F, quote=F)

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/interaction_pval.txt  /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex

# Reference is Male!

###########################
# Idenitify signficant probes
###########################

library(qqman)
DNAm <- data.table::fread(paste0("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/interaction_pval.txt"), header=T)
qq(DNAm$probe_P)

DNAm %>% filter(probeID=="cg26651978")

pthreshold = 0.05/nrow(DNAm)

DNAm %>% filter(interaction_P < pthreshold)  # 8..
DNAm %>% filter(interaction_P < pthreshold) %>% select(probeID, chr, probe_Estimate, probe_StdError, probe_P, sex_Estimate, sex_StdError, sex_P, interaction_Estimate, interaction_StdError, interaction_P)


options(scipen=NA)


# Extract those DNAm values
path=/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

echo cg12269535 > ${path}/DNAm/sig_probes.txt
echo cg12054453 >> ${path}/DNAm/sig_probes.txt
echo cg18181703 >> ${path}/DNAm/sig_probes.txt
echo cg11047325 >> ${path}/DNAm/sig_probes.txt
echo cg19748455 >> ${path}/DNAm/sig_probes.txt
echo cg16936953 >> ${path}/DNAm/sig_probes.txt
echo cg00840791 >> ${path}/DNAm/sig_probes.txt
echo cg09349128 >> ${path}/DNAm/sig_probes.txt
echo cg15125798 >> ${path}/DNAm/sig_probes.txt
echo cg26651978 >> ${path}/DNAm/sig_probes.txt

osca_Linux  --befile ${scratch}/DNAm/OSCA/${filename} \
--extract-probe ${path}/DNAm/sig_probes.txt \
--make-efile --out ${path}/DNAm/DNAm_unrelated_sig_probes.txt


library(tidyverse)
ds <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd_related.txt", header=T)
age <- read.table("/Local_Scratch/Alesha/DNAm/GS_20K_samplesheet_related.txt", header=T)
DNAm <- data.table::fread(paste0("/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/DNAm/DNAm_unrelated_sig_probes.txt"), header=T) 
ds <- ds %>% mutate(sex=as.factor(sex))
DNAm_cov <- ds %>% select(FID=FID_DNAm, IID=IID_DNAm, trait=bmi_RankNorm, sex) %>%
  inner_join(DNAm, by=c("FID", "IID")) %>%
  inner_join(age %>% select(age, Sample_Sentrix_ID), by=c("FID"="Sample_Sentrix_ID"))

write.table(DNAm_cov, file="/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/DNAm/DNAm_unrelated_sig_probes_sex.txt", col.names=T, row.names=F, quote=F)


scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/age_sex_interactions/DNAm/DNAm_unrelated_sig_probes_sex.txt  /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex


library(sjPlot)
library(sjmisc)
library(ggplot2)
library(tidyverse)

sig_probes <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/DNAm_unrelated_sig_probes_sex.txt", header=T) %>% 
  dplyr::rename(BMI = trait) %>% 
  na.omit() 


# Linear model in Males and Female
probelist <- colnames(sig_probes)[5:12] 
length(probelist)

# Set up outut file
pval.matrix <- probelist %>% as.data.frame
colnames(pval.matrix) <- c("probeID")
pval.matrix <- pval.matrix %>% mutate(M_Estimate=NA, M_StdError=NA, M_tvalue=NA, M_P=NA,
                                      F_Estimate=NA, F_StdError=NA, F_tvalue=NA, F_P=NA)

# For each probe, create normalised residuals
for (i in 1:length(probelist)) {

  # Select probevalue
  probelist[i]
  this_probe <- probelist[i]
  this_ds <- sig_probes %>% select(FID, IID, BMI, sex, DNAm=probelist[i])
  
  lm_M <- lm(BMI ~ DNAm, data=this_ds %>% filter(sex=="M"))
  lm_F <- lm(BMI ~ DNAm, data=this_ds %>% filter(sex=="F"))
  
  # Add to normalised residual matrix
  # Male
  pval.matrix[pval.matrix$probeID==this_probe,]$M_Estimate <- summary(lm_M)$coefficients[2,1]
  pval.matrix[pval.matrix$probeID==this_probe,]$M_StdError <- summary(lm_M)$coefficients[2,2]
  pval.matrix[pval.matrix$probeID==this_probe,]$M_tvalue <- summary(lm_M)$coefficients[2,3]
  pval.matrix[pval.matrix$probeID==this_probe,]$M_P <- summary(lm_M)$coefficients[2,4]
  
  # Female
  pval.matrix[pval.matrix$probeID==this_probe,]$F_Estimate <- summary(lm_F)$coefficients[2,1]
  pval.matrix[pval.matrix$probeID==this_probe,]$F_StdError <- summary(lm_F)$coefficients[2,2]
  pval.matrix[pval.matrix$probeID==this_probe,]$F_tvalue <- summary(lm_F)$coefficients[2,3]
  pval.matrix[pval.matrix$probeID==this_probe,]$F_P <- summary(lm_F)$coefficients[2,4]
  
  
  # Re-set all values at the end of each cycle
  this_probe <- NULL
  this_ds <- NULL
  lm <- NULL
}

pval.matrix #Estaimtes for Table2
# Done

plot_model(probe1, type = "pred", terms = c("cg12054453", "sex"))
plot_model(probe1, type = "int")

sig_probes_long <- 
  sig_probes %>% 
  gather(probeID, DNAm, cg12054453:cg09349128, factor_key=TRUE) %>%
  mutate(CHR = case_when(
    probeID %in% c("cg12269535") ~ 6,
    probeID %in% c("cg12054453", "cg18181703", "cg11047325", "cg19748455", "cg16936953") ~ 17,
    probeID %in% c("cg00840791") ~ 19,
    probeID %in% c("cg09349128") ~ 22),
          probe_POS = case_when(
    probeID=="cg12269535" ~ 43142014,
    probeID=="cg12054453" ~ 57915717, 
    probeID=="cg18181703" ~ 76354621, 
    probeID=="cg19748455" ~ 76274856, 
    probeID=="cg16936953" ~ 57915665, 
    probeID=="cg11047325" ~ 76354934, 
    probeID=="cg00840791" ~ 16453259,
    probeID=="cg09349128" ~ 50327986),
    order = CHR + probe_POS/max(probe_POS))

library(RNOmni)
sig_probes_long %>% group_by(probeID) %>% mutate(scale_DNAm = RankNorm(DNAm)/4) %>% ungroup %>% 
  # select(probeID, BMI, sex, DNAm, scale_DNAm, order) %>% write.table(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Genome_Biology/Feb2023/Data/DNAm_by_probe.csv", row.names=F, col.names=T, quote=F, sep=",")
  # 
  mutate(agegrp = case_when(age<50 ~ 1,
                            TRUE ~ 2)) %>%
  ggplot(aes(x = DNAm, y = BMI, color = sex)) +
    # geom_point(alpha=0.1, colour="grey") +
    facet_wrap(~ fct_reorder(probeID, desc(-order)), nrow=2) +
    geom_smooth(method = "lm") +
    scale_colour_manual(values = c(mycolours[2], mycolours[1])) +
    theme_linedraw() +
    scale_x_continuous(limits=c(-1, 1), breaks=c(-1, 0, 1), name="DNAm") +
    guides(color=guide_legend(override.aes=list(fill=NA)))


ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/probe_sex_interaction_clean.png", width = 7, height = 5, dpi = 150, units = "in", device='png')


# Checking co-methylation

sig_probes %>% 
  select(-FID, -IID, -trait, -sex) %>%
  as.matrix %>%
  cor %>%
  as.data.frame %>%
  rownames_to_column(var = 'var1') %>%
  gather(var2, value, -var1) %>% 
  filter(var1!=var)

cor(sig_probes$cg18181703, sig_probes$cg11047325)

cor(sig_probes$cg16936953, sig_probes$cg12054453)
cor(sig_probes$cg12054453, sig_probes$cg12269535)
cor(sig_probes$cg12054453, sig_probes$cg12269535)


sig_probes %>% 
  select(BMI, sex, cg18181703) %>% 
  ggplot(aes(x=BMI, y=cg18181703, colour=sex)) + 
  geom_smooth(method = "lm") 
sig_probes$sex2 <- ifelse(sig_probes$sex=="M" , 1, 0)

probes <- list("cg12269535", "cg16936953", "cg12054453", "cg19748455", "cg18181703", "cg11047325", "cg00840791", "cg09349128")
forms <- paste('BMI ~ ', probes)
lapply(forms, function(x) summary(lm(x, data =sig_probes %>% filter(sex=="F")))$coeff) 
lapply(forms, function(x) summary(lm(x, data =sig_probes %>% filter(sex=="M")))$coeff ) 
forms <- paste('BMI ~ ', probes,"*sex")
lapply(forms, function(x) summary(lm(x , data =sig_probes))$coeff ) 


# Investigating probes previously reported
# Wang et al. one CpG showed a statistically significant DNAm × sex interaction effect on BMI status transition across adolescence in the IoW cohort; at cg15125798, boys with higher DNAm were more likely to be in the transition group from normal weight to overweight or obese, but for girls, the association was in the opposite direction (p value for interaction effect: 2.3 × 10–3, Table 3).

# Mendelson et al. a significant sex interaction was demonstrated in the discovery cohorts for one unannotated CpG (cg26651978 on Chromosome 17q25.3; <3 kbp from the 3′ end of LGALS3BP [lectin galactoside-binding soluble 3-binding protein]), 


DNAm <- data.table::fread(paste0("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/interaction_pval.txt"), header=T)
DNAm %>% filter(probeID %in% c("cg15125798", "cg26651978")) %>% select(probeID, interaction_P)
sig_probes <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_sex/DNAm_unrelated_sig_probes_sex.txt", header=T) %>% 
  dplyr::rename(BMI = trait) %>% 
  na.omit()

library(RNOmni)
pvalue <- DNAm %>% filter(probeID %in% c("cg15125798", "cg26651978")) %>% select(probeID, interaction_P) %>% mutate(sex=)

rbind(sig_probes %>% select(BMI, Sex=sex, DNAm=cg15125798) %>% mutate(probeID="cg15125798"),
      sig_probes %>% select(BMI, Sex=sex, DNAm=cg26651978) %>% mutate(probeID="cg26651978")) %>% 
  group_by(probeID) %>% mutate(scale_DNAm = RankNorm(DNAm)/4) %>% ungroup %>% 
  ggplot(aes(x = DNAm, y = BMI, color = Sex)) +
    facet_wrap(~ probeID, ncol=2) +
    geom_smooth(method = "lm") +
    scale_colour_manual(values = c(mycolours[2], mycolours[1])) +
    theme_linedraw() +
    scale_x_continuous(limits=c(-1, 1), breaks=c(-1, 0, 1), name="DNAm") +
    labs(fill = "Sex") +
    geom_text(
    data    = pvalue,
    mapping = aes(x = -Inf, y = -Inf, label = paste0("Pval  = ",interaction_P %>% round(2))),
    colour="black", 
    hjust   = -0.1,
    vjust   = -1)
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Genome_Biology/Feb2023/Figures/SupplementaryDiscusisionFigure.png", width = 8.5, height = 4, dpi = 300, units = "in", device='png')


```

<br> <br>

## Correlation between traits

```{bash Estimate correlation between traits}

# Bivariate REML

# bmi_RankNorm $3
# whr_RankNorm $4
# body_fat_RankNorm $5
# Glucose_RankNorm $6
# HDL_cholesterol_RankNorm $7
# Total_cholesterol_RankNorm $8

scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}
mkdir ${scratch}/DNAm/GREML/traits/${cov}


cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

# for trait1 in bmi whr body_fat Glucose HDL_cholesterol
for trait1 in bmi whr Total_cholesterol
do

if [ "${trait1}" == "bmi" ];              then  var1=3; fi
if [ "${trait1}" == "whr" ];              then  var1=4; fi
if [ "${trait1}" == "body_fat" ];         then  var1=5; fi
if [ "${trait1}" == "Glucose" ];          then  var1=6; fi
if [ "${trait1}" == "HDL_cholesterol" ];  then  var1=7; fi
if [ "${trait1}" == "Total_cholesterol" ]; then  var1=8; fi


for trait2 in height

do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi
if [ "${trait2}" == "height" ]; then  var2=9; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt > temp3.${trait_pair}

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--thread-num 10 \
--pheno temp3.${trait_pair} \
--out ${scratch}/DNAm/GREML/traits/${cov}/${trait_pair}_0

done
done



cd ${scratch}/DNAm/GREML/traits/${cov}
for trait1 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist
do
for trait2 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
awk -v "trait1=${trait1}" -v "trait2=${trait2}"  '{print trait1, trait2, $1, $2, $3}' ${trait1}_${trait2}.hsq >> trait_correlation.hsq
done
done



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/covar_batch_eversmoke_packyears/trait_correlation.hsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/



```

<br>

```{r, eval=TRUE, warning=FALSE, message=FALSE, out.width="100%"}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

library(stringr)
library(readr)
library(tidyverse)
library(forcats)

traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/trait_correlation.hsq") %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of") %>%
mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "V(O)/Vp1",
                       Source %in% c("V(G)/Vp_tr2") ~ "V(O)/Vp2",
                       Source == "rG" ~ "rG",
                       Source == "Pval" ~ "Pval")) %>%
select(-Source) %>% filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait1"
colnames(traits)[2] <- "Trait2"

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits <- traits %>% mutate(Trait1a = ifelse(Trait1=="whr" & Trait2=="body_fat", "body_fat", Trait1),
                            Trait2a = ifelse(Trait1=="whr" & Trait2=="body_fat", "whr", Trait2)) %>%
         select(-Trait1, -Trait2, Variance , SE, Var, Trait1=Trait1a, Trait2= Trait2a ) 

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`V(O)/Vp1`, `V(O)/Vp2`, `rG`, `Pval`),  .sep = '_')


traits_wide$Trait1 <- recode_factor( traits_wide$Trait1, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
traits_wide$Trait2 <- recode_factor( traits_wide$Trait2, 
                             bmi  = "BMI",
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

plot_ds <- traits_wide %>% filter(!Trait1 %in% c("Waist Circumference", "Weight") & !(Trait2 %in% c("Hip Circumference", "Weight", "Waist Circumference")) ) %>%
         mutate(Trait1 = fct_reorder(Trait1, desc(Trait1)),
                label = paste0(rG_Variance %>% signif(digits = 2), " (", rG_SE%>% signif(digits = 1), ")")) 
        
                # label = paste0(rG_Variance %>% signif(digits = 2))) %>% select(Trait1, Trait2, rG_Variance, label)
plot_ds$Trait1 <- plot_ds$Trait1 %>% as.character
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.character
plot_ds$rG_Variance <- plot_ds$rG_Variance %>% as.character

plot_ds <- plot_ds %>% select(Trait1, Trait2, rG_Variance, label)

plot_ds[nrow(plot_ds) + 1,] <- c("BMI", "BMI", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Waist to Hip Ratio", "Waist to Hip Ratio", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Body Fat %", "Body Fat %", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Glucose", "Glucose",  NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("HDL Cholesterol", "HDL Cholesterol",  NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Total Cholesterol", "Total Cholesterol", NA, NA) %>% as.list
# plot_ds[nrow(plot_ds) + 1,] <- c("Height", "Height",  NA, NA) %>% as.list

plot_ds$Trait1 <- plot_ds$Trait1 %>% as.factor
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.factor



plot_ds$Trait1 <- recode_factor( plot_ds$Trait1, 
                             "BMI"  = "BMI", 
                             "Body Fat %" = "Body Fat %", 
                             "Waist to Hip Ratio" = "Waist to Hip Ratio",
                             "Glucose" = "Glucose",
                             "HDL Cholesterol" = "HDL Cholesterol",
                             "Total Cholesterol" = "Total Cholesterol",
                             "Height" = "Height")
plot_ds$Trait2 <- recode_factor( plot_ds$Trait2, 
                             "BMI"  = "BMI", 
                             "Body Fat %" = "Body Fat %", 
                             "Waist to Hip Ratio" = "Waist to Hip Ratio",
                             "Glucose" = "Glucose",
                             "HDL Cholesterol" = "HDL Cholesterol",
                             "Total Cholesterol" = "Total Cholesterol",
                             "Height" = "Height")
plot_ds$rG_Variance <- plot_ds$rG_Variance %>% as.numeric


# plot_ds1 <- plot_ds %>% left_join(traits_wide %>% select(Trait1, Trait2, Pval_Variance), by=c("Trait1", "Trait2"))
r_DNAm_plot <- plot_ds
rg_DNAm <- ggplot(plot_ds
                 %>% filter(Trait1!="Height" & Trait2!="Height" & Trait1!="Total Cholesterol" & Trait2!="BMI") ,
                  # %>% filter(Trait1!="Waist to Hip Ratio"& Trait2!="Body Fat %") ,
       aes(x = Trait2, y = Trait1, fill = rG_Variance)) +
        # aes(x = fct_reorder(Trait1, desc(Trait1)), y = Trait2, fill = rG_Variance)) +
  geom_raster() +
  # geom_point(aes(size=Pval_Variance), shape=15) +
  geom_text(aes(label = label), size=3) +
  # scale_fill_distiller(palette = "Spectral", na.value = 'white',  limits=c(-0.7,1)) +
  theme_minimal() +
  theme(panel.grid = element_blank())+ 
  labs(fill = "rG (DNAm)", y="", x="") +
  scale_y_discrete(limits=rev) +
  scale_fill_gradient2(low = mycolours[1], mid = "beige", high = mycolours[3], na.value = 'white',  limits=c(-0.7,1), guide = "colourbar")  +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=12)) +
  theme(axis.text.y = element_text(size=12)) +
  theme(legend.title = element_text( size=12), legend.text=element_text(size=12))


ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/rG(DNAm).png", width = 8, height = 6, dpi = 150, units = "in", device='png')



# Combine plots
s <- plot_spacer()
library(patchwork)
rg_DNAm + rg_SNP + rg_pheno + plot_layout(ncol = 3, nrow = 1)

# Add option to share legend

library(ggpubr)
ggarrange(
  rg_DNAm +
        theme(axis.title.x=element_blank(),
        # axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        # axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="none") +
        ggtitle("DNAm"),
  rg_SNP +
        theme(axis.title.x=element_blank(),
        # axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="none") +
        ggtitle("Genetic"),
    rg_pheno +
        theme(axis.title.x=element_blank(),
        # axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="right") +
        ggtitle("Phenotypic"), 
  ncol=3, widths= c(1, 0.85, 1.0))
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/rG_combined_lowertri.png", width = 18, height = 6, dpi = 300, units = "in", device='png')




## Thesis figures

library(ggpubr)

ggarrange(
  rg_DNAm +
        theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        # axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="none") +
        ggtitle("DNAm"),
  rg_SNP +
        theme(axis.title.x=element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        # axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="none") +
        ggtitle("Genetic"),
    rg_pheno +
        theme(axis.title.x=element_blank(),
        # axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y=element_blank(),
        # axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
        theme(legend.position="none") +
        ggtitle("Phenotypic"), 
  ncol=1, heights= c(1, 1, 1.5))



ggsave(file="/Users/uqahatt2/Documents/0_Thesis/Chapters/3_DNAm_correlations/Figures/rG_combined_lowertri.png", width = 5, height = 14, dpi = 300, units = "in", device='png')


```

<br>

```{r Correlation BMI with additional phenotypes}

# Bivariate REML

# bmi_RankNorm $3

scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}
mkdir ${scratch}/DNAm/GREML/traits/BMI_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

trait1=bmi
var1=6
for trait2 in ever_smoke heart_disease high_BP diabetes depression osteo_arthritis rheum_arthritis asthma T2D pack_years_RankNorm education_years_RankNorm alcohol_units_RankNorm avg_sys_RankNorm avg_dia_RankNorm avg_hr_RankNorm FEV_RankNorm FVC_RankNorm
do

if [ "${trait2}" == "ever_smoke" ];      then  var2=7; fi
if [ "${trait2}" == "heart_disease" ];   then  var2=8; fi
if [ "${trait2}" == "high_BP" ];         then  var2=9; fi
if [ "${trait2}" == "diabetes" ];        then  var2=10; fi
if [ "${trait2}" == "depression" ];      then  var2=11; fi
if [ "${trait2}" == "osteo_arthritis" ]; then  var2=12; fi
if [ "${trait2}" == "rheum_arthritis" ]; then  var2=13; fi
if [ "${trait2}" == "asthma" ];          then  var2=14; fi
if [ "${trait2}" == "T2D" ];             then  var2=15; fi
if [ "${trait2}" == "pack_years_RankNorm" ]; then  var2=17; fi
if [ "${trait2}" == "education_years_RankNorm" ]; then  var2=19; fi
if [ "${trait2}" == "alcohol_units_RankNorm" ]; then  var2=21; fi
if [ "${trait2}" == "avg_sys_RankNorm" ]; then  var2=23; fi
if [ "${trait2}" == "avg_dia_RankNorm" ]; then  var2=25; fi
if [ "${trait2}" == "avg_hr_RankNorm" ]; then  var2=27; fi
if [ "${trait2}" == "FEV_RankNorm" ]; then  var2=29; fi
if [ "${trait2}" == "FVC_RankNorm" ]; then  var2=31; fi
 
trait_pair=${trait1}-${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' ${scratch}/DNAm/phenotypes/additional_covariates_agesex_ranknorm_outliers4sd.txt > temp.${trait_pair}

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--pheno temp.${trait_pair} \
--out ${scratch}/DNAm/GREML/traits/BMI_${cov}/${trait_pair}_0

done

cd ${scratch}/DNAm/GREML/traits/BMI_${cov}
for trait in ever_smoke heart_disease high_BP diabetes depression osteo_arthritis rheum_arthritis asthma T2D pack_years_RankNorm education_years_RankNorm alcohol_units_RankNorm avg_sys_RankNorm avg_dia_RankNorm avg_hr_RankNorm FEV_RankNorm FVC_RankNorm
do
awk  -F',' '{print FILENAME, $1, $2, $3, $4}' bmi-${trait}_0.hsq  >> traits_0.hsq
awk  -F',' '{print FILENAME, $1, $2, $3, $4}' bmi-${trait}.hsq  >> traits_1.hsq
done

scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/BMI_covar_batch_eversmoke_packyears/traits*.hsq  /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/BMI_covar_batch_eversmoke_packyears
```

```{r, eval=TRUE, warning=FALSE, message=FALSE}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated
# Difference from 1
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/BMI_covar_batch_eversmoke_packyears/traits_1.hsq") %>% 
select(-X5) %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of") %>%
mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "V(O)/V_BMI",
                       Source %in% c("V(G)/Vp_tr2") ~ "V(O)/V_trait2",
                       Source == "rG" ~ "rG",
                       Source == "Pval" ~ "Pval")) %>%
select(-Source) %>% filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait"

traits$Trait <- str_extract(traits$Trait, ".*(?=\\.)")
traits1 <- traits %>% separate(Trait, into = c('Trait1', 'Trait2'), sep="-")


# Difference from 0
traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/BMI_covar_batch_eversmoke_packyears/traits_0.hsq") %>% 
select(-X5) %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of") %>%
mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "V(O)/V_BMI",
                       Source %in% c("V(G)/Vp_tr2") ~ "V(O)/V_trait2",
                       Source == "rG" ~ "rG",
                       Source == "Pval" ~ "Pval")) %>%
select(-Source) %>% filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait"

traits$Trait <- str_extract(traits$Trait, ".*(?=\\.)")
traits0 <- traits %>% separate(Trait, into = c('Trait1', 'Trait2a'), sep="-") %>%
                      separate(Trait2a, into=c("Trait2", "else"), sep="_0") %>% select(-`else`)


traits <- traits1 %>% left_join(traits0 %>% rename(Variance_0=Variance, SE_0=SE), by=c("Trait1", "Trait2", "Var"))

traits$Trait2 <- recode_factor( traits$Trait2, 
                             alcohol_units_RankNorm  = "Alcohol consumption (units/day)", 
                             asthma = "Asthma", 
                             avg_dia_RankNorm = "Diastolic blood pressure ",
                             avg_sys_RankNorm = "Systolic blood pressure",
                             avg_hr_RankNorm = "Heart rate",
                             depression = "Depression",
                             diabetes = "Diabetes",
                             education_years_RankNorm = "Education years",
                             FEV_RankNorm = "Forced expiratory volume",
                             FVC_RankNorm = "Forced vital capacity",
                             heart_disease = "Heart Disease",
                             high_BP = "High blood pressure",
                             osteo_arthritis = "Osteoarthritis",
                             rheum_arthritis = "Rheumatoid arthritis",
                             T2D = "T2D")

traits$Trait1 <- recode_factor( traits$Trait1, 
                             bmi  = "BMI")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`V(O)/V_BMI`, `V(O)/V_trait2`, `rG`, `Pval`),  .sep = '_')
  
traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`V(O)/V_BMI`, `V(O)/V_trait2`, `rG`, `Pval`),  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO/Vp BMI` = paste0(`V(O)/V_BMI_Variance`, " (", `V(O)/V_BMI_SE`, ")"),
         `VO/Vp Trait2` = paste0(`V(O)/V_trait2_Variance`, " (", `V(O)/V_trait2_SE`, ")"),
         `rG` = paste0(rG_Variance, " (", rG_SE, ")")) %>%
  filter(!is.na(`V(O)/V_BMI_Variance`)) %>%
   select(Trait1, Trait2, `VO/Vp BMI`, `VO/Vp Trait2`, rG, rG_pval=Pval_Variance) 
  # kable(caption="Proportion of Variance captured") 


plot_ds_add <-  traits %>% filter(Var %in% c("Pval", "rG")) %>% select(-Variance_0, -SE_0)   %>%  
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`rG`, `Pval`),  .sep = '_') %>%
  arrange(rG_Variance) %>%
  mutate(Pvalue = case_when(Pval_Variance < 0.05 ~ "p < 0.05", TRUE ~ "p >= 0.05"),
         pval = "Difference from 1") %>%
  select(Trait1, Trait2, Pval_Variance, rG_Variance, rG_SE,  Pvalue, pval) %>%
  rbind(
    traits %>% filter(Var %in% c("Pval", "rG")) %>% select(-Variance, -SE)   %>%  
    nest(col_value = c(Variance_0, SE_0)) %>%
    spread(key = Var, value = col_value) %>%
    unnest(c(`rG`, `Pval`),  .sep = '_') %>%
    arrange(rG_Variance_0) %>%
    mutate(Pval_Variance=as.numeric(Pval_Variance_0), Pval_SE=as.numeric(Pval_SE_0), rG_Variance=as.numeric(rG_Variance_0), 
         rG_SE=as.numeric(rG_SE_0), 
         Pvalue = case_when(Pval_Variance < 0.05 ~ "p < 0.05", TRUE ~ "p >= 0.05"), 
         pval = "Difference from 0") %>%
  select(Trait1, Trait2, Pval_Variance, rG_Variance, rG_SE,  Pvalue, pval)
  ) 

plot_ds_add %>%
  ggplot(aes(y=reorder(Trait2, -rG_Variance), x=rG_Variance, xmin=rG_Variance-rG_SE, xmax = rG_Variance+rG_SE, colour=Pvalue)) +
  geom_errorbar(position = position_dodge(width = 0.2), width=.1) +
  geom_point(position = position_dodge(width = 0.2)) +
  labs(x="rG (DNAm) with BMI", y="Trait") +
  facet_wrap(~pval) +
  scale_colour_manual(values=c("red", "gray28"))

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/rG(DNAm)_BMI_additional.png", width = 10, height = 6, dpi = 150, units = "in", device='png')

  

table_add <- traits %>% filter(Var %in% c("Pval", "rG")) %>% select(-Variance_0, -SE_0)   %>%  
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`rG`, `Pval`),  .sep = '_') %>%
  arrange(rG_Variance) %>% 
  mutate(Pval_1= Pval_Variance %>% signif(digits=2)) %>% select(-Pval_SE) %>%
  left_join(
    traits %>% filter(Var %in% c("Pval", "rG")) %>% select(-Variance, -SE)   %>%  
    nest(col_value = c(Variance_0, SE_0)) %>%
    spread(key = Var, value = col_value) %>%
    unnest(c(`rG`, `Pval`),  .sep = '_') %>%
    arrange(rG_Variance_0) %>%
    mutate(Pval_0=as.numeric(Pval_Variance_0) %>% signif(digits=2)) %>%
   select(Trait1, Trait2, Pval_0), by=c("Trait1", "Trait2")) %>%
  mutate(rG=rG_Variance %>% signif(digits=2), rG_SE = rG_SE %>% signif(digits=2)) %>%
  select(Trait1, Trait2, rG_Variance, rG_SE, Pval_0, Pval_1) 
table_add %>% kable
```

## Phenotypic correlation between traits

```{r Phenotypic correlation}
library(tidyverse)
out="/Local_Scratch/Alesha/DNAm/phenotypes/descriptive"
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)

traits <- c("bmi", "whr", "body_fat", "Glucose", "HDL_cholesterol", "Total_cholesterol", "height", "weight", "waist", "hips")

corr_with_RankNorm <- data.frame("Trait" = traits, 
                                   "Correlation with RankNorm" = c(
    cor(ds_pheno$bmi, ds_pheno$bmi_RankNorm, use = "complete.obs"),
    cor(ds_pheno$whr, ds_pheno$whr_RankNorm, use = "complete.obs"),
    cor(ds_pheno$body_fat, ds_pheno$body_fat_RankNorm, use = "complete.obs"),
    cor(ds_pheno$Glucose, ds_pheno$Glucose_RankNorm, use = "complete.obs"), 
    cor(ds_pheno$HDL_cholesterol, ds_pheno$HDL_cholesterol_RankNorm, use = "complete.obs"),
    cor(ds_pheno$Total_cholesterol, ds_pheno$Total_cholesterol_RankNorm, use = "complete.obs"),
    cor(ds_pheno$height, ds_pheno$height_RankNorm, use = "complete.obs"),
    cor(ds_pheno$weight, ds_pheno$weight_RankNorm, use = "complete.obs"),
    cor(ds_pheno$waist, ds_pheno$waist_RankNorm, use = "complete.obs"),
    cor(ds_pheno$hips, ds_pheno$hips_RankNorm, use = "complete.obs")))

write.table(corr_with_RankNorm, file=paste0(out,"/corr_with_RankNorm.txt"), row.names=F, col.names=T, quote=F)

# Between traits
library("Matrix")
library("psych")
pheno_cor <- ds_pheno %>% select(ends_with("RankNorm")) %>%
  rename_with(~str_remove(., '_RankNorm')) %>% corr.test

pheno_cor_tab <- pheno_cor$r %>% unlist %>% as.data.frame 
pheno_cor_se <- pheno_cor$se %>% unlist %>% as.data.frame 


write.table(pheno_cor_tab, file=paste0(out,"/pheno_correlations.txt"), col.names = T, row.names = T, quote = F)
write.table(pheno_cor_se, file=paste0(out,"/pheno_correlations_se.txt"), col.names = T, row.names = T, quote = F)



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/descriptive/* /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes

```

```{r, eval=TRUE}


corr_with_RankNorm <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes/corr_with_RankNorm.txt", header=T) %>% mutate(Correlation.with.RankNorm=Correlation.with.RankNorm %>% signif(2))

corr_with_RankNorm$Trait <- recode_factor( corr_with_RankNorm$Trait, 
                             bmi  = "BMI", 
                             whr = "Waist to Hip Ratio",
                             body_fat = "Body Fat %", 
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

corr_with_RankNorm %>% select(Trait, Correlation.with.RankNorm) %>% kable(caption=" Correlation between each trait and rank normalised residuals")


pheno_correlations <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes/pheno_correlations.txt", header=T) 
names <- recode_factor(rownames(pheno_correlations) , 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
colnames(pheno_correlations) = rownames(pheno_correlations) = names
pheno_correlations <- pheno_correlations[c(1, 3, 2, 4, 5), c(3, 2, 4, 5, 6)]

pheno_correlations_mat <- pheno_correlations %>% as.matrix %>% signif(2)
pheno_correlations_mat[lower.tri(pheno_correlations_mat)] <- NA

options(knitr.kable.NA = '')
pheno_correlations_mat %>% kable(caption = "Phenotypic correlation between traits")


pheno_correlations_se <- read.table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/phenotypes/pheno_correlations_se.txt", header=T)
colnames(pheno_correlations_se) = rownames(pheno_correlations_se) = names
pheno_correlations_se_mat <- pheno_correlations_se[c(1, 3, 2, 4, 5), c(3, 2, 4, 5, 6)]
pheno_correlations_se_mat <- pheno_correlations_se_mat %>% as.matrix %>% signif(2)
pheno_correlations_se_mat[lower.tri(pheno_correlations_se_mat)] <- NA



library(reshape2)
pheno <- reshape2::melt(pheno_correlations_mat) %>% mutate(Trait1=Var2, Trait2=Var1)
pheno_se <- reshape2::melt(pheno_correlations_se_mat) %>% mutate(Trait1=Var2, Trait2=Var1)

pheno <- pheno %>% left_join(pheno_se %>% select(Trait1, Trait2, se=value), by=c("Trait1", "Trait2"))

                          
plot_ds <- pheno %>% filter(
  # !Trait1 %in% c("Waist Circumference", "Weight", "Hip Circumference") & !(Trait2 %in% c("Hip Circumference", "Weight", "Waist Circumference")) & Trait1!=Trait2 & 
  !is.na(value) ) %>%
         mutate(Trait1 = fct_reorder(Trait1, desc(Trait1)),
                  label = paste0(value %>% signif(digits = 2), " (", se %>% signif(digits=1), ")"))  %>% 
  select(Trait1, Trait2, value, label)

plot_ds$Trait1 <- plot_ds$Trait1 %>% as.character
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.character
plot_ds$rG_Variance <- plot_ds$value %>% as.character



plot_ds[nrow(plot_ds) + 1,] <- c("BMI", "BMI", NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Waist to Hip Ratio", "Waist to Hip Ratio", NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Body Fat %", "Body Fat %", NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Glucose", "Glucose",  NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("HDL Cholesterol", "HDL Cholesterol",  NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Total Cholesterol", "Total Cholesterol", NA, NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Height", "Height",  NA, NA, NA) %>% as.list

plot_ds$Trait1 <- plot_ds$Trait1 %>% as.factor
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.factor

# plot_ds <- plot_ds %>% mutate(Trait1a = ifelse(Trait1=="Waist to Hip Ratio" & Trait2=="Body Fat %" , "Body Fat %", as.character(Trait1)),
#                    Trait2a = ifelse(Trait1=="Waist to Hip Ratio" & Trait2=="Body Fat %" , "Waist to Hip Ratio",  as.character(Trait2)))
#   
plot_ds$Trait1 <- recode_factor( plot_ds$Trait1, 
                             "BMI"  = "BMI", 
                             "Body Fat %" = "Body Fat %", 
                             "Waist to Hip Ratio" = "Waist to Hip Ratio",
                             "Glucose" = "Glucose",
                             "HDL Cholesterol" = "HDL Cholesterol",
                             "Total Cholesterol" = "Total Cholesterol",
                             "Height" = "Height")
plot_ds$Trait2 <- recode_factor( plot_ds$Trait2, 
                             "BMI"  = "BMI", 
                             "Body Fat %" = "Body Fat %", 
                             "Waist to Hip Ratio" = "Waist to Hip Ratio",
                             "Glucose" = "Glucose",
                             "HDL Cholesterol" = "HDL Cholesterol",
                             "Total Cholesterol" = "Total Cholesterol",
                             "Height" = "Height")
plot_ds$value <- plot_ds$value %>% as.numeric


r_pheno_plot <- plot_ds  %>% filter(Trait1!="Height" & Trait2!="Height" & Trait1!="BMI" & Trait2!="Total Cholesterol") %>%
  rename(Trait1=Trait2, Trait2=Trait1) 



rg_pheno <- ggplot(plot_ds %>% filter(Trait1!="Height" & Trait2!="Height" & Trait1!="BMI" & Trait2!="Total Cholesterol"),
       aes(x = Trait1, y = Trait2, fill = value)) +
       # aes(x = fct_reorder(Trait2, desc(Trait2)), y = Trait1, fill = value)) +
  geom_raster() +
  geom_text(aes(label = label), size=3) +
  scale_fill_gradient2(low = mycolours[1], mid = "beige", high = mycolours[3], na.value = 'white',  
                       limits=c(-0.7,1), guide = "colourbar")  +
  theme_minimal() +
  theme(panel.grid = element_blank())+ 
  labs(fill = "Correlation", y="", x="") +
  scale_y_discrete(limits=rev) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=12)) +
  theme(axis.text.y = element_text(size=12)) +
  theme(legend.title = element_text( size=12), legend.text=element_text(size=12)) 

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/rG(pheno).png", width = 8, height = 6, dpi = 150, units = "in", device='png')



```



<br>

## Genetic and epigenetic correlation between traits

We also attempted to partition and estimate the proportions of phenotypic variation captured by all SNPs (i.e., hSNP2) and all the DNAm probes respectively when fitted jointly in a model.

```{bash GRM & ORM}
# Rename GRM ID's to match DNAm ID's
# Save GRM ready for analysis
library(tidyverse)
ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)
id <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_DNAm_0.05.grm.id", header=F)

id %>% 
left_join(ds, by=c("V2"="IID")) %>%
select(FID_DNAm, IID_DNAm) %>%
write.table("/Local_Scratch/Alesha/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated.grm.id", col.names=F, row.names=F, quote=F)

# cp GS_GWAS to wd
scratch=/Local_Scratch/Alesha
filename=mvals-norm20k-18413-831733

cp /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_DNAm_0.05.grm.N.bin ${scratch}/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated.grm.N.bin
cp /Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS_DNAm_0.05.grm.bin ${scratch}/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated.grm.bin

echo ${scratch}/DNAm/ORM/GRM/${filename}_sd0.02_unrelated_covar > ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt
echo ${scratch}/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated >> ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt



scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}
mkdir ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM


cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

for trait1 in body_fat
do

if [ "${trait1}" == "bmi" ];              then  var1=3; fi
if [ "${trait1}" == "whr" ];              then  var1=4; fi
if [ "${trait1}" == "body_fat" ];         then  var1=5; fi
if [ "${trait1}" == "Glucose" ];          then  var1=6; fi
if [ "${trait1}" == "HDL_cholesterol" ];  then  var1=7; fi
if [ "${trait1}" == "Total_cholesterol" ]; then  var1=8; fi
if [ "${trait1}" == "height" ]; then  var1=9; fi


for trait2 in   Glucose HDL_cholesterol Total_cholesterol height
do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi
if [ "${trait2}" == "height" ]; then  var2=9; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt > temp.${trait_pair}_grm_orm

gcta64 --reml-bivar 1 2 \
--mgrm ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt \
--reml-bivar-lrt-rg 1 \
--reml-bivar-no-constrain \
--pheno temp.${trait_pair}_grm_orm \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM/${trait_pair}_no-constrain


gcta64 --reml-bivar 1 2 \
--mgrm ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt \
--reml-bivar-lrt-rg 1 \
--reml-bivar-nocove \
--pheno temp.${trait_pair}_grm_orm \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM/${trait_pair}_nocove

gcta64 --HEreg-bivar 1 2 \
--mgrm ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt \
--reml-bivar-lrt-rg 1 \
--pheno temp.${trait_pair}_grm_orm \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM/${trait_pair}_HEreg


gcta64 --reml-bivar 1 2 \
--mgrm ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt \
--reml-bivar-lrt-rg 1 \
--reml-alg 1 \
--pheno temp.${trait_pair}_grm_orm \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM/${trait_pair}_AI

gcta64 --reml-bivar 1 2 \
--mgrm ${scratch}/DNAm/GREML/traits/GRM_ORM/GRM_ORM.txt \
--reml-bivar-lrt-rg 1 \
--reml-alg 2 \
--pheno temp.${trait_pair}_grm_orm \
--thread-num 30 \
--out ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM/${trait_pair}_EM


done
done





cd ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM
for trait1 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist
do
for trait2 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
awk -v "trait1=${trait1}" -v "trait2=${trait2}"  '{print trait1, trait2, $1, $2, $3}' ${trait1}_${trait2}.hsq >> trait_correlation_GRM_ORM.hsq
done
done



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/covar_batch_eversmoke_packyears/GRM_ORM/trait_correlation_GRM_ORM.hsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/mQTL_and_Illumina_annotation_files/epic_probes_to_exclude/ /Users/uqahatt2/Documents/2_project/4_GS/Cluster/Additional_info_cluster_filespace_cohorts

scp -P 8017 -r v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/covar_batch_eversmoke_packyears/GRM_ORM /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/GRM_ORM/methods

```

```{r, eval=TRUE}

traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/trait_correlation_GRM_ORM.hsq") %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of") %>%
mutate(Var = case_when(Source %in% c("V(G1)/Vp_tr1") ~ "V(O)/Vp1",
                       Source %in% c("V(G1)/Vp_tr2") ~ "V(O)/Vp2",
                       Source %in% c("V(G2)/Vp_tr1") ~ "V(G)/Vp1",
                       Source %in% c("V(G2)/Vp_tr2") ~ "V(G)/Vp2",
                       Source == "rG1" ~ "rO",
                       Source == "rG2" ~ "rG")) %>%
select(-Source) %>% filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait1"
colnames(traits)[2] <- "Trait2"

traits$Trait1 <- recode_factor( traits$Trait1, 
                             bmi  = "BMI", 
                             whr = "Waist to Hip Ratio",
                             body_fat = "Body Fat %", 
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
traits$Trait2 <- recode_factor( traits$Trait2, 
                             bmi  = "BMI", 
                             whr = "Waist to Hip Ratio",
                             body_fat = "Body Fat %", 
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

options(knitr.kable.NA = '')
traits %>%
  filter(Var %in% c("rO", "rG")) %>% 
  mutate(Variance = Variance %>% signif(digits = 2),
         SE = SE %>% signif(digits = 2),
         Var_SE = paste0(Variance, " (", SE,")")) %>%
  select(Trait1, Trait2, Var_SE, Var) %>%
  spread(key = Var, value = Var_SE) %>%
  mutate(Correlation = paste0(rO, "<br>", rG)) %>%
  select(Trait1, Trait2, Correlation) %>%
  spread(key=Trait2, value=Correlation) %>%
  kable(format = 'latex', booktabs = TRUE, escape = FALSE)

  # kable(caption="Epigenetic Correlations")



# Heatmap 
# library(RColorBrewer)
# library(ggplot2)
# library(forcats)
# 
# ggplot(traits %>% filter(Var=="rO") %>% mutate(Trait1 = fct_reorder(Trait1, desc(Trait1))), 
#        aes(x = Trait2, y = Trait1, fill = Variance)) +
#   geom_raster() +
#   geom_text(aes(label = Variance)) +
#   scale_fill_distiller(palette = "Spectral") +
#   theme_minimal() +
#   theme(panel.grid = element_blank())
# 
# ggplot(traits %>% filter(Var=="rG") %>% mutate(Trait1 = fct_reorder(Trait1, desc(Trait1))), 
#        aes(x = Trait2, y = Trait1, fill = Variance)) +
#   geom_raster() +
#   geom_text(aes(label = Variance)) +
#   scale_fill_distiller(palette = "Spectral") +
#   theme_minimal() +
#   theme(panel.grid = element_blank())


traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  select(Trait1, Trait2, rO, rG) %>%
  unnest(c(`rO`,  `rG`),  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(
    # `VO/Vp Trait1` = paste0(`V(O)/Vp1_Variance`, " (", `V(O)/Vp1_SE`, ")"),
    # `VO/Vp Trait2` = paste0(`V(O)/Vp2_Variance`, " (", `V(O)/Vp2_SE`, ")"),
      `rO` = paste0(rO_Variance, " (", rO_SE, ")"),
      `rG` = paste0(rG_Variance, " (", rG_SE, ")")) %>%
 select(Trait1, Trait2, `rO`, `rG`) %>%
  kable(caption="Epigenetic and genetic correlations")
```

<br>

## Genetic correlation between traits

```{bash GRM}

# cp GRM to dir
scratch=/Local_Scratch/Alesha/DNAm/GREML/traits/
cp ${scratch}/GRM_ORM/GS_GWAS_unrelated.grm.id ${scratch}/GRM/GS_GWAS_unrelated.grm.id
cp ${scratch}/GRM_ORM/GS_GWAS_unrelated.grm.N.bin ${scratch}/GRM/GS_GWAS_unrelated.grm.N.bin
cp ${scratch}/GRM_ORM/GS_GWAS_unrelated.grm.bin ${scratch}/GRM/GS_GWAS_unrelated.grm.bin

# bmi_RankNorm $3
# whr_RankNorm $4
# body_fat_RankNorm $5
# Glucose_RankNorm $6
# HDL_cholesterol_RankNorm $7
# Total_cholesterol_RankNorm $8

scratch=/Local_Scratch/Alesha
cd ${scratch}/DNAm/phenotypes/greml.pheno/temp


#for trait1 in bmi whr body_fat Glucose HDL_cholesterol
#do

trait1=bmi
if [ "${trait1}" == "bmi" ];              then  var1=3; fi
if [ "${trait1}" == "whr" ];              then  var1=4; fi
if [ "${trait1}" == "body_fat" ];         then  var1=5; fi
if [ "${trait1}" == "Glucose" ];          then  var1=6; fi
if [ "${trait1}" == "HDL_cholesterol" ];  then  var1=7; fi
if [ "${trait1}" == "Total_cholesterol" ]; then  var1=8; fi
if [ "${trait1}" == "height" ]; then  var1=9; fi

#for trait2 in  height
for trait2 in  body_fat 
do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi
if [ "${trait2}" == "height" ]; then  var2=9; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt > temp.${trait_pair}_grm

# --grm ${scratch}/DNAm/GREML/traits/GRM/GS_GWAS_unrelated \ -->

gcta64 --reml-bivar 1 2 \
--grm /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3 \
--reml-bivar-lrt-rg 0 \
--pheno temp.${trait_pair}_grm \
--reml-bivar-no-constrain \
--thread-num 10 \
--out ${scratch}/DNAm/GREML/traits/GRM/${trait_pair}_imputed_hm3_0

done
#done




cd ${scratch}/DNAm/GREML/traits/GRM
for trait1 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height
do
for trait2 in  whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do
awk -v "trait1=${trait1}" -v "trait2=${trait2}"  '{print trait1, trait2, $1, $2, $3}' ${trait1}_${trait2}_imputed_hm3.hsq >> trait_correlation_GRM_imputed_hm3.hsq
done
done



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/GRM/trait_correlation_GRM_imputed_hm3.hsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/


```




```{r, eval=TRUE, warning=FALSE, message=FALSE, out.width="100%"}
# Order is not importance. Same results are obtained either way e.g. GRM + ORM vs ORM + GRM Compared to literature both V(O) and V(G) are overestimated

library(stringr)
library(readr)
library(tidyverse)

traits <- read_table("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/trait_correlation_GRM_imputed_hm3.hsq") %>% 
filter(!is.na(SE) & SE!="SE" & Variance!="of") %>%
mutate(Var = case_when(Source %in% c("V(G)/Vp_tr1") ~ "V(O)/Vp1",
                       Source %in% c("V(G)/Vp_tr2") ~ "V(O)/Vp2",
                       Source == "rG" ~ "rG",
                       Source == "Pval" ~ "Pval")) %>%
select(-Source) %>% filter(!is.na(Var)) 

colnames(traits)[1] <- "Trait1"
colnames(traits)[2] <- "Trait2"

traits <- traits %>% mutate(Trait1a = ifelse(Trait1=="whr" & Trait2=="body_fat", "body_fat", Trait1),
                            Trait2a = ifelse(Trait1=="whr" & Trait2=="body_fat", "whr", Trait2)) %>%
         select(-Trait1, -Trait2, Variance , SE, Var, Trait1=Trait1a, Trait2= Trait2a ) 


traits$Trait1 <- recode_factor( traits$Trait1, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
traits$Trait2 <- recode_factor( traits$Trait2, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")

# Recode to numeric
traits$Variance <- traits$Variance %>% as.numeric
traits$SE <- traits$SE %>% as.numeric

traits_wide <- traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`V(O)/Vp1`, `V(O)/Vp2`, `rG`, `Pval`),  .sep = '_')

# options(knitr.kable.NA = '')
# traits_wide %>% 
#   mutate(rG = rG_Variance %>% signif(digits = 2)) %>% 
#   select(Trait1, Trait2, rG) %>%
#   spread(key = Trait2, value = rG) %>%
#   kable(caption="Epigenetic Correlations") 


# Heatmap 
# library(RColorBrewer)
# library(ggplot2)
library(forcats)

plot_ds <- traits_wide %>% filter(!Trait1 %in% c("Waist Circumference", "Weight") & !(Trait2 %in% c("Hip Circumference", "Weight", "Waist Circumference")) ) %>%
         mutate(Trait1 = fct_reorder(Trait1, desc(Trait1)),
                label = paste0(rG_Variance %>% signif(digits = 2), " (", rG_SE %>% signif(digits = 1), ")"))
                # label = paste0(rG_Variance %>% signif(digits = 2))) %>% select(Trait1, Trait2, rG_Variance, label)
plot_ds$Trait1 <- plot_ds$Trait1 %>% as.character
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.character
plot_ds$rG_Variance <- plot_ds$rG_Variance %>% as.character


plot_ds <- plot_ds %>% select(Trait1, Trait2, rG_Variance, label)
plot_ds[nrow(plot_ds) + 1,] <- c("BMI", "BMI", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Waist to Hip Ratio", "Waist to Hip Ratio", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Body Fat %", "Body Fat %", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Glucose", "Glucose",  NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("HDL Cholesterol", "HDL Cholesterol",  NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Total Cholesterol", "Total Cholesterol", NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Height", "Height",  NA, NA) %>% as.list
plot_ds[nrow(plot_ds) + 1,] <- c("Total Cholesterol", "HDL Cholesterol",  NA, NA) %>% as.list


plot_ds$Trait1 <- plot_ds$Trait1 %>% as.factor
plot_ds$Trait2 <- plot_ds$Trait2 %>% as.factor
plot_ds$Trait1 <- recode_factor( plot_ds$Trait1, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
plot_ds$Trait2 <- recode_factor( plot_ds$Trait2, 
                             bmi  = "BMI", 
                             body_fat = "Body Fat %", 
                             whr = "Waist to Hip Ratio",
                             Glucose = "Glucose",
                             HDL_cholesterol = "HDL Cholesterol",
                             Total_cholesterol = "Total Cholesterol",
                             height = "Height",
                             weight = "Weight",
                             waist = "Waist Circumference",
                             hips = "Hip Circumference")
plot_ds$rG_Variance <- plot_ds$rG_Variance %>% as.numeric

plot_ds2 <- plot_ds %>% 
  filter( !(Trait1=="Waist to Hip Ratio" & Trait2 %in% c("BMI")) &
          !(Trait1=="Total Cholesterol" & Trait2 %in% c("BMI",  "Waist to Hip Ratio", "Body Fat %", "Glucose", "HDL Cholesterol","Height")) &
          !(Trait1=="Height" & Trait2 %in% c("BMI", "Waist to Hip Ratio")))


r_G_plot <- plot_ds2
rg_SNP <- ggplot(plot_ds2 %>% filter(Trait1!="Height" & Trait2!="Height" & Trait1!="Total Cholesterol" & Trait2!="" & !(Trait1=="BMI" & Trait2=="BMI")),
       aes(x = Trait2, y = Trait1, fill = rG_Variance)) +
  geom_raster() +
  geom_text(aes(label = label), size=3) +
  # scale_fill_distiller(palette = "Spectral", na.value = 'white',  limits=c(-0.7,1)) +
  scale_fill_gradient2(low = mycolours[1], mid = "beige", high = mycolours[3], na.value = 'white',  limits=c(-0.7,1), guide = "colourbar")  +
  theme_minimal() +
  theme(panel.grid = element_blank())+ 
  labs(fill = "Correlation", y="", x="") +
  scale_y_discrete(limits=rev) +

  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=12)) +
  theme(axis.text.y = element_text(size=12)) +
  theme(legend.title = element_text( size=12), legend.text=element_text(size=12)) 

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/Figures/excluding height/rG(DNA)_imputed_hm3.png", width = 8, height = 6, dpi = 150, units = "in", device='png')




rg <- traits_wide

traits %>%
  nest(col_value = c(Variance, SE)) %>%
  spread(key = Var, value = col_value) %>%
  unnest(c(`V(O)/Vp1`, `V(O)/Vp2`, `rG`, `Pval`),  .sep = '_') %>%
  mutate(across(where(is.character), is.numeric)) %>%
  mutate(across(where(is.numeric), round, 3)) %>%
  mutate(`VO/Vp Trait1` = paste0(`V(O)/Vp1_Variance`, " (", `V(O)/Vp1_SE`, ")"),
         `VO/Vp Trait2` = paste0(`V(O)/Vp2_Variance`, " (", `V(O)/Vp2_SE`, ")"),
         `rG` = paste0(rG_Variance, " (", rG_SE, ")")) %>%
 select(Trait1, Trait2, `VO/Vp Trait1`, `VO/Vp Trait2`, rG, rG_pval=Pval_Variance) %>%
  kable(caption="Proportion of Variance captured") 
```




```{r Combine all heatmaps}
r_DNAm_plot 
r_G_plot 
r_pheno_plot

merge_heatmap <- rbind(
  r_DNAm_plot %>% select(Trait1, Trait2,  rG_Variance, label) %>% mutate(var="DNAm"),
  r_G_plot %>% select(Trait1, Trait2,  rG_Variance, label) %>% mutate(var="Genetic"),
  r_pheno_plot %>% select(Trait1, Trait2,  rG_Variance, label) %>% mutate(var="Phenotypic")) %>%
  filter(Trait2!="BMI" & Trait1!="Height" & Trait2!="Height" & Trait1!="Total Cholesterol" & Trait1!=Trait2) %>%
  mutate(rG_Variance = rG_Variance %>% as.numeric %>% round(2))


ggplot(merge_heatmap ,
    aes(x = Trait2, y = Trait1, fill = rG_Variance)) +
  geom_raster() +
  geom_text(aes(label =  rG_Variance), size=3) +
  theme_minimal() +
  theme(panel.grid = element_blank())+ 
  labs(fill = "Correlation", y="", x="") +
  scale_y_discrete(limits=rev) +
  scale_fill_gradient2(low = mycolours[1], mid = "beige", high = mycolours[3], na.value = 'white',  limits=c(-0.7,1), guide = "colourbar")  +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, size=8)) +
  theme(axis.text.y = element_text(size=8)) +
  theme(legend.title = element_text( size=8), legend.text=element_text(size=7)) +
  facet_grid(~var) +
  theme(strip.text.x = element_text(size = 11, colour = "black")) +
  theme(panel.spacing = unit(0.1, "lines"))

ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/manuscript/Genome_Biology/Feb2023/Figure2.png", width = 8.5, height = 3.75, dpi = 300, units = "in", device='png')
 
```


## Epigenetic correlation for Males and Females

```{bash Correlation for Males and Females}

# Bivariate REML

# bmi_RankNorm $3
# whr_RankNorm $4
# body_fat_RankNorm $5
# Glucose_RankNorm $6
# HDL_cholesterol_RankNorm $7
# Total_cholesterol_RankNorm $8

# Make Male and Female ds
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)
ds_pheno %>% filter(sex=="M") %>% select(FID_DNAm, IID_DNAm, bmi_RankNorm, whr_RankNorm, body_fat_RankNorm ,  Glucose_RankNorm, HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm) %>%
write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Males.txt", col.name=T, row.name=F, quote=F)

ds_pheno %>% filter(sex=="F") %>% select(FID_DNAm, IID_DNAm, bmi_RankNorm, whr_RankNorm, body_fat_RankNorm ,  Glucose_RankNorm, HDL_cholesterol_RankNorm, Total_cholesterol_RankNorm) %>%
write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Females.txt", col.name=T, row.name=F, quote=F)


scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}
mkdir ${scratch}/DNAm/GREML/traits/${cov}/GRM_ORM

mkdir ${scratch}/DNAm/GREML/traits/Male
mkdir ${scratch}/DNAm/GREML/traits/Female

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

trait1=bmi
var1=3
for trait2 in whr body_fat Glucose HDL_cholesterol Total_cholesterol
do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi

trait_pair=${trait1}_${trait2}

# Male
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Males.txt > temp.${trait_pair}_Male

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 1 \
--pheno temp.${trait_pair}_Male \
--out ${scratch}/DNAm/GREML/traits/Male/${trait_pair}_Male

# Female
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno_Females.txt > temp.${trait_pair}_Female

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 1 \
--pheno temp.${trait_pair}_Female \
--out ${scratch}/DNAm/GREML/traits/Female/${trait_pair}_Female

done

# scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/*hsq /Users/uqahatt2/Documents/2_project/4_GS/Data_preparation/GREML_traits/


```


## Pre-adjusting for kinship

```{r adjusting for kinship}

scp -P 8017 /Volumes/marioni-lab/Generation_Scotland_data_Sep2021/2022-01-17_pedigree.csv v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/

library(dplyr)  
library(kinship2)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliers4sd.txt", header=T)
ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov.txt", header=T)
oii <- read.table("/Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.orm.id")

ped_rds_path = "/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/kinship_matrix_using_fixed_2022-01-28-pedigree.rds"
ped <- read.csv("/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/2022-01-17_pedigree.csv")
ds_pheno <- ds_pheno %>% left_join(ds %>% select( FID_DNAm, IID_DNAm,  FID, IID, age), by=c("FID_DNAm", "IID_DNAm")) %>%
  mutate(sex=as.factor(sex)) %>% filter(IID_DNAm %in% oii$V2)
table(ds_pheno$IID %in% ped$volid)

# How many people with another family member
ped_n <- ped %>% filter(volid %in% ds_pheno$IID) 
ped_n %>% group_by(famid) %>% filter(n() == 2 ) %>% as.data.frame %>% dim # 2116 individuals with unrelated people in the family
counts <- ped_n %>% group_by(famid) %>% summarise(count = n_distinct(volid)) 
table(counts$count)
#    1    2    3    4    5    6 
# 4894 1058  126   18    7    4 

# # Families of 6
# 1  1346     6
# 2  2979     6
# 3  3636     6
# 4  6828     6

ds_pheno %>% filter(FID==1346) %>% select(FID_DNAm) 
# 1 200826750088_R06C01
# 2 200876170046_R04C01
# 3 200875800044_R02C01
# 4 200876170037_R02C01
# 5 200876170002_R08C01
# 6 200875800029_R03C01
#   famid  volid father mother sex age
# 1  1346   7013   5942 159236   F  71
# 2  1346  81963  15433 159648   M  44
# 3  1346  88448  55356 134470   F  60
# 4  1346 140738  75608  44449   F  69
# 5  1346 101828 121489  49331   M  44
# 6  1346  74484 999056 106893   F  22

# Extract GRM value for these people..
echo 200826750088_R06C01 200876170046_R04C01 200875800044_R02C01 200876170037_R02C01 200876170002_R08C01 200875800029_R03C01 > /Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/famid_1346.txt

gcta64 \
--threads 30 \
--grm /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3_0.05 \
--keep /Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/famid_1346.txt \
--make-grm \
--out /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/famid_1346

library(tidyverse)
# GRM function
grm="/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/famid_1346"
prefix_grm=paste0(grm, sep="")
ReadGRMBin=function(prefix, AllN=F, size=4){
  sum_i=function(i){
    return(sum(1:i))
  }
  BinFileName=paste(prefix,".grm.bin",sep="")
  NFileName=paste(prefix,".grm.N.bin",sep="")
  IDFileName=paste(prefix,".grm.id",sep="")
  id = read.table(IDFileName)
  n=dim(id)[1]
  BinFile=file(BinFileName, "rb");
  grm=readBin(BinFile, n=n*(n+1)/2, what=numeric(0), size=size)
  NFile=file(NFileName, "rb");
  if(AllN==T){
    N=readBin(NFile, n=n*(n+1)/2, what=numeric(0), size=size)
  }
  else N=readBin(NFile, n=1, what=numeric(0), size=size)
  i=sapply(1:n, sum_i)
  return(list(diag=grm[i], off=grm[-i], id=id, N=N))
}

GRM = ReadGRMBin(prefix_grm)
library(reshape2)
GRM.matrix = data.frame(matrix(0, ncol=nrow(GRM$id), nrow=nrow(GRM$id)))
colnames(GRM.matrix)=GRM$id$V2
rownames(GRM.matrix)=GRM$id$V2
GRM.matrix[lower.tri(GRM.matrix)] <- NA

GRM.ds <- melt(as.matrix(GRM.matrix))
GRM.ds <- GRM.ds[complete.cases(GRM.ds), ]
GRM.ds <- GRM.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(GRM = GRM$off) %>% 
  select(-value)

# > GRM.ds
#                   Var1                Var2           GRM
# 1  200875800029_R03C01 200826750088_R06C01 -5.489559e-03
# 2  200875800029_R03C01 200876170046_R04C01 -8.397039e-04
# 3  200826750088_R06C01 200876170046_R04C01 -2.929975e-03
# 4  200875800029_R03C01 200875800044_R02C01 -4.176658e-03
# 5  200826750088_R06C01 200875800044_R02C01 -4.371815e-03
# 6  200876170046_R04C01 200875800044_R02C01 -6.447181e-04
# 7  200875800029_R03C01 200876170002_R08C01 -7.890753e-05
# 8  200826750088_R06C01 200876170002_R08C01 -9.766904e-04
# 9  200876170046_R04C01 200876170002_R08C01  3.677856e-03
# 10 200875800044_R02C01 200876170002_R08C01  1.497912e-03
# 11 200875800029_R03C01 200876170037_R02C01 -3.771286e-03
# 12 200826750088_R06C01 200876170037_R02C01  9.626647e-03
# 13 200876170046_R04C01 200876170037_R02C01 -4.968245e-03
# 14 200875800044_R02C01 200876170037_R02C01  3.076000e-03
# 15 200876170002_R08C01 200876170037_R02C01 -6.444100e-04




          
kin <- with(ped, pedigree(volid, father, mother, sex, famid=famid))
# value of 'momid' not found in the id list 4091/103027

id_with_issues = as.numeric(rownames(ped[which(ped$mother %in% 103027), ]))
ped[id_with_issues, "mother"] = 0
ped[id_with_issues, "father"] = 0
  
kin <- with(ped, pedigree(volid, father, mother, sex, famid=famid))

# ped_fixed <- with(ped,fixParents(id = volid, dadid=father, momid = mother, sex = sex))
# kin <- with(ped_fixed, pedigree(id, dadid, momid, sex))
kin_model <- kinship(kin) 

library(coxme)

# BMI
ds_bmi <- ds_pheno %>% filter(!is.na(bmi_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, bmi, bmi_RankNorm)
ds_bmi$bmi_kin <- scale(resid(lmekin(bmi ~ age + sex + (1|IID), data=ds_bmi, varlist = kin_model*2)))
ds_bmi$bmi_RankNorm_kin <- scale(resid(lmekin(bmi_RankNorm ~ (1|IID), data=ds_bmi, varlist = kin_model*2)))

# WHR
ds_whr <- ds_pheno %>% filter(!is.na(whr_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, whr, whr_RankNorm)
ds_whr$whr_kin <- scale(resid(lmekin(whr ~ age + sex + (1|IID), data=ds_whr, varlist = kin_model*2)))
ds_whr$whr_RankNorm_kin <- scale(resid(lmekin(whr_RankNorm ~ (1|IID), data=ds_whr, varlist = kin_model*2)))

# Body Fat
ds_body_fat <- ds_pheno %>% filter(!is.na(body_fat_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, body_fat, body_fat_RankNorm)
ds_body_fat$body_fat_kin <- scale(resid(lmekin(body_fat ~ age + sex + (1|IID), data=ds_body_fat, varlist = kin_model*2)))
ds_body_fat$body_fat_RankNorm_kin <- scale(resid(lmekin(body_fat_RankNorm ~ (1|IID), data=ds_body_fat, varlist = kin_model*2)))

# Glucose
ds_Glucose <- ds_pheno %>% filter(!is.na(Glucose_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, Glucose, Glucose_RankNorm)
ds_Glucose$Glucose_kin <- scale(resid(lmekin(Glucose ~ age + sex + (1|IID), data=ds_Glucose, varlist = kin_model*2)))
ds_Glucose$Glucose_RankNorm_kin <- scale(resid(lmekin(Glucose_RankNorm ~ (1|IID), data=ds_Glucose, varlist = kin_model*2)))

# HDL_cholesterol
ds_HDL_cholesterol <- ds_pheno %>% filter(!is.na(HDL_cholesterol_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, HDL_cholesterol, HDL_cholesterol_RankNorm)
ds_HDL_cholesterol$HDL_cholesterol_kin <- scale(resid(lmekin(HDL_cholesterol ~ age + sex + (1|IID), data=ds_HDL_cholesterol, varlist = kin_model*2)))
ds_HDL_cholesterol$HDL_cholesterol_RankNorm_kin <- scale(resid(lmekin(HDL_cholesterol_RankNorm ~ (1|IID), data=ds_HDL_cholesterol, varlist = kin_model*2)))

# Total_cholesterol
ds_Total_cholesterol <- ds_pheno %>% filter(!is.na(Total_cholesterol_RankNorm)) %>% select(FID_DNAm, IID_DNAm, IID, age, sex, Total_cholesterol, Total_cholesterol_RankNorm)
ds_Total_cholesterol$Total_cholesterol_kin <- scale(resid(lmekin(Total_cholesterol ~ age + sex + (1|IID), data=ds_Total_cholesterol, varlist = kin_model*2)))
ds_Total_cholesterol$Total_cholesterol_RankNorm_kin <- scale(resid(lmekin(Total_cholesterol_RankNorm ~ (1|IID), data=ds_Total_cholesterol, varlist = kin_model*2)))

ds_kin <- ds_pheno %>% select(FID_DNAm, IID_DNAm, IID, age, sex) %>%
  left_join(ds_bmi, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) %>%
  left_join(ds_whr, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) %>%
  left_join(ds_body_fat, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) %>%
  left_join(ds_Glucose, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) %>%
  left_join(ds_HDL_cholesterol, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) %>%
  left_join(ds_Total_cholesterol, by=c("FID_DNAm", "IID_DNAm", "IID", "age", "sex")) 
  
ds_kin %>% write.table(file="/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/pheno_RankNorm_kin.txt", col.names=T, row.names=F, quote=F)
ds_kin <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/pheno_RankNorm_kin.txt", header=T)

```

```{bash OREML_kinship}

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/pheno_RankNorm_kin.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/kinship_adjusted

for trait in bmi_kin bmi_RankNorm_kin whr_kin whr_RankNorm_kin body_fat_kin body_fat_RankNorm_kin Glucose_kin Glucose_RankNorm_kin HDL_cholesterol_kin HDL_cholesterol_RankNorm_kin Total_cholesterol_kin Total_cholesterol_RankNorm_kin 
do

if [ "${trait}" == "bmi_kin" ];                       then  var=8; fi
if [ "${trait}" == "bmi_RankNorm_kin" ];              then  var=9; fi

if [ "${trait}" == "whr_kin" ];                       then  var=12; fi
if [ "${trait}" == "whr_RankNorm_kin" ];              then  var=13; fi

if [ "${trait}" == "body_fat_kin" ];                  then  var=16; fi
if [ "${trait}" == "body_fat_RankNorm_kin" ];         then  var=17; fi

if [ "${trait}" == "Glucose_kin" ];                   then  var=20; fi
if [ "${trait}" == "Glucose_RankNorm_kin" ];          then  var=21; fi

if [ "${trait}" == "HDL_cholesterol_kin" ];           then  var=24; fi
if [ "${trait}" == "HDL_cholesterol_RankNorm_kin" ];  then  var=25; fi

if [ "${trait}" == "Total_cholesterol_kin" ];         then  var=28; fi
if [ "${trait}" == "Total_cholesterol_RankNorm_kin" ]; then  var=29; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.prop

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--pheno temp.${trait}.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/kinship_adjusted/${trait}

done

# Correlations kinship
cd /Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/kinship_adjusted
mkdir bivariate

wd=/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/kinship_adjusted/bivariate

# Bivariate REML

# bmi_RankNorm $8
# whr_RankNorm $12
# body_fat_RankNorm $16
# Glucose_RankNorm $20
# HDL_cholesterol_RankNorm $24
# Total_cholesterol_RankNorm $28

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/kinship_adj/pheno_RankNorm_kin.txt
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp

for trait1 in Glucose_kin HDL_cholesterol_kin 
do
# trait1=Glucose_kin HDL_cholesterol_kin
if [ "${trait1}" == "bmi_kin" ];              then  var1=8; fi
if [ "${trait1}" == "whr_kin" ];              then  var1=12; fi
if [ "${trait1}" == "body_fat_kin" ];         then  var1=16; fi
if [ "${trait1}" == "Glucose_kin" ];          then  var1=20; fi
if [ "${trait1}" == "HDL_cholesterol_kin" ];  then  var1=24; fi
if [ "${trait1}" == "Total_cholesterol_kin" ]; then  var1=28; fi

for trait2 in HDL_cholesterol_kin Total_cholesterol_kin
do


if [ "${trait2}" == "bmi_kin" ];              then  var2=8; fi
if [ "${trait2}" == "whr_kin" ];              then  var2=12; fi
if [ "${trait2}" == "body_fat_kin" ];         then  var2=16; fi
if [ "${trait2}" == "Glucose_kin" ];          then  var2=20; fi
if [ "${trait2}" == "HDL_cholesterol_kin" ];  then  var2=24; fi
if [ "${trait2}" == "Total_cholesterol_kin" ]; then  var2=28; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' ${pheno} > temp.${trait_pair}_kin

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--thread-num 10 \
--pheno temp.${trait_pair}_kin \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/kinship_adjusted/bivariate/${trait_pair}_0

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 1 \
--thread-num 10 \
--pheno temp.${trait_pair}_kin \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/kinship_adjusted/bivariate/${trait_pair}_1

done




cd ${scratch}/DNAm/GREML/traits/${cov}
for trait1 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist
do
for trait2 in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height weight waist hips
do
awk -v "trait1=${trait1}" -v "trait2=${trait2}"  '{print trait1, trait2, $1, $2, $3}' ${trait1}_${trait2}.hsq >> trait_correlation.hsq
done
done



scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/GREML/traits/covar_batch_eversmoke_packyears/trait_correlation.hsq /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GREML_traits/covar_batch_eversmoke_packyears/


```


## Adjusting for DNAm PCs

```{r DNAm PCs}

### PC

#The most variable normalized probes (m=20,000) were extracted, decomposed into principal components, and each component regressed against slide, chip column, chip row, and sex to test for batch effects. The association detection p value threshold was set to 0.01.

wd=/Local_Scratch/Alesha/DNAm/ORM
orm=mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears

# Calculate 20PCs from ORM 
osca_Linux --orm ${wd}/${orm} --pca 20 --out ${wd}/PC/${orm}-20PCS --thread-num 20

# scp -P 8017 v1ahatt4@localhost:/Local_Scratch/Alesha/DNAm/ORM/PC/* /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/PC
# library(tidyverse)
# # Regress PC against covariates
# path="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/PC"
# path="/Local_Scratch/Alesha/DNAm/ORM/PC"
# 
# eigenval <- read.table(paste0(path,"/mvals-norm20k-18413-831733_sd0.02_unrelated_covar-10PCS.eigenval"))
# eigenvec <- read.table(paste0(path,"/mvals-norm20k-18413-831733_sd0.02_unrelated_covar-10PCS.eigenvec"))
# 
# eigenval <- eigenval %>% mutate(total = sum(V1),
#                                 prop = V1/total * 100) %>% head(n=10)
# 
# samplesheet <- read.table("/Local_Scratch/Alesha/DNAm/samplesheet/GS_20K_samplesheet_unrelated.txt", header=T)
# 
# PC <- eigenvec %>% left_join(samplesheet, by=c("V1"="Sample_Sentrix_ID"))
# 
# PC1_lm <- lm(V3 ~  Sentrix_ID  + age + sex + Set + Batch + CD8T + CD4T + NK + Bcell + Mono + Gran , data=PC)
```

```{bash OREML adj DNAm PCs}

pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha

cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj

# Put headers on e10PCS.eigenvec
echo "FID" "IID" "PC1" "PC2" "PC3" "PC4" "PC5" "PC6" "PC7" "PC8" "PC9" "PC10" "PC11" "PC12" "PC13" "PC14" "PC15" "PC16" "PC17" "PC18" "PC19" "PC20" > ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/20PCS.eigenvec
more ${scratch}/DNAm/ORM/PC/${filename}-20PCS.eigenvec >> ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/20PCS.eigenvec

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}_DNAmPC.prop

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/20PCS.eigenvec \
--pheno temp.${trait}_DNAmPC.prop \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/${trait}

done



# Correlations DNAm PC
pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/bivariate

# for trait1 in bmi whr body_fat Glucose HDL_cholesterol
# for trait1 in bmi whr body_fat Glucose HDL_cholesterol
# do

trait1=HDL_cholesterol

if [ "${trait1}" == "bmi" ];              then  var1=3; fi
if [ "${trait1}" == "whr" ];              then  var1=4; fi
if [ "${trait1}" == "body_fat" ];         then  var1=5; fi
if [ "${trait1}" == "Glucose" ];          then  var1=6; fi
if [ "${trait1}" == "HDL_cholesterol" ];  then  var1=7; fi
if [ "${trait1}" == "Total_cholesterol" ]; then  var1=8; fi


for trait2 in  Total_cholesterol

do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi
if [ "${trait2}" == "height" ]; then  var2=9; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt > temp.${trait_pair}_DNAmPC

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--thread-num 10 \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/20PCS.eigenvec \
--pheno temp.${trait_pair}_DNAmPC \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/bivariate/${trait_pair}_0

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 1 \
--thread-num 10 \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/20PCS.eigenvec \
--pheno temp.${trait_pair}_DNAmPC \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/DNAm_PCs_adj/bivariate/${trait_pair}_1
done



```


## Adjusting for SNP PCs

```{r SNP PCs}
wd=/Local_Scratch/Alesha/DNAm/ORM

# Calculate 20PCs from SNPs
gcta64  --grm /Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3 --pca 20 --threads 50 --out ${wd}/PC/SNP_PC/GS_GWAS_DNAm_0.05_hm3-20PCS

# Put headers on e10PCS.eigenvec
# Match SNP PC ID's to DNAm IDs
library(tidyverse)
eigenvec <- read.table("/Local_Scratch/Alesha/DNAm/ORM/PC/SNP_PC/GS_GWAS_DNAm_0.05_hm3-20PCS.eigenvec", header=F)
data <- readRDS("/Cluster_Filespace/Marioni_Group/GS/GS_methylation/GS20k/GS20k_Targets.rds")

colnames(eigenvec) <- c("FID_SNP", "IID_SNP", "PC1", "PC2", "PC3", "PC4", "PC5", "PC6", "PC7", "PC8", "PC9", "PC10", "PC11", "PC12", "PC13", "PC14", "PC15", "PC16", "PC17", "PC18", "PC19", "PC20")
eigenvec_clean <- eigenvec %>% left_join(data %>% select(Sample_Sentrix_ID, Sample_Name),
                                         by=c("IID_SNP"="Sample_Sentrix_ID")) %>%
                               mutate(FID=IID_SNP, IID=IID_SNP) %>% 
                               select(-IID_SNP, -FID_SNP, -IID_SNP, -Sample_Name) 

eigenvec_clean[, c(21, 22, 1:20)] %>% write.table(file="/Local_Scratch/Alesha/DNAm/GREML/prop_var_explained/covar_batch_eversmoke_packyears/SNP_PCs_adj/20PCS_hm3.eigenvec", row.names=F, col.names=T, quote=F)


# Prop Var adjusting for SNP PCs
pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj

for trait in bmi whr body_fat Glucose HDL_cholesterol Total_cholesterol height 
do

if [ "${trait}" == "bmi" ];              then  var=3; fi
if [ "${trait}" == "whr" ];              then  var=4; fi
if [ "${trait}" == "body_fat" ];         then  var=5; fi
if [ "${trait}" == "Glucose" ];          then  var=6; fi
if [ "${trait}" == "HDL_cholesterol" ];  then  var=7; fi
if [ "${trait}" == "Total_cholesterol" ]; then  var=8; fi
if [ "${trait}" == "height" ]; then  var=9; fi

awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}_SNP_PC

osca_Linux --reml \
--orm ${scratch}/DNAm/ORM/${filename} \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/20PCS_hm3.eigenvec \
--pheno temp.${trait}_SNP_PC \
--thread-num 40 \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/${trait}

done




# Correlations SNP PC
pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
scratch=/Local_Scratch/Alesha
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
mkdir ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/bivariate

# for trait1 in bmi whr body_fat Glucose HDL_cholesterol
# for trait1 in bmi whr body_fat Glucose HDL_cholesterol
# do

trait1=HDL_cholesterol

if [ "${trait1}" == "bmi" ];              then  var1=3; fi
if [ "${trait1}" == "whr" ];              then  var1=4; fi
if [ "${trait1}" == "body_fat" ];         then  var1=5; fi
if [ "${trait1}" == "Glucose" ];          then  var1=6; fi
if [ "${trait1}" == "HDL_cholesterol" ];  then  var1=7; fi
if [ "${trait1}" == "Total_cholesterol" ]; then  var1=8; fi


for trait2 in   Total_cholesterol

do

if [ "${trait2}" == "bmi" ];              then  var2=3; fi
if [ "${trait2}" == "whr" ];              then  var2=4; fi
if [ "${trait2}" == "body_fat" ];         then  var2=5; fi
if [ "${trait2}" == "Glucose" ];          then  var2=6; fi
if [ "${trait2}" == "HDL_cholesterol" ];  then  var2=7; fi
if [ "${trait2}" == "Total_cholesterol" ]; then  var2=8; fi
if [ "${trait2}" == "height" ]; then  var2=9; fi


trait_pair=${trait1}_${trait2}
awk -v "col1=${var1}" -v "col2=${var2}" '{print $1, $2, $col1, $col2}' /Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt > temp.${trait_pair}_SNPPC

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 0 \
--thread-num 10 \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/20PCS.eigenvec \
--pheno temp.${trait_pair}_SNPPC \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/bivariate/${trait_pair}_0

gcta64 --reml-bivar 1 2 \
--grm ${scratch}/DNAm/ORM/GRM/${filename} \
--reml-bivar-lrt-rg 1 \
--thread-num 10 \
--qcovar ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/20PCS.eigenvec \
--pheno temp.${trait_pair}_SNPPC \
--out ${scratch}/DNAm/GREML/prop_var_explained/${cov}/SNP_PCs_adj/bivariate/${trait_pair}_1
done


```





# Literature

<https://nealelab.github.io/UKBB_ldsc/h2_browser.html>

## Heritability

Heritability results are obtained from <https://nealelab.github.io/UKBB_ldsc/downloads.html> ukb31063_h2_all.02Oct2019.tsv.gz 73.7 MB LDSR results for all of the UKB Neale Lab GWAS, including partitioning, sex-stratified results, and alternative versions of phenotypes (e.g. biomarker dilution, non-normalized)

```{r, eval=TRUE, results='asis'}
h2_UKB <- data.table::fread("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/Literature/ukb31063_h2_all.02Oct2019_subset.txt", header=T)
h2_UKB <- h2_UKB %>% select(Phenotype=description, sex, h2 = h2_observed, `h2 (se)`=h2_observed_se, n) %>%
  mutate(order = case_when(Phenotype=="Body mass index (BMI)" ~ 1,
                           Phenotype=="Waist circumference" ~ 2,
                           Phenotype=="Hip circumference" ~ 3,
                           Phenotype=="Body fat percentage" ~ 4,
                           Phenotype=="Glucose (mmol/L)" ~ 5,
                           Phenotype=="HDL cholesterol (mmol/L)" ~ 6,
                           Phenotype=="Cholesterol (mmol/L)" ~ 7,
                           Phenotype=="Standing height" ~ 8)) %>% 
  arrange(order) %>% 
  select(-order) %>% 
  mutate(across(where(is.numeric), round, 3))


kable(h2_UKB %>% filter(sex=="both_sexes") %>% select(-sex), caption="Heritability of phenotypes from UKB") 

kable(h2_UKB %>% filter(sex=="female") %>% select(-sex), caption="Heritability of phenotypes in Females from UKB")         

kable(h2_UKB %>% filter(sex=="male") %>% select(-sex), caption="Heritability of phenotypes in Males from UKB")  
```

## Correlations

```{r}
grep -E '23099_irnt|21001_irnt|30690_irnt|30740_irnt|30760_irnt|49_irnt|50_irnt|48_irnt' correlation_female.r2 > correlation_female_r2_subset.txt
grep -E '23099_irnt|21001_irnt|30690_irnt|30740_irnt|30760_irnt|49_irnt|50_irnt|48_irnt' correlation_male.r2 > correlation_male_r2_subset.txt
grep -E '23099_irnt|21001_irnt|30690_irnt|30740_irnt|30760_irnt|49_irnt|50_irnt|48_irnt' geno_correlation.r2 > correlation_all_r2_subset.txt

cor_female <- data.table::fread("/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/Literature/rg_and_pheno_correlations/correlation_female_r2_subset.txt")
colnames(cor_female) <- c("p1" , "p2", "rg",  "se", "z",  "p", "h2_obs", "h2_obs_se",  "h2_int", "h2_int_se", "gcov_int", "gcov_int_se")
cor_female %>% head

cor_female <- cor_female %>% mutate(pheno_1 = case_when(grepl("/23099_irnt.", p1) ~ "Body fat percentage",
                                          grepl("/21001_irnt.", p1) ~ "Body mass index (BMI)",
                                          grepl("/30690_irnt.", p1) ~ "Cholesterol (mmol/L)",
                                          grepl("/30740_irnt.", p1) ~ "Glucose (mmol/L)",
                                          grepl("/30760_irnt.", p1) ~ "HDL cholesterol (mmol/L)",
                                          grepl("/49_irnt.", p1) ~ "Hip circumference",
                                          grepl("/50_irnt.", p1) ~ "Standing height",
                                          grepl("/48_irnt.", p1) ~ "Waist circumference"),
                      pheno_2 = case_when(grepl("/23099_irnt.", p2) ~ "Body fat percentage",
                                          grepl("/21001_irnt.", p2) ~ "Body mass index (BMI)",
                                          grepl("/30690_irnt.", p2) ~ "Cholesterol (mmol/L)",
                                          grepl("/30740_irnt.", p2) ~ "Glucose (mmol/L)",
                                          grepl("/30760_irnt.", p2) ~ "HDL cholesterol (mmol/L)",
                                          grepl("/49_irnt.", p2) ~ "Hip circumference",
                                          grepl("/50_irnt.", p2) ~ "Standing height",
                                          grepl("/48_irnt.", p2) ~ "Waist circumference")) %>%
                mutate(order1 = case_when(pheno_1=="Body mass index (BMI)" ~ 1,
                                       pheno_1=="Waist circumference" ~ 2,
                                       pheno_1=="Hip circumference" ~ 3,
                                       pheno_1=="Body fat percentage" ~ 4,
                                       pheno_1=="Glucose (mmol/L)" ~ 5,
                                       pheno_1=="HDL cholesterol (mmol/L)" ~ 6,
                                       pheno_1=="Cholesterol (mmol/L)" ~ 7,
                                       pheno_1=="Standing height" ~ 8),
                    order2 = case_when(pheno_2=="Body mass index (BMI)" ~ 1,
                                       pheno_2=="Waist circumference" ~ 2,
                                       pheno_2=="Hip circumference" ~ 3,
                                       pheno_2=="Body fat percentage" ~ 4,
                                       pheno_2=="Glucose (mmol/L)" ~ 5,
                                       pheno_2=="HDL cholesterol (mmol/L)" ~ 6,
                                       pheno_2=="Cholesterol (mmol/L)" ~ 7,
                                       pheno_2=="Standing height" ~ 8),
                    order= order1 + order2/10)
        
cor_female %>% filter(!is.na(pheno_1) & !is.na(pheno_2)) %>% arrange(order) %>%
  select(pheno_1, pheno_2, rg, se, z, p, h2_obs, h2_obs_se, gcov_int, gcov_int_se)
     
```

BMI = /psych/genetics_data/dpalmer/UKbb/ldsc-additive-export/sumstats-files-female/23099_irnt.ldsc.imputed_v3.female.tsv.bgz

## Genetic correlation between traits

Bulik-Sullivan, B., Finucane, H., Anttila, V. *et al.* An atlas of genetic correlations across human diseases and traits. *Nat Genet* **47,** 1236--1241 (2015). <https://doi.org/10.1038/ng.3406>

| Trait                 | BMI            | **Waist-Hip Ratio** | **Fasting Glucose** | HDL           |
|---------------|---------------|---------------|---------------|---------------|
| **Waist-Hip Ratio**   | 0.579 (0.025)  |                     |                     |               |
| **Fasting Glucose**   | 0.312 (0.053)  | 0.209 (0.062)       |                     |               |
| **HDL**               | -0.382 (0.039) | -0.525 (0.07623)    | -0.285 (0.072)      |               |
| **Total Cholesterol** | 0.023 (0.032)  | 0.117 (0.049)       | -0.119 (0.077)      | 0.179 (0.092) |

Fasting glucose != Glucose

<br>

**Heritability**

Zheng J et al. LD Hub: a centralized database and web interface to perform LD score regression that maximizes the potential of summary level GWAS data for SNP heritability and genetic correlation analysis. Bioinformatics. 2017 Jan 15;33(2):272-279. doi: 10.1093/bioinformatics/btw613. Epub 2016 Sep 22. PMID: 27663502; PMCID: PMC5542030.\

| Trait                        | Consortium, first author, database | Sample size | H2     | SE_H2  | Download_link                                                                             |
|----------|----------|----------|----------|----------|-------------------------|
| Body mass index              | EGG                                | 123912      | 0.1855 | 0.0089 | <http://www.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files> |
| Waist-to-hip ratio           | GIANT                              | 212244      | 0.1086 | 0.0059 | <http://www.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files> |
| Glucose                      | MAGNETIC                           | 24679       | 0.0861 | 0.0189 | <http://www.computationalmedicine.fi/data#NMR_GWAS>                                       |
| 2hr glucose adjusted for BMI | MAGIC                              | 15234       | 0.1017 | 0.0344 | <http://www.magicinvestigators.org/downloads/>                                            |
| HDL cholesterol              | GLGC                               | 100184      | 0.1572 | 0.019  | <http://csg.sph.umich.edu//abecasis/public/lipids2013/>                                   |
| Total Cholesterol            | GLGC                               | 99900       | 0.1505 | 0.0168 | <http://csg.sph.umich.edu//abecasis/public/lipids2013/>                                   |

<br>

BMI = 27% (s.e. = 2.5%) (Yang 2015) <br>

# Power calculations

## Sample Sizes

```{r}
library(tidyverse)
ds_pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/bodyfat_covariates_agesex_ranknorm_outliersremove.txt", header=T)

colSums(!is.na(ds_pheno %>% filter(!is.na(body_fat_RankNorm))))


```

Univariate

| Trait             | Total Sample Size | Males | Females | Wave1 | Wave3 | Wave4 |
|-------------------|-------------------|-------|---------|-------|-------|-------|
| BMI               | 7687              | 3358  | 4329    | 2022  | 4258  | 1407  |
| WHR               | 7635              | 3322  | 4313    | 2005  | 4227  | 1403  |
| Body fat %        | 7555              | 3304  | 4251    | 1992  | 4179  | 1384  |
| Glucose           | 7437              | 3216  | 4221    | 1955  | 4119  | 1364  |
| HDL               | 7686              | 3342  | 4344    | 2019  | 4257  | 1410  |
| Total Cholesterol | 7699              | 3348  | 4351    | 2023  | 4261  | 1415  |

Bivariate

| Trait                  | BMI  | WHR  | Body fat % | Glucose | HDL cholesterol |
|:-----------------------|:-----|:-----|:-----------|:--------|:----------------|
| **Waist-to-hip ratio** | 7594 |      |            |         |                 |
| **Body fat %**         | 7530 | 7475 |            |         |                 |
| **Glucose**            | 7377 | 7320 | 7247       |         |                 |
| **HDL cholesterol**    | 7618 | 7563 | 7486       | 7405    |                 |
| **Total cholesterol**  | 7630 | 7576 | 7499       | 7413    | 7685            |

## Power

**Reference:** [Visscher et al. (2014) Statistical power to detect genetic (co)variance of complex traits using SNP data in unrelated samples. PLoS Genetics, 10(4): e1004269.](http://redirect.viglink.com/?key=71fe2139a887ad501313cd8cce3053c5&subId=5425874&u=http%3A//journals.plos.org/plosgenetics/article%3Fid%3D10.1371/journal.pgen.1004269)

Power is the the probability of detecting h2\>0 for the given the user-specified type I error rate and the SNP-heritability assumed in the population. <br>

Statistical power is calculated from the population value of the parameter and its sampling variance. If the parameter is θ, where θ is either the proportion of phenotypic variance captured by SNPs in the univariate case or the genetic correlation in the bivariate case, then θ\^2/var(θ) is asymptotically distributed as a non-central χ2 with 1 degree of freedom and non-centrality parameter (NCP) of θ\^2/var(θ) . Give nλ and the type-I error rate of α, statistical power is the probability that a non-central χ2 variable is larger than the central χ2 threshold that is determined by α. <br>

We perform power calculations for both GRM and ORM. Note: var(θ) is inversely proportion to the variance of off diagonal elements of GRM/ORM

<br> I am assuming these power calculations are based on single variance component analysis and incorperating multiple variance compents may affect the power.

```{r GRM function - calculate var off diag}
library(tidyverse)
# GRM function
grm="/Local_Scratch/Alesha/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated"
orm="/Local_Scratch/Alesha/DNAm/ORM/GRM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar"

prefix_grm=paste0(grm, sep="")
prefix_orm=paste0(orm, sep="")
ReadGRMBin=function(prefix, AllN=F, size=4){
  sum_i=function(i){
    return(sum(1:i))
  }
  BinFileName=paste(prefix,".grm.bin",sep="")
  NFileName=paste(prefix,".grm.N.bin",sep="")
  IDFileName=paste(prefix,".grm.id",sep="")
  id = read.table(IDFileName)
  n=dim(id)[1]
  BinFile=file(BinFileName, "rb");
  grm=readBin(BinFile, n=n*(n+1)/2, what=numeric(0), size=size)
  NFile=file(NFileName, "rb");
  if(AllN==T){
    N=readBin(NFile, n=n*(n+1)/2, what=numeric(0), size=size)
  }
  else N=readBin(NFile, n=1, what=numeric(0), size=size)
  i=sapply(1:n, sum_i)
  return(list(diag=grm[i], off=grm[-i], id=id, N=N))
}

GRM = ReadGRMBin(prefix_grm)
ORM = ReadGRMBin(prefix_orm)

offdiag_GRM <- GRM$off %>% as.data.frame
offdiag_GRM %>% as.vector %>% var()
offdiag_ORM <- ORM$off %>% as.data.frame
offdiag_ORM %>% as.vector %>% var()

# GRM = 1.842512e-05
# ORM = 0.0132483

```

```{r, eval=TRUE}
# Functions used in the functions above
var_vg_func <- function(n, var_pi=2e-5){
	return(2/(n^2*var_pi))
}

var_rg_func <- function(N1, N2, hsq1, hsq2, rg, rp, overlap=TRUE, var_pi=2e-5){
	if(overlap==T) var_rg=((1-rg*rp)^2+(rg-rp)^2)/(hsq1*hsq2*N1^2*var_pi)
	if(overlap==F) var_rg=(rg^2*(N1^2*hsq1^2+N2^2*hsq2^2)+2*hsq1*hsq2*N1*N2)/(2*hsq1^2*hsq2^2*N1^2*N2^2*var_pi)
	return(var_rg)
}

power_func <- function(ncp, alpha){
	pchisq(qchisq(alpha, df=1,lower.tail=F), ncp=ncp, df=1, lower.tail=F)
}

```

<br>

Power for univatiate analysis for varying sample sizes and proportion of variance captured. Note that the variance of the off-diagonal elements of the GRM was 1.842512e-05 and ORM was 0.0132483.

<br>

```{r Univaraite Power, eval=TRUE, out.width = "100%"}
# Function for a quantitative trait
# n = sample size
# hsq = variance explained by all SNPs
# alpha = significance level
# var_pi = variance of the off-diagonal elements of the GRM
# The output are: se (standard error), ncp (non-centrality parameter) and power
calcUniQt <- function(
	n     =1000, 
	hsq   =0.5, 
	alpha =0.05,
  var_pi=2e-5){
    	l <- list()
    	var_vg <- var_vg_func(n, var_pi)
    	l$se <- sqrt(var_vg)
    	l$ncp <- hsq^2/var_vg;
    	l$power <- power_func(l$ncp, alpha)
    	l$samplesize <- n
    	l$hsq <- hsq
    	return(l %>% as.data.frame)
}

# calcUniQt(3000, 0.3, 0.05, 1.842512e-05) # GRM
# calcUniQt(3000, 0.3, 0.05, 0.0132483) # ORM

# GRM_power <- lapply(2000:10000, function(samplesize) {
#   calcUniQt(samplesize,  hsq=0.5, var_pi=1.842512e-05) # GRM
# })
# GRM_power_vector <- do.call(rbind, GRM_power) 
# ORM_power <- lapply(2000:10000, function(samplesize) {
#   calcUniQt(samplesize,  hsq=0.5, var_pi=0.0132483) # ORM
# })
# ORM_power_vector <- do.call(rbind, ORM_power) 


GRM_power <- Map(function(samplesize, hsq) {
  calcUniQt(samplesize,  hsq=hsq, var_pi=1.842512e-05)}, 
    samplesize=rep(seq(from=2000, to=10000, by=10), 5), hsq=rep(c(0.1, 0.2, 0.3, 0.4, 0.5), each=801))
GRM_power_vector <- do.call(rbind, GRM_power)  

ORM_power <- Map(function(samplesize, hsq) {
  calcUniQt(samplesize,  hsq=hsq, var_pi=0.0132483)}, 
    samplesize=rep(seq(from=2000, to=10000, by=10), 5), hsq=rep(c(0.1, 0.2, 0.3, 0.4, 0.5), each=801))
ORM_power_vector <- do.call(rbind, ORM_power)  


#Combine and plot
power_vector <- rbind(
  GRM_power_vector %>% as.data.frame %>% mutate(data="GRM"),
  ORM_power_vector %>% as.data.frame %>% mutate(data="ORM")) %>%
  mutate(samplesize = as.numeric(samplesize),
         power = as.numeric(power),
         se = as.numeric(se),
         ncp = as.numeric(ncp),
         lower = power - se,
         upper = power + se)


ggplot(data=power_vector, aes(x=samplesize, y=power, colour=data)) + 
  geom_point(size=0.5) + 
  geom_ribbon(aes(ymin=lower, ymax=upper), linetype=2, alpha=0.1) +
  facet_grid( ~ hsq) +
  
    labs(title = "Power to detect variance components for univariate analysis",  x = "Sample size", y = "Power", color = "Variance component") +
   theme(legend.position="bottom") +
   theme(axis.text = element_text(size = 7)) +
   geom_vline(xintercept=7500, alpha=0.2) +
  scale_y_continuous(limits = c(-0.7, 1.2))
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/univariate_power.png", width = 8, height = 6, dpi = 150, units = "in", device='png')

  


```

<br> <br>

We performed the same power calculations for the bivariate analysis (genetic correlation). Here we set the sample sizes for the two traits to be equal and well as the proportion of variance captured. We consider the case where rg and rp are 0.5 with complete sample overlap.

```{r Bivariate power, eval=TRUE, out.width = "100%"}
# Function for bivariate analysis of two quantitative traits
# rg = genetic correlation
# rp = phenotypic correlation
# overlap = whether or not the traits are measured on the same samples
calcBiQt <- function(
	n1      = 1000, 
	n2      = 1000, 
	hsq1    = 0.5, 
	hsq2    = 0.5, 
	rg      = 0.5, 
	rp      = 0.5, 
	overlap = FALSE, 
	alpha   = 0.05,
	var_pi=2e-5){
    	var_rg <- var_rg_func(n1, n2, hsq1, hsq2, rg, rp, overlap, var_pi)
    	l <- list()
    	l$se <- sqrt(var_rg)
    	l$ncp <- rg^2/var_rg;
    	l$power <- power_func(l$ncp, alpha)
    	l$samplesize <- n1
    	l$hsq <- hsq1
    	return(l)
}
# calcBiQt(3000,3000, 0.3, 0.3, 0.5, 0.5, TRUE, 0.05, 1.842512e-05) # GRM
# calcBiQt(3000,3000, 0.3, 0.3, 0.5, 0.5, TRUE, 0.05, 0.0132483) # ORM


GRM_power_bivar <- Map(function(samplesize, hsq) {
  calcBiQt(samplesize, samplesize, hsq, hsq, 0.8, hsq, TRUE, 0.05, 1.842512e-05) }, 
    samplesize=rep(seq(from=2000, to=10000, by=10), 5), hsq=rep(c(0.1, 0.2, 0.3, 0.4, 0.5), each=801))
GRM_power_bivar_vector <- do.call(rbind, GRM_power_bivar)  

ORM_power_bivar <- Map(function(samplesize, hsq) {
    calcBiQt(samplesize, samplesize, hsq, hsq, 0.8, hsq, TRUE, 0.05, 0.0132483) }, 
    samplesize=rep(seq(from=2000, to=10000, by=10), 5), hsq=rep(c(0.1, 0.2, 0.3, 0.4, 0.5), each=801))
ORM_power_bivar_vector <- do.call(rbind, ORM_power_bivar)  


#Combine and plot
power_bivar_vector <- rbind(
  GRM_power_bivar_vector %>% as.data.frame %>% mutate(data="GRM"),
  ORM_power_bivar_vector %>% as.data.frame %>% mutate(data="ORM")) %>%
  mutate(samplesize = as.numeric(samplesize),
         power = as.numeric(power),
         hsq = as.numeric(hsq),
         se = as.numeric(se),
         ncp = as.numeric(ncp),
         lower = power - se,
         upper = power + se)


ggplot(data=power_bivar_vector, aes(x=samplesize, y=power, colour=data)) + 
  geom_point(size=0.5) + 
  geom_ribbon(aes(ymin=lower, ymax=upper), linetype=2, alpha=0.1) +
  facet_grid( ~ hsq) +
    labs(title = "",  x = "Sample size", y = "Power", color = "Variance component") +
   theme(legend.position="bottom") +
    theme(axis.text = element_text(size = 7)) +
     geom_vline(xintercept=7500, alpha=0.2) +
  scale_y_continuous(limits = c(-0.7, 1.2))
ggsave(file="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/results/bivariate_power_rg0.1.png", width = 8, height = 6, dpi = 150, units = "in", device='png')



# Standard error (SE): Standard error of the SNP-heritability (h2).
# NCP: Non-centrality paramter of the chi-squared test statistic, which equal to h4/(SE)2.
# Power: The probability of detecting h2>0 for the given the user-specified type I error rate and the SNP-heritability assumed in the population.
```

<br> <br>

How does having multiple variance components effects power? e.g. ORM and GRM fitted simultaneously?

## Investigating ORM

```{r GRM/ORM}
library(tidyverse)
# GRM function
grm="/Local_Scratch/Alesha/DNAm/GREML/traits/GRM_ORM/GS_GWAS_unrelated"
orm="/Local_Scratch/Alesha/DNAm/ORM/GRM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar"
grm_related="/Cluster_Filespace/Marioni_Group/Alesha/unrelated/GS_GWAS"

prefix_grm=paste0(grm, sep="")
prefix_orm=paste0(orm, sep="")
ReadGRMBin=function(prefix, AllN=F, size=4){
  sum_i=function(i){
    return(sum(1:i))
  }
  BinFileName=paste(prefix,".grm.bin",sep="")
  NFileName=paste(prefix,".grm.N.bin",sep="")
  IDFileName=paste(prefix,".grm.id",sep="")
  id = read.table(IDFileName)
  n=dim(id)[1]
  BinFile=file(BinFileName, "rb");
  grm=readBin(BinFile, n=n*(n+1)/2, what=numeric(0), size=size)
  NFile=file(NFileName, "rb");
  if(AllN==T){
    N=readBin(NFile, n=n*(n+1)/2, what=numeric(0), size=size)
  }
  else N=readBin(NFile, n=1, what=numeric(0), size=size)
  i=sapply(1:n, sum_i)
  return(list(diag=grm[i], off=grm[-i], id=id, N=N))
}

GRM = ReadGRMBin(prefix_grm)
GRM_related = ReadGRMBin(grm_related)
ORM = ReadGRMBin(prefix_orm)

save(GRM, file="/Cluster_Filespace/Marioni_Group/Alesha/GRM.Rdata")
# save(GRM_related, file="/Cluster_Filespace/Marioni_Group/Alesha/GRM_related.Rdata")
save(ORM, file="/Cluster_Filespace/Marioni_Group/Alesha/ORM.Rdata")
# save(ORM_related, file="/Cluster_Filespace/Marioni_Group/Alesha/ORM_related.Rdata")

scp -p 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/*.Rdata /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM/
scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/*.Rdata /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM

```

```{r plotting ORM}

library(tidyverse)
path="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM"
# load(paste0(path,"/GRM_related.Rdata"))

load(paste0(path,"/GRM.Rdata"))
load(paste0(path,"/ORM.Rdata"))


offdiag_GRM <- GRM$off %>% as.data.frame
offdiag_GRM %>% as.vector %>% var()
offdiag_ORM <- ORM$off %>% as.data.frame
offdiag_ORM %>% as.vector %>% var()

# GRM = 1.842512e-05
# GRM_related = 4.447155e-05
# ORM = 0.0132483

# plot offdiagonals of GRM & ORM for comparison

offdiag_GRM <- offdiag_GRM %>% mutate(data="GRM")
offdiag_ORM <- offdiag_ORM %>% mutate(data="ORM")

offdiag <- rbind(offdiag_GRM, offdiag_ORM)
colnames(offdiag) <- c("offdiag", "data")

p1 <- ggplot(offdiag, aes(x=offdiag)) +
  geom_histogram() +
  facet_grid(~data,  scales = "free") +
  scale_x_continuous(limits=c(-0.1, 0.1))+
  labs(x="Off-diagonal elements of variance component") 

offdiag %>% filter(data=="ORM" & offdiag > 5)
offdiag %>% filter(data=="ORM" & offdiag > 5)
#      offdiag data
# 1   7.288937  ORM
# 2   8.076982  ORM
# 3   5.474079  ORM
# 4   8.708009  ORM
# 5   5.009297  ORM
# 6   5.472075  ORM
# 7   9.358624  ORM
# 8   5.302905  ORM
# 9   5.447984  ORM
# 10  5.560289  ORM
# 11 11.168067  ORM
# 12  6.202472  ORM
# 13  7.356836  ORM

# Min.   :-6.175478   
#  1st Qu.:-0.043258   
#  Median :-0.000101  
#  Mean   :-0.000129                     
#  3rd Qu.: 0.042362                     
#  Max.   :11.168067   

p1 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/GRM_ORM_offdiag_axis_cut.png", width = 9, height = 6, dpi = 150, units = "in", device='png')

# scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/GRM_ORM_offdiag.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/pheno_QC


# Diagonal Elements

diag_GRM <- GRM$diag %>% as.data.frame
diag_GRM %>% as.vector %>% var()
diag_ORM <- ORM$diag %>% as.data.frame
diag_ORM %>% as.vector %>% var()

# GRM = 9.303124e-05
# ORM = 0.3050898

# plot offdiagonals of GRM & ORM for comparison
diag_GRM <- diag_GRM %>% mutate(data="GRM")
diag_ORM <- diag_ORM %>% mutate(data="ORM")

diag_ORM %>% summary
 #     .               data          
 # Min.   : 0.5744   Length:7758       
 # 1st Qu.: 0.7850   Class :character  
 # Median : 0.8775   Mode  :character  
 # Mean   : 0.9999                     
 # 3rd Qu.: 1.0493                     
 # Max.   :24.9557          

diag <- rbind(diag_GRM, diag_ORM)
colnames(diag) <- c("diag", "data")

p2 <- ggplot(diag, aes(x=diag)) +
  geom_histogram() +
  facet_grid(~data,  scales = "free") +
  labs(x="Diagonal elements of variance component") 

p2 %>% ggsave(file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/GRM_ORM_diag.png", width = 9, height = 6, dpi = 150, units = "in", device='png')

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/GRM_ORM_diag.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/pheno_QC
```

![Off-diagonal elements of GRM and ORM](data/GRM_ORM/GRM_ORM_offdiag.png)

![Diagonal elements of GRM and ORM](data/GRM_ORM/GRM_ORM_diag.png)

<br> <br>

```{r Correlation between offdiagonal elements of GRM and ORM}

library(tidyverse)
library(reshape2)

path="/Cluster_Filespace/Marioni_Group/Alesha"
path="/Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM"

load(paste0(path,"/GRM.Rdata"))
load(paste0(path,"/ORM.Rdata"))


grm.matrix = data.frame(matrix(0, ncol=nrow(GRM$id), nrow=nrow(GRM$id)))
colnames(grm.matrix)=GRM$id$V2
rownames(grm.matrix)=GRM$id$V2
grm.matrix[lower.tri(grm.matrix)] <- NA

grm.ds <- melt(as.matrix(grm.matrix))
grm.ds <- grm.ds[complete.cases(grm.ds), ]
grm.ds <- grm.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(GRM = GRM$off) %>% 
  select(-value)


orm.matrix = data.frame(matrix(0, ncol=nrow(ORM$id), nrow=nrow(ORM$id)))
colnames(orm.matrix)=ORM$id$V2
rownames(orm.matrix)=ORM$id$V2
orm.matrix[lower.tri(orm.matrix)] <- NA

orm.ds <- melt(as.matrix(orm.matrix))
orm.ds <- orm.ds[complete.cases(orm.ds), ]
orm.ds <- orm.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(ORM = ORM$off) %>% 
  select(-value)


offdiag <- orm.ds %>% 
  left_join(grm.ds, by=c("Var1", "Var2")) %>% 
  left_join(grm.ds, by=c("Var1"="Var2", "Var2"="Var1"))

offdiag <- offdiag %>% mutate(GRM = case_when(is.na(GRM.x) ~ GRM.y,
                                              is.na(GRM.y) ~ GRM.x,
                                              TRUE ~ as.numeric(NA)))

library(ggpubr)

ggraphs2 <- ggplot(offdiag,aes(x = GRM , y = ORM)) +
              geom_point(size=0.9) +
              scale_y_continuous("MRM", limits=c(-7, 10.5)) +
              stat_cor(aes(label = ..r.label..), method = "pearson", label.x =0.01, label.y = 10.96, size=4) + theme_linedraw()
ggsave(ggraphs2, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/correlation_ORM_GRM2.png", width = 6, height = 6, dpi = 150, units = "in", device='png')

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/correlation_ORM_GRM2.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM/

```

```{r mQTL nomQTL correlation}

library(tidyverse)
# GRM function
path="/Local_Scratch/Alesha/DNAm/ORM"
mQTL="/Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_mQTLs"
nomQTL="/Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears_nomQTLs"

prefix_mQTL=paste0(mQTL, sep="")
prefix_nomQTL=paste0(nomQTL, sep="")

ReadGRMBin=function(prefix, AllN=F, size=4){
  sum_i=function(i){
    return(sum(1:i))
  }
  BinFileName=paste(prefix,".orm.bin",sep="")
  NFileName=paste(prefix,".orm.N.bin",sep="")
  IDFileName=paste(prefix,".orm.id",sep="")
  id = read.table(IDFileName)
  n=dim(id)[1]
  BinFile=file(BinFileName, "rb");
  grm=readBin(BinFile, n=n*(n+1)/2, what=numeric(0), size=size)
  NFile=file(NFileName, "rb");
  if(AllN==T){
    N=readBin(NFile, n=n*(n+1)/2, what=numeric(0), size=size)
  }
  else N=readBin(NFile, n=1, what=numeric(0), size=size)
  i=sapply(1:n, sum_i)
  return(list(diag=grm[i], off=grm[-i], id=id, N=N))
}

ORM_mQTL = ReadGRMBin(prefix_mQTL)
ORM_nomQTL = ReadGRMBin(prefix_nomQTL)

## Correlations between off diagonals
library(reshape2)
mQTL.matrix = data.frame(matrix(0, ncol=nrow(ORM_mQTL$id), nrow=nrow(ORM_mQTL$id)))
colnames(mQTL.matrix)=ORM_mQTL$id$V2
rownames(mQTL.matrix)=ORM_mQTL$id$V2
mQTL.matrix[lower.tri(mQTL.matrix)] <- NA

mQTL.ds <- melt(as.matrix(mQTL.matrix))
mQTL.ds <- mQTL.ds[complete.cases(mQTL.ds), ]
mQTL.ds <- mQTL.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(ORM_mQTL = ORM_mQTL$off) %>% 
  select(-value)


nomQTL.matrix = data.frame(matrix(0, ncol=nrow(ORM_nomQTL$id), nrow=nrow(ORM_nomQTL$id)))
colnames(nomQTL.matrix)=ORM_nomQTL$id$V2
rownames(nomQTL.matrix)=ORM_nomQTL$id$V2
nomQTL.matrix[lower.tri(nomQTL.matrix)] <- NA

nomQTL.ds <- melt(as.matrix(nomQTL.matrix))
nomQTL.ds <- nomQTL.ds[complete.cases(nomQTL.ds), ]
nomQTL.ds <- nomQTL.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(ORM_nomQTL = ORM_nomQTL$off) %>% 
  select(-value)


offdiag <- mQTL.ds %>% 
  left_join(nomQTL.ds, by=c("Var1", "Var2")) %>% 
  left_join(nomQTL.ds, by=c("Var1"="Var2", "Var2"="Var1"))

offdiag <- offdiag %>% mutate(nomQTL = case_when(is.na(ORM_nomQTL.x) ~ ORM_nomQTL.y,
                                              is.na(ORM_nomQTL.y) ~ ORM_nomQTL.x,
                                              TRUE ~ as.numeric(NA)))

library(ggpubr)

ggraphs <- ggplot(offdiag,aes(x = ORM_mQTL , y = nomQTL)) +
              geom_point() +
              scale_y_continuous(limits=c(-7, 11)) +
              stat_cor(aes(label = ..r.label..), method = "pearson", label.x =0.01, label.y = 10.96, size=3)
ggsave(ggraphs, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/correlation_ORM_mQTL_nomQTL.png", width = 9, height = 12, dpi = 150, units = "in", device='png')

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/correlation_ORM_mQTL_nomQTL.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM



```


<br> <br>

<br> <br>

```{r Wave - GRM/ORM distribution by wave and batch}

library(tidyverse)
library(reshape2)

path="/Cluster_Filespace/Marioni_Group/Alesha"

load(paste0(path,"/GRM.Rdata"))
load(paste0(path,"/ORM.Rdata"))

grm.matrix = data.frame(matrix(0, ncol=nrow(GRM$id), nrow=nrow(GRM$id)))
colnames(grm.matrix)=GRM$id$V2
rownames(grm.matrix)=GRM$id$V2
grm.matrix[lower.tri(grm.matrix)] <- NA

grm.ds <- melt(as.matrix(grm.matrix))
grm.ds <- grm.ds[complete.cases(grm.ds), ]
grm.ds <- grm.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(GRM = GRM$off) %>% 
  select(-value)


orm.matrix = data.frame(matrix(0, ncol=nrow(ORM$id), nrow=nrow(ORM$id)))
colnames(orm.matrix)=ORM$id$V2
rownames(orm.matrix)=ORM$id$V2
orm.matrix[lower.tri(orm.matrix)] <- NA

orm.ds <- melt(as.matrix(orm.matrix))
orm.ds <- orm.ds[complete.cases(orm.ds), ]
orm.ds <- orm.ds %>% 
  filter(Var1!=Var2) %>% 
  cbind.data.frame(ORM = ORM$off) %>% 
  select(-value)

offdiag <- orm.ds %>% 
  left_join(grm.ds, by=c("Var1", "Var2")) %>% 
  left_join(grm.ds, by=c("Var1"="Var2", "Var2"="Var1"))

offdiag <- offdiag %>% mutate(GRM = case_when(is.na(GRM.x) ~ GRM.y,
                                              is.na(GRM.y) ~ GRM.x,
                                              TRUE ~ as.numeric(NA)))

ds <- read.table("/Cluster_Filespace/Marioni_Group/Alesha/covariates/oii_cov_unrelated.txt", header=T)
ds_set <- ds %>% select(FID_DNAm, Set)


offdiag_set <- offdiag %>% 
  left_join(ds_set, by=c("Var1"="FID_DNAm")) %>%
  mutate(Set_var1=Set) %>% select(-Set) %>%
  left_join(ds_set, by=c("Var2"="FID_DNAm")) %>%
  mutate(Set_var2=Set) %>% select(-Set) 

GRM_set <- ggplot(offdiag_set, aes(x=GRM)) +
  geom_histogram() +
  facet_grid(Set_var1~Set_var2,  scales = "free") +
  labs(x="Off-diagonal elements of variance component") 

ggsave(GRM_set, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/GRM_wave.png", width = 9, height = 12, dpi = 150, units = "in", device='png')


ORM_set <- ggplot(offdiag_set, aes(x=ORM)) +
  geom_histogram() +
  facet_grid(Set_var1~Set_var2,  scales = "free") +
  labs(x="Off-diagonal elements of variance component") 

ggsave(ORM_set, file="/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/ORM_wave.png", width = 9, height = 12, dpi = 150, units = "in", device='png')

scp -P 8017 v1ahatt4@localhost:/Cluster_Filespace/Marioni_Group/Alesha/covariates/figures/ORM_wave.png /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/GRM_ORM


offdiag_set %>% mutate(grp = case_when(Set_var1==Set_var2 ~ "Within",
                                   TRUE ~ "Between")) %>%
  group_by(grp) %>% 
  summarise(mean=mean(ORM), median=median(ORM),
            min = min(ORM), max=max(ORM),
            var = var(ORM))

  grp          mean     median   min   max    var
  <chr>       <dbl>      <dbl> <dbl> <dbl>  <dbl>
1 Between  1.99e-13  0.0000523 -4.63 11.2  0.0127
2 Within  -3.15e- 4 -0.000331  -6.18  8.08 0.0140

  Set_var1 Set_var2      mean     median   min   max     var
  <chr>    <chr>        <dbl>      <dbl> <dbl> <dbl>   <dbl>
1 wave1    wave1    -5.80e- 4  0.000566  -6.18  8.08 0.0439 
2 wave1    wave3     1.25e-13  0.000239  -3.64  9.36 0.0172 
3 wave1    wave4     6.49e-13  0.0000122 -4.63 11.2  0.0133 
4 wave3    wave3    -2.20e- 4 -0.000360  -2.04  5.30 0.00817
5 wave3    wave4     9.20e-14 -0.000118  -2.74  7.36 0.00616
6 wave4    wave4    -6.37e- 4 -0.00104   -2.30  3.37 0.00580


offdiag_set %>% mutate(grp = case_when(Set_var1==Set_var2 ~ "Within",
                                   TRUE ~ "Between")) %>%
  group_by(grp) %>% 
  summarise(mean=mean(GRM), median=median(GRM),
            min = min(GRM), max=max(GRM),
            var = var(GRM))

  Set_var1 Set_var2       mean     median     min    max       var
  <chr>    <chr>         <dbl>      <dbl>   <dbl>  <dbl>     <dbl>
1 wave1    wave1    -0.0000386 -0.0000838 -0.0196 0.0496 0.0000185
2 wave1    wave3    -0.0000708 -0.000110  -0.0216 0.0495 0.0000184
3 wave1    wave4    -0.0000823 -0.000123  -0.0222 0.0500 0.0000184
4 wave3    wave3    -0.0000474 -0.0000885 -0.0218 0.0497 0.0000184
5 wave3    wave4    -0.0000760 -0.000120  -0.0205 0.0494 0.0000184
6 wave4    wave4    -0.0000979 -0.000149  -0.0201 0.0497 0.0000187

  grp           mean     median     min    max       var
  <chr>        <dbl>      <dbl>   <dbl>  <dbl>     <dbl>
1 Between -0.0000745 -0.000115  -0.0222 0.0500 0.0000184
2 Within  -0.0000501 -0.0000925 -0.0218 0.0497 0.0000185
```

![Distribution of off-digonal elements of GRM by Wave](data/GRM_ORM/GRM_wave.png)

<br>

![Distribution of off-diagonal elements of ORM by Wave](data/GRM_ORM/ORM_wave.png)

| grp         | mean      | median    | min   | max  | var     |
|-------------|-----------|-----------|-------|------|---------|
| Between     | 1.99e-13  | 0.0000523 | -4.63 | 11.2 | 0.0127  |
| Within      | -3.15e- 4 | -0.000331 | -6.18 | 8.08 | 0.0140  |
|             |           |           |       |      |         |
| wave1 wave1 | -5.80e- 4 | 0.000566  | -6.18 | 8.08 | 0.0439  |
| wave1 wave3 | 1.25e-13  | 0.000239  | -3.64 | 9.36 | 0.0172  |
| wave1 wave4 | 6.49e-13  | 0.0000122 | -4.63 | 11.2 | 0.0133  |
| wave3 wave3 | -2.20e- 4 | -0.000360 | -2.04 | 5.30 | 0.00817 |
| wave3 wave4 | 9.20e-14  | -0.000118 | -2.74 | 7.36 | 0.00616 |
| wave4 wave4 | -6.37e- 4 | -0.00104  | -2.30 | 3.37 | 0.00580 |

<br>

<br>

# Ella simulations
```{r GS}
scratch=/Local_Scratch/Alesha
mkdir ${scratch}/Ella_simulations
# 1. BMI GWAS using fast-linear and MOA
scratch=/Local_Scratch/Alesha
pheno=/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt
cov=covar_batch_eversmoke_packyears
filename=mvals-norm20k-18413-831733_sd0.02_unrelated_${cov}

cd ${scratch}/DNAm/phenotypes/greml.pheno/temp
trait=bmi
var=3
awk -v "col1=${var}" '{print $1, $2, $col1}' ${pheno} > temp.${trait}.EWAS

osca_Linux \
--linear \
--befile ${scratch}/DNAm/OSCA/${filename} \
--pheno temp.${trait}.EWAS \
--fast-linear \
--thread-num 30 \
--out ${scratch}/Ella_simulations/${trait}_fast-linear

osca_Linux \
--befile ${scratch}/DNAm/OSCA/${filename} \
--pheno temp.${trait}.EWAS \
--moa \
--thread-num 30 \
--out ${scratch}/Ella_simulations/${trait}_moa


# 2. Simulating 1000 causal probes - fast-linear, MOA

# How are causal probes selected?
library(dplyr)
probes <- read.table("/Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.opi", header=F) 
probes %>% filter(V1 %in% seq(1, 21, 2)) %>%
  sample_n(size=1000, replace=F) %>% 
  select(V2) %>%
  write.table(file="/Local_Scratch/Alesha/Ella_simulations/random_probes_1000_odd.txt", col.names=F, row.names=F, quote=F)


for h2 in 0.1 0.2 0.3 0.4 0.5
do
osca_Linux \
--simu-qt \
--simu-rsq ${h2} \
--befile ${scratch}/DNAm/OSCA/${filename} \
--simu-causal-loci ${scratch}/Ella_simulations/random_probes_1000_odd.txt \
--out ${scratch}/Ella_simulations/pheno_1000probes_h${h2}

osca_Linux \
--befile ${scratch}/DNAm/OSCA/${filename} \
--pheno ${scratch}/Ella_simulations/pheno_1000probes_h${h2}.phen  \
--linear \
--fast-linear \
--thread-num 30 \
--out ${scratch}/Ella_simulations/pheno_1000probes_h${h2}_fast-linear

osca_Linux \
--befile ${scratch}/DNAm/OSCA/${filename} \
--pheno ${scratch}/Ella_simulations/pheno_1000probes_h${h2}.phen  \
--moa \
--thread-num 30 \
--out ${scratch}/Ella_simulations/pheno_1000probes_h${h2}_moa

done

# Compare to simulated causal
awk '$8 < 1e-8 {print}' ${scratch}/Ella_simulations/pheno_1000probes_h${h2}_fast-linear.linear  > ${scratch}/Ella_simulations/pheno_1000probes_h${h2}_fast-linear_sig.linear 


library(dplyr)
causal_probes <- read.table("/Local_Scratch/Alesha/Ella_simulations/random_probes_1000_odd.txt", header=F) 
colnames(causal_probes) <- c("probe")
h2_0.1 <- data.table::fread("/Local_Scratch/Alesha/Ella_simulations/pheno_1000probes_h0.1_fast-linear.linear", header=T)
h2_0.1 %>% filter(Probe %in% causal_probes$probe) %>% summary
h2_0.1 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% dim # 97


# 1. Plot probe effects




# --simu-qt smulates a quantitative trait.
# --simu-rsq specifies the proportion of variance in phenotype explained by the causal probes. The default value is 0.1.
# --simu-causal-loci reads a list of probes as causal probes. If the effect sizes are not specified in the file, they will be generated from a standard normal distribution.

# How are you dealing with covariates?
# If they arent included during simulation does this create some sort of bias/confounding during EWAS?



```

```{r PD}
/home/uqahatt2/wd/PD_EWASdata/PD_controls
/home/uqahatt2/wd/PD_EWASdata/simulations

# 2. Simulating 1000 causal probes - fast-linear, MOA

# How are causal probes selected?
module load R/4.0.0
R
library(dplyr)
probes <- read.table("/home/uqahatt2/wd/PD_EWASdata/PD_controls.opi", header=F) 
probes %>% 
  sample_n(size=1000, replace=F) %>% 
  select(V2) %>%
  write.table(file="/home/uqahatt2/wd/PD_EWASdata/simulations/random_probes_1000_odd.txt", col.names=F, row.names=F, quote=F)


for h2 in 0.1 0.2 0.3 0.4 0.5
do
osca \
--simu-qt \
--simu-rsq ${h2} \
--befile /home/uqahatt2/wd/PD_EWASdata/PD_controls \
--simu-causal-loci /home/uqahatt2/wd/PD_EWASdata/simulations/random_probes_1000_odd.txt \
--out /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}

osca \
--befile /home/uqahatt2/wd/PD_EWASdata/PD_controls \
--pheno /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}.phen  \
--linear \
--fast-linear \
--thread-num 30 \
--out /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}_fast-linear

awk '$8 < 1e-8 {print}' /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}_fast-linear.linear  > /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}_fast-linear_sig.linear 

done

# osca \
# --befile /home/uqahatt2/wd/PD_EWASdata/PD_controls \
# --pheno /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}.phen  \
# --moa \
# --thread-num 30 \
#--out /home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h${h2}_moa



# Compare to simulated causal


library(dplyr)
causal_probes <- read.table("/home/uqahatt2/wd/PD_EWASdata/simulations/random_probes_1000_odd.txt", header=F) 
colnames(causal_probes) <- c("probe")
h2_0.1 <- data.table::fread("/home/uqahatt2/wd/PD_EWASdata/simulations/pheno_1000probes_h0.1_fast-linear.linear", header=T)
h2_0.1 %>% filter(Probe %in% causal_probes$probe) %>% summary
h2_0.1 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% dim # 97
h2_0.1 %>% filter(p < 1e-8) %>% dim # 97



```

```{r LBC}

mkdir ~/wd/DNAm_sims/LBC

library(dplyr)
setwd("~/wd/DNAm_sims")

Targets <- read.table("~/rdm/1_project/methylation_data/LBCW1/qcovar_age.txt", header=F) 

load("~/rdm/1_project/methylation_data/LBCW1/norm.beta.Robj")

# Select normalisation values from norm.beta for these individuals
sub.norm.beta <- norm.beta[,colnames(norm.beta) %in% Targets$V1]

tnorm.beta <- t(sub.norm.beta) %>% as.data.frame
tnorm.beta$IID <- rownames(tnorm.beta)

tnorm.beta <- tnorm.beta %>% select(IID, everything())
dim(tnorm.beta)

# Write covariates
write.table(Targets %>% filter(V1 %in% tnorm.beta$IID) %>% select(V1), file="~/wd/DNAm_sims/LBC/IID.txt", col.names=F, row.names=F, quote=F)

# Write norm.beta
data.table::fwrite(tnorm.beta, row.names=F, quote=F, sep=" ", file="/home/uqahatt2/wd/DNAm_sims/LBC/norm.beta.txt")


# Make BOD file
cd ~/wd/DNAm_sims/LBC
osca --efile norm.beta.txt --thread-num 30 --methylation-beta --make-bod --no-fid --out LBC



# Simulate

scratch=~/wd/DNAm_sims/LBC
filename=LBC

# How are causal probes selected?
library(dplyr)
set.seed(1)
probes <- read.table("/home/uqahatt2/wd/DNAm_sims/LBC/LBC.opi", header=F) 
probes %>% 
  sample_n(size=1000, replace=F) %>% 
  select(V2) %>%
  write.table(file="/home/uqahatt2/wd/DNAm_sims/LBC/random_probes_1000_odd.txt", col.names=F, row.names=F, quote=F)


for h2 in 0.1 0.2 0.3 0.4 0.5
do

osca \
--simu-qt \
--simu-rsq ${h2} \
--befile ${scratch}/${filename} \
--simu-causal-loci ${scratch}/random_probes_1000_odd.txt \
--out ${scratch}/pheno_1000probes_h${h2}

osca \
--befile ${scratch}/${filename} \
--pheno ${scratch}/pheno_1000probes_h${h2}.phen  \
--linear \
--fast-linear \
--thread-num 30 \
--out ${scratch}/pheno_1000probes_h${h2}_fast-linear

# osca \
# --befile ${scratch}/${filename} \
# --pheno ${scratch}/pheno_1000probes_h${h2}.phen  \
# --moa \
# --thread-num 30 \
# --out ${scratch}/pheno_1000probes_h${h2}_moa

done

# Compare to simulated causal
awk '$8 < 1e-8 {print}' ${scratch}/pheno_1000probes_h${h2}_fast-linear.linear  > ${scratch}/pheno_1000probes_h${h2}_fast-linear_sig.linear 


library(dplyr)
causal_probes <- read.table("/home/uqahatt2/wd/DNAm_sims/LBC/random_probes_1000_odd.txt", header=F) 
colnames(causal_probes) <- c("probe")
h2_0.1 <- data.table::fread("/home/uqahatt2/wd/DNAm_sims/LBC/pheno_1000probes_h0.1_fast-linear.linear", header=T)
h2_0.2 <- data.table::fread("/home/uqahatt2/wd/DNAm_sims/LBC/pheno_1000probes_h0.2_fast-linear.linear", header=T)
h2_0.3 <- data.table::fread("/home/uqahatt2/wd/DNAm_sims/LBC/pheno_1000probes_h0.3_fast-linear.linear", header=T)
h2_0.4 <- data.table::fread("/home/uqahatt2/wd/DNAm_sims/LBC/pheno_1000probes_h0.4_fast-linear.linear", header=T)
h2_0.5 <- data.table::fread("/home/uqahatt2/wd/DNAm_sims/LBC/pheno_1000probes_h0.5_fast-linear.linear", header=T)

h2_0.1 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow/h2_0.1 %>% filter(p < 1e-8) %>% nrow * 100 # 0.2135148%
h2_0.1 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow
h2_0.1 %>% filter(p < 1e-8) %>% nrow  # 0.2135148%

h2_0.2 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow/h2_0.2 %>% filter(p < 1e-8) %>% nrow * 100 # 0.2853067%
h2_0.2 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow # 273
h2_0.2 %>% filter(p < 1e-8) %>% nrow  # 127860

h2_0.3 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow/h2_0.3 %>% filter(p < 1e-8) %>% nrow * 100 # 0.1896747%
h2_0.3 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow # 87
h2_0.3 %>% filter(p < 1e-8) %>% nrow  # 45868

h2_0.4 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow/h2_0.4 %>% filter(p < 1e-8) %>% nrow * 100 # 0.2515337%
h2_0.4 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow # 41
h2_0.4 %>% filter(p < 1e-8) %>% nrow  # 16300

h2_0.5 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow/h2_0.5 %>% filter(p < 1e-8) %>% nrow * 100 # 0.1838193%
h2_0.5 %>% filter(Probe %in% causal_probes$probe & p < 1e-8) %>% nrow # 79
h2_0.5 %>% filter(p < 1e-8) %>% nrow  # 42977


```

## CORE-GREML

```{r}

scp -P 8017 -r /Users/uqahatt2/Documents/2_project/4_GS/1_Correlations/data/CORE-GREML/public_access v1ahatt4@localhost:/Local_Scratch/Alesha/core_greml/

# i. ensure positive definite kernel matrices

/Cluster_Filespace/Marioni_Group/Alesha
/Local_Scratch/Alesha/DNAm
fam=/Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.oii
grm=/Local_Scratch/Alesha/DNAm/ORM/GRM_GS_unrelated/imputed_HRS/GS20K_HRC.r1-1_nomono_I4_cpra_unrelated_hm3.orm.bin
mrm=/Local_Scratch/Alesha/DNAm/ORM/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.orm.bin

# Covert files to different formats
# fam
# 1 1 0 0 -9 -9
# 2 2 0 0 -9 -9

awk '{print $1, $2, $3, $4, -9, -9}' /Local_Scratch/Alesha/DNAm/OSCA/mvals-norm20k-18413-831733_sd0.02_unrelated_covar_batch_eversmoke_packyears.oii > /Local_Scratch/Alesha/core_greml/GS.fam

# Have same order for pheno
library(tidyverse)
GS.fam <- read.table("/Local_Scratch/Alesha/core_greml/GS.fam", header=F)
pheno <- read.table("/Local_Scratch/Alesha/DNAm/phenotypes/greml.pheno/ds_pheno.txt", header=T)
pheno_order <- GS.fam %>% select(V1, V2) %>% left_join(pheno, by=c("V1"="IID_DNAm"))

write.table(pheno_order %>% select(V1, V2, bmi_RankNorm), file="/Local_Scratch/Alesha/core_greml/bmi.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, whr_RankNorm), file="/Local_Scratch/Alesha/core_greml/whr.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, body_fat_RankNorm), file="/Local_Scratch/Alesha/core_greml/body_fat.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, Glucose_RankNorm), file="/Local_Scratch/Alesha/core_greml/glucose.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, HDL_cholesterol_RankNorm), file="/Local_Scratch/Alesha/core_greml/HDL.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, Total_cholesterol_RankNorm), file="/Local_Scratch/Alesha/core_greml/Total_cholesterol.txt", col.names=F, row.names=F, quote=F)
write.table(pheno_order %>% select(V1, V2, height_RankNorm), file="/Local_Scratch/Alesha/core_greml/height.txt", col.names=F, row.names=F, quote=F)

   

 # i. ensure positive definite kernel matrices
cd /Local_Scratch/Alesha/core_greml
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -bg GS.grm -thread 80 -bend 2
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -bg GS.mrm -thread 80 -bend 2

# ii. Cholesky decomposition
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -g GS.grm.bend -thread 80 -chol 1
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -g GS.mrm.bend -thread 80 -chol 1

# iii. compute Q
echo GS.grm.bend.chol > GS_grm_mrm.chol
echo GS.mrm.bend.chol >> GS_grm_mrm.chol
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -mg GS_grm_mrm.chol -thread 80 -matmat 2

# 3. Parameter estimation
# i. model fitting

echo GS.grm > GS_greml.matlist
echo GS.mrm >> GS_greml.matlist

echo GS.grm > GS_coregreml.matlist
echo GS.mrm >> GS_coregreml.matlist
echo GS_grm_mrm.chol.matmat2 >> GS_coregreml.matlist


for trait in bmi whr body_fat glucose HDL Total_cholesterol height 
do

# GREML
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -mbg GS_greml.matlist -d ${trait}.txt -mod 1 -thread 100 -out GS_greml_${trait}.out > GS_greml_${trait}.log

# CORE-GREML
/Cluster_Filespace/Marioni_Group/Alesha/./mtg2 -p GS.fam -mbg GS_coregreml.matlist -d ${trait}.txt -mod 1 -sv test.sv -thread 100 -out GS_coregreml_${trait}.out > GS_coregreml_${trait}.log

done

pchisq(-2*(-3019.4460+3018.9473), df=1, lower.tail=F) # bmi 0.3179405
pchisq(-2*(-3437.2856+3437.2745), df=1, lower.tail=F) # whr 0.8815563
pchisq(-2*(-3019.4264+3019.1763), df=1, lower.tail=F) # body fat 0.4794123
pchisq(-2*(-3610.9032+3610.8027), df=1, lower.tail=F) # glucose 0.6539149
pchisq(-2*(-3368.8995+3368.7420), df=1, lower.tail=F) # HDL 0.5746281
pchisq(-2*(-3625.4130+3625.3529), df=1, lower.tail=F) # Total 0.7288177
pchisq(-2*(-3637.2051+3635.8549), df=1, lower.tail=F) # height 0.1003231

```





```{r}
# LBCs
LBC_sampleID <- read.csv("~/Documents/2_project/4_GS/1_Correlations/manuscript/12_2022/LBC_sampleID.txt", sep="")
info <- read.csv("~/Downloads/information.for.final.txt")
final <- read.csv("~/Downloads/final.txt", header=F, sep=" ")

LBC_sampleID %>% head
info %>% head
final %>% head

info %>% filter(LBC.subject.no %in% LBC_sampleID$IID | bloodnum..LBC1921.w1.only. %in% LBC_sampleID$IID) %>% dim

# 1403
LBC_keep <- final %>% filter(V3 %in% LBC_sampleID$IID) # 1437
Daniel_exclude <- c("LBC360558", "LBC360760", "LBC360536", "LBC361272", "LBC360213", "LBC360262", "LBC361264", "LBC361030", "LBC360412", "LBC361076", "LBC360721")

info %>% filter(LBC.subject.no %in% Daniel_exclude & barcode %in% LBC_keep$V1)
# We retained 9 of the samples Daniel excluded
# What are these?
unknown <- LBC_sampleID %>% filter( !(IID %in% info$LBC.subject.no) & !(IID %in% info$bloodnum..LBC1921.w1.only.) )
unknown %>% dim # 34

excluded <- info %>% filter(!(bloodnum..LBC1921.w1.only. %in% LBC_sampleID$IID) & !(LBC.subject.no %in% LBC_sampleID$IID)) # 81

# We have LBC wave 1 data for 1484 people in total (Tian's RDM)
# I have included 1437 people in my analysis
# Of these, 1403 are definitely in the wave1 list (bloodnum..LBC1921.w1.only. or LBC.subject.no)
# There are 34 individuals I have included in my analysis which I can not trace back to wave1
# There were 81 individuals in wave1 which appear to have been filtered out in my analysis

LBC_sampleID %>% filter(!IID %in% info$LBC.subject.no) %>% head 



info[,2] %>% head
```

